<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>TCGA-Nigerian-RNASeq Project</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">TCGA-Nigerian-RNASeq Project</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-06-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 1 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>~/Research-Local/2019-rnaseq/TCGA-Nigerian-RNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomparajagoTCGANigerianRNAseqtree965fa5aafe42392b49e984b0b57b72d1e9f94c40targetblank965fa5aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/tree/965fa5aafe42392b49e984b0b57b72d1e9f94c40" target="_blank">965fa5a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomparajagoTCGANigerianRNAseqtree965fa5aafe42392b49e984b0b57b72d1e9f94c40targetblank965fa5aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/figure/.DS_Store
    Ignored:    plots/.DS_Store

Untracked files:
    Untracked:  NigerianTCGArawcountsDeSeq2-pc2.Rmd
    Untracked:  NigerianTCGArawcountsDeSeq2-proteincoding-IHC.Rmd
    Untracked:  NigerianTCGArawcountslimma-voomDE-PAM50-linc.Rmd
    Untracked:  NigerianTCGArawcountslimma-voomDE-PAM50.Rmd
    Untracked:  NigerianTCGArawcountslimma-voomDE-PAM50_cache/
    Untracked:  _site.yml
    Untracked:  about.Rmd
    Untracked:  docs/figure/NigerianTCGArawcountslimma-voomDE-PAM50.Rmd/
    Untracked:  plots/Nanostring-genes-in-RNAseq-results.jpeg

Unstaged changes:
    Modified:   .Rhistory
    Deleted:    analysis/NigerianTCGArawcountsDeSeq2-pc2.Rmd
    Deleted:    analysis/NigerianTCGArawcountsDeSeq2-proteincoding-IHC.Rmd
    Deleted:    analysis/NigerianTCGArawcountslimma-voomDE-PAM50.Rmd
    Deleted:    analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Modified:   code/proteincodingparse.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/965fa5aafe42392b49e984b0b57b72d1e9f94c40/analysis/index.Rmd" target="_blank">965fa5a</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-12-04
</td>
<td>
renamed Perou project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/83061c68cbc1b08a26f0946dd35ef061cdf9cd2a/analysis/index.Rmd" target="_blank">83061c6</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-08-19
</td>
<td>
Differential expression with IHC and LOH subanalysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/9fc69d9422c2383ba1e592f5306824b6f3dd3906/analysis/index.Rmd" target="_blank">9fc69d9</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-06-23
</td>
<td>
diff exp analysis limma voom
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/c7160b426e119663dcd352a580df04407057e3d5/analysis/index.Rmd" target="_blank">c7160b4</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-06-17
</td>
<td>
Updated quantile normalization
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/1b2b589826cd88e354cc43649ac394ec0f7767b6/analysis/index.Rmd" target="_blank">1b2b589</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-05-08
</td>
<td>
batch effect assessment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/80f9f32d29f9e5576471c3309d953608f577f341/analysis/index.Rmd" target="_blank">80f9f32</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-03-11
</td>
<td>
read counts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/303135894e957747263e7211bf29d9d74eacdbbc/analysis/index.Rmd" target="_blank">3031358</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-03-06
</td>
<td>
Analysis repeated with protein coding only
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/44f9151ff87c7e5ac9629473a2d0e5ac6ffd4963/analysis/index.Rmd" target="_blank">44f9151</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-02-07
</td>
<td>
not adding tempus here
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/63cf3641c57bf9cfdfee9830efb775ecd487a3a1/analysis/index.Rmd" target="_blank">63cf364</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-02-07
</td>
<td>
adding Tempus site info
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/8600fc69f0fb914f88dcd96ca9355ad70eda3b53/analysis/index.Rmd" target="_blank">8600fc6</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-01-22
</td>
<td>
removed normalization description from main page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/d47a998f13b96ba2dab3394fb2b26251eb015be4/analysis/index.Rmd" target="_blank">d47a998</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-01-22
</td>
<td>
Cleanup of website, no formal changes made
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/c927d91afe56f0e449f9d6ce6e3e8ffcae080ea0/analysis/index.Rmd" target="_blank">c927d91</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-01-16
</td>
<td>
normalization to correct link site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/56355691ebddd476e09fd33762e5b165b4008c50/analysis/index.Rmd" target="_blank">5635569</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-01-16
</td>
<td>
updated index with links
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/df3d95c16ea9c3ec811ad6142b6cac56d2468275/analysis/index.Rmd" target="_blank">df3d95c</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-01-16
</td>
<td>
trying to upload data for lab meeting
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/0ad8ebc0b6566ff025a638847f5624863c92be77/analysis/index.Rmd" target="_blank">0ad8ebc</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-01-16
</td>
<td>
adding DeSeq data to date
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/c3c1e743aec418ed3c294e6f332dd4e3e580fdf1/analysis/index.Rmd" target="_blank">c3c1e74</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-01-16
</td>
<td>
“Updated workflowR with basic information”
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/fcaf7a23e445dcce4e41640f730b89f457b27398/analysis/index.Rmd" target="_blank">fcaf7a2</a>
</td>
<td>
Sheila Rajagopal
</td>
<td>
2019-01-14
</td>
<td>
WorkflowR structure added
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Padma Sheila Rajagopal, MD MPH</p>
<p>Olopade Lab / Im Lab Postdoctoral Fellow</p>
<p>This is the site I will use for a project in which I am assessing differential expression between a population of Nigerian breast cancer patients and breast cancer patients from the TCGA.</p>
<p>Workflow and analysis published here:</p>
<p>#Protocols * <em>Alignment</em>: STAR (completed prior to analysis by Dominic Fitzgerald), initial steps of [PsychENCODE pipeline]: (<a href="https://gist.github.com/djf604/a3d6925c3775d6cf1053ded0e7a8b148" class="uri">https://gist.github.com/djf604/a3d6925c3775d6cf1053ded0e7a8b148</a>)</p>
<ul>
<li><p><em>Raw read counts</em>: HTSeq (completed prior to analysis by Dominic Fitzgerald)</p></li>
<li><p><em>Differential expression</em>: Limma-voom w/ quantile normalization</p></li>
</ul>
<p>#Differential expression: * Batch effect correction exploration: <a href="https://parajago.github.io/TCGA-Nigerian-RNAseq/NigerianTCGArawcountsDeSeq2-pc2.html" class="uri">https://parajago.github.io/TCGA-Nigerian-RNAseq/NigerianTCGArawcountsDeSeq2-pc2.html</a></p>
<ul>
<li><p>Differential expression analysis (protein coding genes only) comparing directly TCGA and Nigerian samples: <a href="https://parajago.github.io/TCGA-Nigerian-RNAseq/NigerianTCGArawcountslimma-voomDE-PAM50.html" class="uri">https://parajago.github.io/TCGA-Nigerian-RNAseq/NigerianTCGArawcountslimma-voomDE-PAM50.html</a></p></li>
<li><p>Differential expression analysis (protein coding genes only) using samples with whole genome sequencing available as part of an accompanying project to the Life History Analysis: <a href="https://parajago.github.io/TCGA-Nigerian-RNAseq/NigerianTCGArawcountsDeSeq2-proteincoding-IHC.html" class="uri">https://parajago.github.io/TCGA-Nigerian-RNAseq/NigerianTCGArawcountsDeSeq2-proteincoding-IHC.html</a></p></li>
</ul>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
