<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Padma Sheila Rajagopal, MD MPH" />


<title>Nigerian-TCGA Differential Expression Analysis Using DeSeq2</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Olopade Lab: TCGA-Nigerian-RNAseq Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="NigerianTCGArawcountsDeSeq2.html">Differential Expression Analysis</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/parajago/TCGA-Nigerian-RNAseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Nigerian-TCGA Differential Expression Analysis Using DeSeq2</h1>
<h4 class="author"><em>Padma Sheila Rajagopal, MD MPH</em></h4>
<h4 class="date"><em>2/19/2019</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-02-26</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20190113)</code> </summary></p>
<p>The command <code>set.seed(20190113)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/tree/44f9151ff87c7e5ac9629473a2d0e5ac6ffd4963" target="_blank">44f9151</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/figure/.DS_Store
    Ignored:    plots/.DS_Store

Unstaged changes:
    Modified:   analysis/NigerianTCGArawcountsDeSeq2.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/d131c0dbfb62cc8d44b3856323bf63ba77a38373/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">d131c0d</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-02-07
</td>
<td style="text-align:left;">
check for differential expression for TCGA black/white
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/81057beabb2162a8de45383c340f611744cdfe08/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">81057be</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-02-06
</td>
<td style="text-align:left;">
removed sensitive files from public
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
full analysis 1/30 with race/subset typing and initial batch correction applied
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/b237165f9643faff44320705011faeb4d7e21cce/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
full analysis 1/30 with race/subset typing and initial batch correction applied
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/39e89b8bf84293c89d0b87a82a6b12fcc07bd614/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">39e89b8</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
breast cancer subtype information updated
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/39e89b8bf84293c89d0b87a82a6b12fcc07bd614/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">39e89b8</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
breast cancer subtype information updated
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/bb42865c36f25e0657a0881f01403860a27236b9/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">bb42865</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
breast cancer subtype information added
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/bb42865c36f25e0657a0881f01403860a27236b9/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">bb42865</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
breast cancer subtype information added
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/45f3be0ed3e9bfa9a545b381c78eb70f96eb3482/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">45f3be0</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
<td style="text-align:left;">
cleaned files and DEanalysis currently contains only through sva of batch effect given errors
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/45f3be0ed3e9bfa9a545b381c78eb70f96eb3482/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">45f3be0</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
<td style="text-align:left;">
cleaned files and DEanalysis currently contains only through sva of batch effect given errors
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
<td style="text-align:left;">
batch effect correction implemented
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
<td style="text-align:left;">
batch effect correction implemented
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/616d47012ac13731ce150d005296f7bcdcac52b9/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">616d470</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
<td style="text-align:left;">
Batch effect corrected noted in Rmd file
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/d47a998f13b96ba2dab3394fb2b26251eb015be4/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">d47a998</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-22
</td>
<td style="text-align:left;">
Cleanup of website, no formal changes made
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/d47a998f13b96ba2dab3394fb2b26251eb015be4/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">d47a998</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-22
</td>
<td style="text-align:left;">
Cleanup of website, no formal changes made
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/cd53cd0d2448d0eb970cffed8785f32d085fb4ff/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">cd53cd0</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-16
</td>
<td style="text-align:left;">
initial diff exp pass
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/0ad8ebc0b6566ff025a638847f5624863c92be77/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">0ad8ebc</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-16
</td>
<td style="text-align:left;">
adding DeSeq data to date
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/c3c1e743aec418ed3c294e6f332dd4e3e580fdf1/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">c3c1e74</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-16
</td>
<td style="text-align:left;">
“Updated workflowR with basic information”
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="translation-from-htseq---deseq2-table" class="section level1">
<h1>Translation from HTSeq -&gt; DESeq2 table</h1>
<p>I have 86 TCGA patients with whole-genome sequencing data and RNAseq data as well as 99 Nigerian patients with RNA-seq data. Raw counts were initially processed using HTSeq, so HTSeq data is being formatted for use with DESeq2.</p>
<pre class="r"><code>FOLDER &lt;- &quot;/Users/parajago/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants&quot;
sampleFiles &lt;- grep(&quot;htseq.counts&quot;,list.files(FOLDER),value=TRUE)

#Differential gene expression setup based on race (b/w/other)
sampleConditionrace &lt;- sampleFiles
countVar2=1
for (sample in sampleConditionrace){
  if (stri_detect_fixed(sample,&quot;LIB&quot;)==TRUE){
    sampleConditionrace[countVar2] &lt;- &quot;Nigerian&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;black&quot;)==TRUE){
    sampleConditionrace[countVar2] &lt;- &quot;TCGA_black&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;white&quot;)==TRUE){
    sampleConditionrace[countVar2] &lt;- &quot;TCGA_white&quot;
    countVar2=countVar2+1
  } else{
    sampleConditionrace[countVar2] &lt;- &quot;TCGA_other&quot;
    countVar2=countVar2+1
  }
}

#Condition based on PAM50 subtype 
sampleConditionPAM50 &lt;- sampleFiles
countVar2=1
for (sample in sampleConditionPAM50){
  if (stri_detect_fixed(sample,&quot;Her2&quot;)==TRUE){
    sampleConditionPAM50[countVar2] &lt;- &quot;Her2&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;Basal&quot;)==TRUE){
    sampleConditionPAM50[countVar2] &lt;- &quot;Basal&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;LumA&quot;)==TRUE){
    sampleConditionPAM50[countVar2] &lt;- &quot;LumA&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;LumB&quot;)==TRUE){
    sampleConditionPAM50[countVar2] &lt;- &quot;LumB&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;PAMNL&quot;)==TRUE){
    sampleConditionPAM50[countVar2] &lt;- &quot;Normal&quot;
    countVar2=countVar2+1
  } else{
    sampleConditionPAM50[countVar2] &lt;- &quot;PAM_other&quot;
    countVar2=countVar2+1
  }
}

#Condition based on batch (relevant to the Nigerian patients only; no difference in batch for the TCGA patients)
batchval &lt;- sampleFiles
countVar2=1
for (sample in batchval){
  if (stri_detect_fixed(sample,&quot;batch1&quot;)==TRUE){
    batchval[countVar2] &lt;- &quot;batch1&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;batch23&quot;)==TRUE){
    batchval[countVar2] &lt;- &quot;batch23&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;batch4&quot;)==TRUE){
    batchval[countVar2] &lt;- &quot;batch4&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;batch5&quot;)==TRUE){
    batchval[countVar2] &lt;- &quot;batch5&quot;
    countVar2=countVar2+1
  } else{
    batchval[countVar2] &lt;- &quot;batchT&quot;
    countVar2=countVar2+1
  }
}

table(sampleConditionrace, sampleConditionPAM50)</code></pre>
<pre><code>                   sampleConditionPAM50
sampleConditionrace Basal Her2 LumA LumB Normal PAM_other
         Nigerian      32   26   16   18      7         0
         TCGA_black    25    0    4    4      0         0
         TCGA_other     0    0    0    0      0        14
         TCGA_white    17    5    8    9      0         0</code></pre>
<pre class="r"><code>sampleTable2 &lt;- data.frame(sampleName=gsub(&quot;.htseq.counts&quot;,&quot;&quot;,sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          batch=batchval)

sampleTable2$sampleCondition &lt;- paste(sampleTable2$condition1, sampleTable2$condition2, sep=&quot;.&quot;)

ddsHTSeqMF &lt;- DESeqDataSetFromHTSeqCount(sampleTable=sampleTable2,
                                       directory=FOLDER,
                                       design=~sampleCondition)</code></pre>
</div>
<div id="variance-transformation-batch-effect-assessment-and-initial-exploratory-analysis" class="section level1">
<h1>Variance transformation, batch effect assessment and initial exploratory analysis</h1>
<p>I am now performing different visualization approaches to assess the variance of the data, batch effect and expected differences between the two populations.</p>
<pre class="r"><code>ddsHTSeqMF &lt;- ddsHTSeqMF[rowSums(counts(ddsHTSeqMF)) &gt; 1, ] #Pre-filtering the dataset by removing the rows without information about gene expression

dds &lt;- estimateSizeFactors(ddsHTSeqMF) #The size factor is the median ratio of the sample over a &quot;pseudosample&quot;: for each gene, the geometric mean of all samples.
dds$sampleCondition &lt;- relevel(dds$sampleCondition, ref = &quot;Nigerian.Basal&quot;) #explicitly making the Nigerian Basal patients the reference population

vsd &lt;- vst(ddsHTSeqMF, blind = TRUE) #Variance-stabilizing transformation

df &lt;- bind_rows( #Graphical representation of the variance-stabilizing transformation
          as_data_frame(log2(counts(dds, normalized=TRUE)[, 1:2]+1)) %&gt;%
          mutate(transformation = &quot;log2(x + 1)&quot;),
          as_data_frame(assay(vsd)[, 1:2]) %&gt;% mutate(transformation = &quot;vst&quot;))
          colnames(df)[1:2] &lt;- c(&quot;x&quot;, &quot;y&quot;)

ggplot(df, aes(x = x, y = y)) + geom_hex(bins = 80) + coord_fixed() + facet_grid( . ~ transformation)</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance%20transformation%20and%20batch%20effect%20assessment-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Variance transformation and batch effect assessment-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/45f3be0ed3e9bfa9a545b381c78eb70f96eb3482/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance transformation and batch effect assessment-1.png" target="_blank">45f3be0</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance transformation and batch effect assessment-1.png" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Figure 1: Graphical representation of the variance-stabilizing transformation relative to the standard transformation of log2(x+1).This reduces variance purely due to low read counts</p>
<pre class="r"><code>#PCAs at this point of the analysis are all of the HTSeq-input raw counts
plotPCA(vsd, intgroup=c(&quot;condition1&quot;, &quot;batch&quot;)) </code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Batch%20effect%20assessment-PCAs-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Batch effect assessment-PCAs-1-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Batch effect assessment-PCAs-1-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>PCA 1-1:PCA demonstrating the raw difference between Nigerian and TCGA patients, which would be expected given that these experiments were done in different settings at different times. Nigerian patients do not appear to have significant differneces based on their batches. (TCGA patients are known not to have significant difference based on batch performed.) However, there does appear to be some clustering mechanism that is separating these TCGA patients into two groups that is not race dependent.</p>
<pre class="r"><code>plotPCA(vsd, intgroup=c(&quot;condition2&quot;))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Batch%20effect%20assessment-PCAs-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Batch effect assessment-PCAs-2-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Batch effect assessment-PCAs-2-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>PCA 1-2:PCA demonstrating association with breast cancer subtype and gene expression. This is noted in both groups and not likely due to batch effect given known biology.</p>
<pre class="r"><code>plotPCA(vsd, intgroup=c(&quot;condition1&quot;, &quot;condition2&quot;))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Batch%20effect%20assessment-PCAs-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Batch effect assessment-PCAs-3-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Batch effect assessment-PCAs-3-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>PCA 1-3: Integrated PCA demonstrating combined effect of breast cancer subtype and race. Subtype appears to account for the clustering seen in the TCGA group.</p>
<pre class="r"><code>#Heatmap to assess clusters pre-analysis
topVarGenes &lt;- head(order(rowVars(assay(vsd)), decreasing = TRUE), 100)
hm  &lt;- assay(vsd)[ topVarGenes, ]
hm  &lt;- hm - rowMeans(hm)
anno &lt;- as.data.frame(colData(vsd)[, c(&quot;condition1&quot;, &quot;condition2&quot;)])
rownames(anno) &lt;- colnames(vsd)
pheatmap(hm, annotation_col = anno, cluster_rows=TRUE, cluster_cols=TRUE, clustering_distance_rows = &quot;euclidean&quot;, clustering_distance_cols = &quot;euclidean&quot;, clustering_method = &quot;complete&quot;, show_rownames=FALSE, show_colnames = FALSE, main=&quot;Heat map: Relative VST-transformed\ncounts across samples before batch correction&quot;)</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Heatmap%20to%20assess%20clusters%20pre-analysis-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Heatmap to assess clusters pre-analysis-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Heatmap to assess clusters pre-analysis-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Heatmap 1-1: This heatmaps is only using the top 100 genes and is competely heirarchically clustered. These clusters appear consistent with the PCA results.</p>
<pre class="r"><code>#PCA2: Removing the batch effect created by the separate batch processing of the TCGA and Nigerian samples (although this may overcorrect for the actual difference between these two populations).  
mat &lt;- assay(vsd) 
mat &lt;- limma::removeBatchEffect(mat, vsd$condition1, vsd$batch)</code></pre>
<pre><code>Coefficients not estimable: batch24 </code></pre>
<pre class="r"><code>assay(vsd) &lt;- mat
plotPCA(vsd, intgroup=c(&quot;condition1&quot;, &quot;condition2&quot;))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Batch%20effect%20assessment-PCAs-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Batch effect assessment-PCAs-4-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Batch effect assessment-PCAs-4-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>PCA 2-1: We have accounted for batch effect in this PCA only using the limma removeBatchEffect method.This largely accounts for the batch effect noticed in PCA1-Nigerian vs TCGA. There remain some outliers, primarily from the Nigerian patients.</p>
<pre class="r"><code>#Heatmap to assess clusters post-analysis
topVarGenes &lt;- head(order(rowVars(assay(vsd)), decreasing = TRUE), 100)
hm  &lt;- assay(vsd)[ topVarGenes, ]
hm  &lt;- hm - rowMeans(hm)
anno &lt;- as.data.frame(colData(vsd)[, c(&quot;condition1&quot;, &quot;condition2&quot;)])
rownames(anno) &lt;- colnames(vsd)
pheatmap(hm, annotation_col = anno, cluster_rows=TRUE, cluster_cols=TRUE, clustering_distance_rows = &quot;euclidean&quot;, clustering_distance_cols = &quot;euclidean&quot;, clustering_method = &quot;complete&quot;, show_rownames=FALSE, show_colnames = FALSE, main=&quot;Heat map: Relative VST-transformed\ncounts across samples after batch correction&quot;)</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Heatmap%20to%20assess%20clusters%20after%20batch%20correction-1.png" width="672" style="display: block; margin: auto;" /> Heatmap 2-1: This heatmap after batch correction with limma’s removeBatchEffect method is only using the top 100 genes and is competely heirarchically clustered. These clusters appear consistent with the PCA results as well with clustering enabled between the breast cancer subtypes and finer interspersing of the Nigerian vs. TCGA characteristic based on breast cancer subtype.</p>
<pre class="r"><code>boxplot(log2(counts(dds)), las=2, ylab=&quot;log2(Raw Counts)&quot;, xlab=&quot;Samples&quot;, main=&quot;Boxplot of raw counts before\n batch correction/transformation&quot;)</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/:%20Boxplots%20of%20counts%20before%20and%20after%20DESeq2%20normalization-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of : Boxplots of counts before and after DESeq2 normalization-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/: Boxplots of counts before and after DESeq2 normalization-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>boxplot(log2(counts(dds, normalized=TRUE)), las=2, ylab=&quot;log2(Raw Counts)&quot;, xlab=&quot;Samples&quot;, main=&quot;Boxplot of batch corrected/transformed counts&quot;)</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/:%20Boxplots%20of%20counts%20before%20and%20after%20DESeq2%20normalization-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of : Boxplots of counts before and after DESeq2 normalization-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/: Boxplots of counts before and after DESeq2 normalization-2.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>These boxplots demonstrate the effect of the DESeq2 normalization process. The normalized DESeq2 values are incorporated into the batch corrected model (as below).</p>
</div>
<div id="using-sva-to-account-for-batch-effect-related-to-the-nigeriantcga-samples" class="section level1">
<h1>Using sva to account for batch effect related to the Nigerian/TCGA samples</h1>
<pre class="r"><code>dat  &lt;- counts(dds, normalized = TRUE)
idx  &lt;- rowMeans(dat) &gt; 1
dat  &lt;- dat[idx, ]
mod  &lt;- model.matrix(~condition1, colData(dds))
mod0 &lt;- model.matrix(~   1, colData(dds))
svseq &lt;- svaseq(dat, mod, mod0, n.sv = 2)</code></pre>
<pre><code>Number of significant surrogate variables is:  2 
Iteration (out of 5 ):1  2  3  4  5  </code></pre>
<pre class="r"><code>sampleTablebe &lt;- data.frame(sampleName=gsub(&quot;.htseq.counts&quot;,&quot;&quot;,sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          be1=svseq$sv[,1],
                          be2=svseq$sv[,2],
                          batch=batchval)

sampleTablebe$sampleCondition &lt;- paste(sampleTablebe$condition1, sampleTablebe$condition2, sep=&quot;.&quot;)

ddsHTSeqMFbe &lt;- DESeqDataSetFromHTSeqCount(sampleTable=sampleTablebe,
                                       directory=FOLDER,
                                       design=~sampleCondition + be1 + be2)

ddsHTSeqMFbe &lt;- ddsHTSeqMFbe[rowSums(counts(ddsHTSeqMFbe)) &gt; 1, ]</code></pre>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential expression analysis</h1>
<p>This is the actual differential expression analysis with false discovery threshold of 0.05. I am performing this for both the standard and batch-effect-corrected versions of the DESeq2 model to allow comparison.</p>
<pre class="r"><code>ddsMF &lt;- DESeq(ddsHTSeqMF) #This will refer to a version that does not include batch correction outside of what is accomplished through DESeq2 alone. 
ddsMFbe &lt;-DESeq(ddsHTSeqMFbe) #This will refer to a version that batch-corrects using sva for 2 variables and overtly acknowledges that the TCGA-Nigerian platform is part of this batch effect issue. 

resultsNames(ddsMF)</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                             
 [2] &quot;sampleCondition_Nigerian.Her2_vs_Nigerian.Basal&quot;       
 [3] &quot;sampleCondition_Nigerian.LumA_vs_Nigerian.Basal&quot;       
 [4] &quot;sampleCondition_Nigerian.LumB_vs_Nigerian.Basal&quot;       
 [5] &quot;sampleCondition_Nigerian.Normal_vs_Nigerian.Basal&quot;     
 [6] &quot;sampleCondition_TCGA_black.Basal_vs_Nigerian.Basal&quot;    
 [7] &quot;sampleCondition_TCGA_black.LumA_vs_Nigerian.Basal&quot;     
 [8] &quot;sampleCondition_TCGA_black.LumB_vs_Nigerian.Basal&quot;     
 [9] &quot;sampleCondition_TCGA_other.PAM_other_vs_Nigerian.Basal&quot;
[10] &quot;sampleCondition_TCGA_white.Basal_vs_Nigerian.Basal&quot;    
[11] &quot;sampleCondition_TCGA_white.Her2_vs_Nigerian.Basal&quot;     
[12] &quot;sampleCondition_TCGA_white.LumA_vs_Nigerian.Basal&quot;     
[13] &quot;sampleCondition_TCGA_white.LumB_vs_Nigerian.Basal&quot;     </code></pre>
<pre class="r"><code>resultsNames(ddsMFbe)</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                             
 [2] &quot;sampleCondition_Nigerian.Her2_vs_Nigerian.Basal&quot;       
 [3] &quot;sampleCondition_Nigerian.LumA_vs_Nigerian.Basal&quot;       
 [4] &quot;sampleCondition_Nigerian.LumB_vs_Nigerian.Basal&quot;       
 [5] &quot;sampleCondition_Nigerian.Normal_vs_Nigerian.Basal&quot;     
 [6] &quot;sampleCondition_TCGA_black.Basal_vs_Nigerian.Basal&quot;    
 [7] &quot;sampleCondition_TCGA_black.LumA_vs_Nigerian.Basal&quot;     
 [8] &quot;sampleCondition_TCGA_black.LumB_vs_Nigerian.Basal&quot;     
 [9] &quot;sampleCondition_TCGA_other.PAM_other_vs_Nigerian.Basal&quot;
[10] &quot;sampleCondition_TCGA_white.Basal_vs_Nigerian.Basal&quot;    
[11] &quot;sampleCondition_TCGA_white.Her2_vs_Nigerian.Basal&quot;     
[12] &quot;sampleCondition_TCGA_white.LumA_vs_Nigerian.Basal&quot;     
[13] &quot;sampleCondition_TCGA_white.LumB_vs_Nigerian.Basal&quot;     
[14] &quot;be1&quot;                                                   
[15] &quot;be2&quot;                                                   </code></pre>
<pre class="r"><code>#MAPlot: log2 fold changes attributable to the condition over the mean of normalized counts for all the samples in the DESeqDataSet. Points will be colored red if the adjusted p value is less than 0.1. Points which fall out of the window are plotted as open triangles pointing either up or down.
cat(&quot;MA Plot 1: Original differential expression&quot;)</code></pre>
<pre><code>MA Plot 1: Original differential expression</code></pre>
<pre class="r"><code>res &lt;- lfcShrink(ddsMF, coef=&quot;sampleCondition_TCGA_white.Basal_vs_Nigerian.Basal&quot;, type=&quot;apeglm&quot;)
DESeq2::plotMA(res, ylim=c(-10,10), xlim=c(0.1,200))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Differential%20expression%20analysis%20and%20visualization-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Differential expression analysis and visualization-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Differential expression analysis and visualization-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Differential expression analysis and visualization-1.png" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>cat(&quot;MA Plot 2: Batch-corrected differential expression&quot;)</code></pre>
<pre><code>MA Plot 2: Batch-corrected differential expression</code></pre>
<pre class="r"><code>res2 &lt;- lfcShrink(ddsMFbe, coef=&quot;sampleCondition_TCGA_white.Basal_vs_Nigerian.Basal&quot;, type=&quot;apeglm&quot;)
DESeq2::plotMA(res2, ylim=c(-10,10), xlim=c(0.1,200))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Differential%20expression%20analysis%20and%20visualization-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Differential expression analysis and visualization-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Differential expression analysis and visualization-2.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Differential expression analysis and visualization-2.png" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>This is just an MA-plot demonstration of the efficacy of the batch effect correction (using TCGA-white basal breast cancer patients vs Nigerian basal breast cancer patients as an example). The second MA plot with batch-corrected differential expression looks like it is more appropriately assessing significantly differentially expressed genes between these two populations.</p>
<p>This is a check between the TCGA black and white patients to verify this differential expression analysis based on prior work by Stewart et al (PLOS One 2013) and Huo et al (JAMA Oncology 2017). This was only performed with the basal patients as this was the largest subsample.</p>
<pre class="r"><code>diffTCGA&lt;- results(ddsMFbe, contrast=c(&quot;sampleCondition&quot;, &quot;TCGA_black.Basal&quot;, &quot;TCGA_white.Basal&quot;), alpha=0.05)

diffTCGA &lt;- subset(diffTCGA, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
diffTCGA$temp &lt;- row.names(diffTCGA)
diffTCGA$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, diffTCGA$temp)

diffTCGA$symbol &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

diffTCGA$biotype &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column=&quot;GENEBIOTYPE&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

diffTCGA$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

diffTCGA$locstart &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column=&quot;GENESEQSTART&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

diffTCGA$locend &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column=&quot;GENESEQEND&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

with(diffTCGA, plot(log2FoldChange, -log10(padj), pch=20, main=&quot;Volcano plot of differential gene expression between basal \nbreast cancers in TCGA black and \nTCGA white breast cancer patients&quot;, xlim=c(-50,50), ylim=c(0,70)))
with(subset(diffTCGA, diffTCGA$biotype==&quot;protein_coding&quot; &amp; padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;blue&quot;))
with(subset(diffTCGA, padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/DE:%20TCGA%20black%20and%20TCGA%20white-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>diffTCGAres &lt;- as.data.frame(diffTCGA)[, ]
write.csv(diffTCGAres, file = &quot;TCGAblackwhite-basal-test.csv&quot;)</code></pre>
<p>This is a comparison of 32 Nigerian basal cases and 17 TCGA white basal cases.</p>
<pre class="r"><code>resBasal&lt;- results(ddsMFbe, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.Basal&quot;, &quot;TCGA_white.Basal&quot;), alpha=0.05)
resBasal &lt;- subset(resBasal, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resBasal$temp &lt;- row.names(resBasal)
resBasal$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resBasal$temp)

resBasal$symbol &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resBasal$biotype &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column=&quot;GENEBIOTYPE&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resBasal$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resBasal$locstart &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column=&quot;GENESEQSTART&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

resBasal$locend &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column=&quot;GENESEQEND&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

with(resBasal, plot(log2FoldChange, -log10(padj), pch=20, main=&quot;Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and \nTCGA white breast cancer patients&quot;, xlim=c(-50,50), ylim=c(0,70)))
with(subset(resBasal, resBasal$biotype==&quot;protein_coding&quot; &amp; padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;blue&quot;))
with(subset(resBasal, padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/DE:%20Basal%20breast%20-%20Nigerian%20and%20TCGA%20white-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of DE: Basal breast - Nigerian and TCGA white-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/DE: Basal breast - Nigerian and TCGA white-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>diffNigerianwhitebasal &lt;- as.data.frame(resBasal)[, ]
write.csv(diffNigerianwhitebasal, file = &quot;Nigerian-white-basal.csv&quot;)</code></pre>
<p>This is a comparison of 32 Nigerian basal cases and 25 TCGA black basal cases.</p>
<pre class="r"><code>resBasalblack&lt;- results(ddsMFbe, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.Basal&quot;, &quot;TCGA_black.Basal&quot;), alpha=0.05)
resBasalblack &lt;- subset(resBasalblack, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resBasalblack$temp &lt;- row.names(resBasalblack)
resBasalblack$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resBasalblack$temp)

resBasalblack$symbol &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resBasalblack$biotype &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column=&quot;GENEBIOTYPE&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resBasalblack$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resBasalblack$locstart &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column=&quot;GENESEQSTART&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

resBasalblack$locend &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column=&quot;GENESEQEND&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

with(resBasalblack, plot(log2FoldChange, -log10(padj), pch=20, main=&quot;Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and \nTCGA black breast cancer patients&quot;, xlim=c(-50,50), ylim=c(0,70)))
with(subset(resBasalblack, resBasalblack$biotype==&quot;protein_coding&quot; &amp; padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;blue&quot;))
with(subset(resBasalblack, padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/DE:%20Basal%20breast%20-%20Nigerian%20and%20TCGA%20black-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of DE: Basal breast - Nigerian and TCGA black-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/DE: Basal breast - Nigerian and TCGA black-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>diffNigerianblackbasal &lt;- as.data.frame(resBasalblack)[, ]
write.csv(diffNigerianblackbasal, file = &quot;Nigerian-black-basal.csv&quot;)</code></pre>
<p>This is a comparison of 26 Nigerian Her2 cases and 5 TCGA white Her2 cases. There are no TCGA black Her2 cases available for this analysis.</p>
<pre class="r"><code>resHer2&lt;- results(ddsMFbe, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.Her2&quot;, &quot;TCGA_white.Her2&quot;), alpha=0.05)
resHer2 &lt;- subset(resHer2, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resHer2$temp &lt;- row.names(resHer2)
resHer2$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resHer2$temp)

resHer2$symbol &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resHer2$biotype &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column=&quot;GENEBIOTYPE&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resHer2$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resHer2$locstart &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column=&quot;GENESEQSTART&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

resHer2$locend &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column=&quot;GENESEQEND&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

with(resHer2, plot(log2FoldChange, -log10(padj), pch=20, main=&quot;Volcano plot of differential gene expression between Her2 \nbreast cancers in Nigerian and \nTCGA white breast cancer patients&quot;, xlim=c(-50,50), ylim=c(0,70)))
with(subset(resHer2, resHer2$biotype==&quot;protein_coding&quot; &amp; padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;blue&quot;))
with(subset(resHer2, padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/DE%20Her2%20breast%20-%20Nigerian%20and%20TCGA%20white-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of DE Her2 breast - Nigerian and TCGA white-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/DE Her2 breast - Nigerian and TCGA white-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>diffNigerianwhiteHER2 &lt;- as.data.frame(resHer2)[, ]
write.csv(diffNigerianwhiteHER2, file = &quot;Nigerian-white-HER2.csv&quot;)</code></pre>
<p>This is a comparison of 16 Nigerian luminal A cases and 8 TCGA white luminal A cases.</p>
<pre class="r"><code>resLumA&lt;- results(ddsMFbe, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.LumA&quot;, &quot;TCGA_white.LumA&quot;), alpha=0.05)
resLumA &lt;- subset(resLumA, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resLumA$temp &lt;- row.names(resLumA)
resLumA$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resLumA$temp)

resLumA$symbol &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumA$biotype &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column=&quot;GENEBIOTYPE&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumA$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumA$locstart &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column=&quot;GENESEQSTART&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

resLumA$locend &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column=&quot;GENESEQEND&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

with(resLumA, plot(log2FoldChange, -log10(padj), pch=20, main=&quot;Volcano plot of differential gene expression between Luminal A \nbreast cancers in Nigerian and \nTCGA white breast cancer patients&quot;, xlim=c(-50,50), ylim=c(0,70)))
with(subset(resLumA, resLumA$biotype==&quot;protein_coding&quot; &amp; padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;blue&quot;))
with(subset(resLumA, padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/DE%20LuminalA%20breast%20-%20Nigerian%20and%20TCGA%20white-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of DE LuminalA breast - Nigerian and TCGA white-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/DE LuminalA breast - Nigerian and TCGA white-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>diffNigerianwhiteLumA &lt;- as.data.frame(resLumA)[, ]
write.csv(diffNigerianwhiteLumA, file = &quot;Nigerian-white-LumA.csv&quot;)</code></pre>
<p>This is a comparison of 16 Nigerian luminal A cases and 4 TCGA black luminal A cases.</p>
<pre class="r"><code>resLumAblack&lt;- results(ddsMFbe, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.LumA&quot;, &quot;TCGA_black.LumA&quot;), alpha=0.05)
resLumAblack &lt;- subset(resLumAblack, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resLumAblack$temp &lt;- row.names(resLumAblack)
resLumAblack$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resLumAblack$temp)

resLumAblack$symbol &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumAblack$biotype &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column=&quot;GENEBIOTYPE&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumAblack$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumAblack$locstart &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column=&quot;GENESEQSTART&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

resLumAblack$locend &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column=&quot;GENESEQEND&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

with(resLumAblack, plot(log2FoldChange, -log10(padj), pch=20, main=&quot;Volcano plot of differential gene expression between Luminal A \nbreast cancers in Nigerian and \nTCGA black breast cancer patients&quot;, xlim=c(-50,50), ylim=c(0,70)))
with(subset(resLumAblack, resLumAblack$biotype==&quot;protein_coding&quot; &amp; padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;blue&quot;))
with(subset(resLumAblack, padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/DE%20LuminalA%20breast%20-%20Nigerian%20and%20TCGA%20black-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of DE LuminalA breast - Nigerian and TCGA black-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/DE LuminalA breast - Nigerian and TCGA black-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>diffNigerianblackLumA &lt;- as.data.frame(resLumAblack)[, ]
write.csv(diffNigerianblackLumA, file = &quot;Nigerian-black-LumA.csv&quot;)</code></pre>
<p>This is a comparison of 18 nigerian luminal B cases and 9 TCGA white luminal B cases.</p>
<pre class="r"><code>resLumB&lt;- results(ddsMFbe, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.LumB&quot;, &quot;TCGA_white.LumB&quot;), alpha=0.05)
resLumB &lt;- subset(resLumB, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resLumB$temp &lt;- row.names(resLumB)
resLumB$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resLumB$temp)

resLumB$symbol &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumB$biotype &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column=&quot;GENEBIOTYPE&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumB$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumB$locstart &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column=&quot;GENESEQSTART&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

resLumB$locend &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column=&quot;GENESEQEND&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

with(resLumB, plot(log2FoldChange, -log10(padj), pch=20, main=&quot;Volcano plot of differential gene expression between Luminal B \nbreast cancers in Nigerian and \nTCGA white breast cancer patients&quot;, xlim=c(-50,50), ylim=c(0,70)))
with(subset(resLumB, resLumB$biotype==&quot;protein_coding&quot; &amp; padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;blue&quot;))
with(subset(resLumB, padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/DE%20LuminalB%20breast%20-%20Nigerian%20and%20TCGA%20white-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of DE LuminalB breast - Nigerian and TCGA white-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/DE LuminalB breast - Nigerian and TCGA white-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>diffNigerianwhiteLumB &lt;- as.data.frame(resLumB)[, ]
write.csv(diffNigerianwhiteLumB, file = &quot;Nigerian-white-LumB.csv&quot;)</code></pre>
<p>This is a comparison of 18 Nigerian luminal B cases and 4 TCGA black luminal B cases.</p>
<pre class="r"><code>resLumBblack&lt;- results(ddsMFbe, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.LumB&quot;, &quot;TCGA_black.LumB&quot;), alpha=0.05)
resLumBblack &lt;- subset(resLumBblack, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resLumBblack$temp &lt;- row.names(resLumBblack)
resLumBblack$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resLumBblack$temp)

resLumBblack$symbol &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumBblack$biotype &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column=&quot;GENEBIOTYPE&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumBblack$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)

resLumBblack$locstart &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column=&quot;GENESEQSTART&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

resLumBblack$locend &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column=&quot;GENESEQEND&quot;,
                     keytype=&quot;GENEID&quot;,
                     multiVals=&quot;first&quot;)

with(resLumBblack, plot(log2FoldChange, -log10(padj), pch=20, main=&quot;Volcano plot of differential gene expression between Luminal B \nbreast cancers in Nigerian and \nTCGA black breast cancer patients&quot;, xlim=c(-50,50), ylim=c(0,70)))
with(subset(resLumBblack, resLumBblack$biotype==&quot;protein_coding&quot; &amp; padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;blue&quot;))
with(subset(resLumBblack, padj&lt;0.01 &amp; abs(log2FoldChange)&gt;25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/DE%20LuminalB%20breast%20-%20Nigerian%20and%20TCGA%20black-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of DE LuminalB breast - Nigerian and TCGA black-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/b237165f9643faff44320705011faeb4d7e21cce/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/DE LuminalB breast - Nigerian and TCGA black-1.png" target="_blank">b237165</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>diffNigerianblackLumB &lt;- as.data.frame(resLumBblack)[, ]
write.csv(diffNigerianblackLumB, file = &quot;Nigerian-black-LumB.csv&quot;)</code></pre>
</div>
<div id="validation-of-results" class="section level1">
<h1>Validation of results</h1>
<pre class="r"><code>chromecheck &lt;- factor(diffNigerianblackbasal$chr)
plot(table(chromecheck))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Checking%20distribution%20of%20all%20results%20by%20chromosome-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromecheck2 &lt;- factor(diffNigerianwhitebasal$chr)
plot(table(chromecheck2))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Checking%20distribution%20of%20all%20results%20by%20chromosome-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromecheck3 &lt;- factor(diffNigerianwhiteHER2$chr)
plot(table(chromecheck3))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Checking%20distribution%20of%20all%20results%20by%20chromosome-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromecheck4 &lt;- factor(diffNigerianwhiteLumA$chr)
plot(table(chromecheck4))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Checking%20distribution%20of%20all%20results%20by%20chromosome-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromecheck5 &lt;- factor(diffNigerianblackLumA$chr)
plot(table(chromecheck5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Checking%20distribution%20of%20all%20results%20by%20chromosome-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromecheck6 &lt;- factor(diffNigerianwhiteLumB$chr)
plot(table(chromecheck6))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Checking%20distribution%20of%20all%20results%20by%20chromosome-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromecheck7 &lt;- factor(diffNigerianblackLumB$chr)
plot(table(chromecheck7))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Checking%20distribution%20of%20all%20results%20by%20chromosome-7.png" width="672" style="display: block; margin: auto;" /> *****Given that each of the differential expression analyses produce a similar distribution along the chromosomes, this is concerning for a problem with how the differential expression is run.*****</p>
</div>
<div id="repeat-validation-of-results-without-batch-effect" class="section level1">
<h1>Repeat validation of results without batch effect</h1>
<pre class="r"><code>resregBasal&lt;- results(ddsMF, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.Basal&quot;, &quot;TCGA_white.Basal&quot;), alpha=0.05)
resregBasal &lt;- subset(resregBasal, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resregBasal$temp &lt;- row.names(resregBasal)
resregBasal$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resregBasal$temp)

resregBasal$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregBasal$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)
chrome2check &lt;- factor(resregBasal$chr)
plot(table(chrome2check))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Repeat%20analysis%20without%20batch%20effect-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>resregBasalblack&lt;- results(ddsMF, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.Basal&quot;, &quot;TCGA_black.Basal&quot;), alpha=0.05)
resregBasalblack &lt;- subset(resregBasalblack, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resregBasalblack$temp &lt;- row.names(resregBasalblack)
resregBasalblack$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resregBasalblack$temp)

resregBasalblack$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregBasalblack$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)
chrome2check2 &lt;- factor(resregBasalblack$chr)
plot(table(chrome2check2))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Repeat%20analysis%20without%20batch%20effect-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>resregHer2&lt;- results(ddsMF, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.Her2&quot;, &quot;TCGA_white.Her2&quot;), alpha=0.05)
resregHer2 &lt;- subset(resregHer2, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resregHer2$temp &lt;- row.names(resregHer2)
resregHer2$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resregHer2$temp)

resregHer2$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregHer2$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)
chrome2check3 &lt;- factor(resregHer2$chr)
plot(table(chrome2check3))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Repeat%20analysis%20without%20batch%20effect-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>resregLumA&lt;- results(ddsMF, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.LumA&quot;, &quot;TCGA_white.LumA&quot;), alpha=0.05)
resregLumA &lt;- subset(resregLumA, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resregLumA$temp &lt;- row.names(resregLumA)
resregLumA$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resregLumA$temp)

resregLumA$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregLumA$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)
chrome2check4 &lt;- factor(resregLumA$chr)
plot(table(chrome2check4))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Repeat%20analysis%20without%20batch%20effect-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>resregLumAblack&lt;- results(ddsMF, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.LumA&quot;, &quot;TCGA_black.LumA&quot;), alpha=0.05)
resregLumAblack &lt;- subset(resregLumAblack, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resregLumAblack$temp &lt;- row.names(resregLumAblack)
resregLumAblack$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resregLumAblack$temp)

resregLumAblack$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregLumAblack$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)
chrome2check5 &lt;- factor(resregLumAblack$chr)
plot(table(chrome2check5))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Repeat%20analysis%20without%20batch%20effect-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>resregLumB&lt;- results(ddsMF, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.LumB&quot;, &quot;TCGA_white.LumB&quot;), alpha=0.05)
resregLumB &lt;- subset(resregLumB, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resregLumB$temp &lt;- row.names(resregLumB)
resregLumB$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resregLumB$temp)

resregLumB$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregLumB$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)
chrome2check6 &lt;- factor(resregLumB$chr)
plot(table(chrome2check6))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Repeat%20analysis%20without%20batch%20effect-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>resregLumBblack&lt;- results(ddsMF, contrast=c(&quot;sampleCondition&quot;, &quot;Nigerian.LumB&quot;, &quot;TCGA_black.LumB&quot;), alpha=0.05)
resregLumBblack &lt;- subset(resregLumBblack, padj &lt; 0.05) #Selecting only the lowest threshold of significant findings
resregLumBblack$temp &lt;- row.names(resregLumBblack)
resregLumBblack$temp &lt;- gsub(&quot;[.].+&quot;, &quot;&quot;, resregLumBblack$temp)

resregLumBblack$chr &lt;- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregLumBblack$temp,
                     column=&quot;SEQNAME&quot;,
                     keytype=&quot;GENEID&quot;,           
                     multiVals=&quot;first&quot;)
chrome2check7 &lt;- factor(resregLumBblack$chr)
plot(table(chrome2check7))</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Repeat%20analysis%20without%20batch%20effect-7.png" width="672" style="display: block; margin: auto;" /> *****The concerning, persistent distribution in RNA results across chromosomes is persistent across differential expression results regardless of batch effect correction.*****</p>
</div>
<div id="protein-coding-assessment---pending-resolution-of-validation-issues" class="section level1">
<h1>Protein-coding assessment -&gt; pending resolution of validation issues</h1>
<pre class="r"><code>proteincoding &lt;- (subset(resBasal, biotype==&quot;protein_coding&quot;))
proteincoding$analysis &lt;- &quot;Nigerian-TCGAwhite.Basal&quot;

proteincoding2 &lt;- (subset(resBasalblack, biotype==&quot;protein_coding&quot;))
proteincoding2$analysis &lt;- &quot;Nigerian-TCGAblack.Basal&quot;

proteincoding3 &lt;- (subset(resHer2, biotype==&quot;protein_coding&quot;))
proteincoding3$analysis &lt;- &quot;Nigerian-TCGAwhite.Her2&quot;

proteincoding4 &lt;- (subset(resLumA, biotype==&quot;protein_coding&quot;))
proteincoding4$analysis &lt;- &quot;Nigerian-TCGAwhite.LuminalA&quot;

proteincoding5 &lt;- (subset(resLumAblack, biotype==&quot;protein_coding&quot;))
proteincoding5$analysis &lt;- &quot;Nigerian-TCGAblack.LuminalA&quot;

proteincoding6 &lt;- (subset(resLumB, biotype==&quot;protein_coding&quot;))
proteincoding6$analysis &lt;- &quot;Nigerian-TCGAwhite.LuminalB&quot;

proteincoding7 &lt;- (subset(resLumBblack, biotype==&quot;protein_coding&quot;))
proteincoding7$analysis &lt;- &quot;Nigerian-TCGAblack.LuminalB&quot;

proteincoding=rbind(proteincoding, proteincoding2, proteincoding3, proteincoding4, proteincoding5, proteincoding6, proteincoding7)

table(proteincoding$analysis)</code></pre>
<pre><code>
   Nigerian-TCGAblack.Basal Nigerian-TCGAblack.LuminalA 
                       8340                        2490 
Nigerian-TCGAblack.LuminalB    Nigerian-TCGAwhite.Basal 
                       2640                        8694 
    Nigerian-TCGAwhite.Her2 Nigerian-TCGAwhite.LuminalA 
                       4266                        4546 
Nigerian-TCGAwhite.LuminalB 
                       4609 </code></pre>
<pre class="r"><code>summary(proteincoding)</code></pre>
<pre><code>
out of 35585 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 22664, 64%
LFC &lt; 0 (down)     : 12921, 36%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code>proteincodingsig &lt;- head(proteincoding[order(proteincoding$log2FoldChange, proteincoding$padj), ],100)
proteincodingsig2 &lt;- tail(proteincoding[ order(proteincoding$log2FoldChange, proteincoding$padj), ],100)
head(proteincodingsig)</code></pre>
<pre><code>log2 fold change (MLE): sampleCondition Nigerian.Basal vs TCGA_white.Basal 
Wald test p-value: sampleCondition Nigerian.Basal vs TCGA_white.Basal 
DataFrame with 6 rows and 13 columns
                          baseMean    log2FoldChange            lfcSE
                         &lt;numeric&gt;         &lt;numeric&gt;        &lt;numeric&gt;
ENSG00000196758.3 1.07266575682475 -33.0504026526818 5.44278386573842
ENSG00000244057.2  3.8664791422837 -29.9124978232476 8.06142794499249
ENSG00000172154.4 1.09978591470531 -27.9594191562178 8.05577288959801
ENSG00000132446.5 6.94679531324786 -27.3715053489322 7.24205499200452
ENSG00000187714.5 2.81206422058059 -26.6697077688643 2.43565294515868
ENSG00000081800.4 3.61494230545257 -26.3142743221192 3.89049286603071
                               stat               pvalue
                          &lt;numeric&gt;            &lt;numeric&gt;
ENSG00000196758.3   -6.072334207634 1.26064224389568e-09
ENSG00000244057.2 -3.71057063678506 0.000206792583852525
ENSG00000172154.4 -3.47073080874962 0.000519044053595334
ENSG00000132446.5 -3.77952188697149 0.000157129781836489
ENSG00000187714.5 -10.9497158952286 6.66563523371499e-28
ENSG00000081800.4 -6.76373797054829 1.34476203066463e-11
                                  padj            temp      symbol
                             &lt;numeric&gt;     &lt;character&gt; &lt;character&gt;
ENSG00000196758.3 4.08306559618518e-08 ENSG00000196758  AC079612.1
ENSG00000244057.2  0.00180810692505404 ENSG00000244057       LCE3C
ENSG00000172154.4  0.00389883669766717 ENSG00000172154       OR8I2
ENSG00000132446.5  0.00258696307879189 ENSG00000132446      FTHL17
ENSG00000187714.5 1.67490211783945e-25 ENSG00000187714     SLC18A3
ENSG00000081800.4 6.71232538442775e-10 ENSG00000081800     SLC13A1
                         biotype         chr  locstart    locend
                     &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt;
ENSG00000196758.3 protein_coding           2 240499995 240507788
ENSG00000244057.2 protein_coding           1 152573138 152573562
ENSG00000172154.4 protein_coding          11  55860753  55861762
ENSG00000132446.5 protein_coding           X  31089360  31090170
ENSG00000187714.5 protein_coding          10  50818347  50820765
ENSG00000081800.4 protein_coding           7 122753585 122840040
                                     analysis
                                  &lt;character&gt;
ENSG00000196758.3 Nigerian-TCGAwhite.LuminalB
ENSG00000244057.2 Nigerian-TCGAwhite.LuminalB
ENSG00000172154.4 Nigerian-TCGAwhite.LuminalB
ENSG00000132446.5 Nigerian-TCGAblack.LuminalB
ENSG00000187714.5 Nigerian-TCGAwhite.LuminalB
ENSG00000081800.4 Nigerian-TCGAwhite.LuminalA</code></pre>
<pre class="r"><code>tail(proteincodingsig2)</code></pre>
<pre><code>log2 fold change (MLE): sampleCondition Nigerian.Basal vs TCGA_white.Basal 
Wald test p-value: sampleCondition Nigerian.Basal vs TCGA_white.Basal 
DataFrame with 6 rows and 13 columns
                          baseMean   log2FoldChange            lfcSE
                         &lt;numeric&gt;        &lt;numeric&gt;        &lt;numeric&gt;
ENSG00000203933.2 1.46781544351778 32.6095667377862 9.67044693560393
ENSG00000111732.6 2.54501192302528  32.765170136005 1.59166201615886
ENSG00000091482.5  9.4529745707773 32.7849502606895 1.64482664502282
ENSG00000203933.2 1.46781544351778 33.2047492441598 8.50486089531779
ENSG00000081051.3 45.1461520068687 35.3425708262877 2.15086517296123
ENSG00000203933.2 1.46781544351778 40.3339581130024 9.55776587792804
                              stat               pvalue
                         &lt;numeric&gt;            &lt;numeric&gt;
ENSG00000203933.2 3.37208476039786 0.000746014856674372
ENSG00000111732.6 20.5855073522938 3.70142383123073e-94
ENSG00000091482.5 19.9321614590179 2.14112337962345e-88
ENSG00000203933.2 3.90420838775154 9.45343009873791e-05
ENSG00000081051.3  16.431792783008 1.13260374732445e-60
ENSG00000203933.2 4.22001947192978 2.44281218318888e-05
                                  padj            temp      symbol
                             &lt;numeric&gt;     &lt;character&gt; &lt;character&gt;
ENSG00000203933.2  0.00883478939728623 ENSG00000203933     CXorf66
ENSG00000111732.6  1.6475566247641e-90 ENSG00000111732       AICDA
ENSG00000091482.5 8.33914028278842e-85 ENSG00000091482        SMPX
ENSG00000203933.2  0.00104174175653441 ENSG00000203933     CXorf66
ENSG00000081051.3 1.76448337795676e-56 ENSG00000081051         AFP
ENSG00000203933.2 0.000555976201634763 ENSG00000203933     CXorf66
                         biotype         chr  locstart    locend
                     &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt;
ENSG00000203933.2 protein_coding           X 139037884 139047679
ENSG00000111732.6 protein_coding          12   8754762   8765467
ENSG00000091482.5 protein_coding           X  21724090  21776281
ENSG00000203933.2 protein_coding           X 139037884 139047679
ENSG00000081051.3 protein_coding           4  74296855  74321891
ENSG00000203933.2 protein_coding           X 139037884 139047679
                                     analysis
                                  &lt;character&gt;
ENSG00000203933.2 Nigerian-TCGAblack.LuminalA
ENSG00000111732.6 Nigerian-TCGAwhite.LuminalB
ENSG00000091482.5 Nigerian-TCGAwhite.LuminalB
ENSG00000203933.2     Nigerian-TCGAwhite.Her2
ENSG00000081051.3 Nigerian-TCGAblack.LuminalB
ENSG00000203933.2 Nigerian-TCGAblack.LuminalB</code></pre>
<pre class="r"><code>write.table(proteincodingsig,&quot;all-analysis-protein-coding-genes-head&quot;,sep=&quot;\t&quot;,row.names=TRUE)
write.table(proteincodingsig2,&quot;all-analysis-protein-coding-genes-tail&quot;,sep=&quot;\t&quot;,row.names=TRUE)</code></pre>
<p>These are the strongest differentially-expressed protein-coding genes across all analyses.</p>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.5

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] gplots_3.0.1.1              calibrate_1.7.2            
 [3] MASS_7.3-51.1               limma_3.38.3               
 [5] sva_3.30.1                  mgcv_1.8-26                
 [7] nlme_3.1-137                pheatmap_1.0.12            
 [9] genefilter_1.64.0           apeglm_1.4.2               
[11] EnsDb.Hsapiens.v75_2.99.0   ensembldb_2.6.3            
[13] AnnotationFilter_1.6.0      GenomicFeatures_1.34.1     
[15] org.Hs.eg.db_3.7.0          AnnotationDbi_1.44.0       
[17] hexbin_1.27.2               ggplot2_3.1.0              
[19] dplyr_0.7.8                 stringi_1.2.4              
[21] DESeq2_1.22.2               SummarizedExperiment_1.12.0
[23] DelayedArray_0.8.0          BiocParallel_1.16.5        
[25] matrixStats_0.54.0          Biobase_2.42.0             
[27] GenomicRanges_1.34.0        GenomeInfoDb_1.18.1        
[29] IRanges_2.16.0              S4Vectors_0.20.1           
[31] BiocGenerics_0.28.0        

loaded via a namespace (and not attached):
 [1] colorspace_1.4-0         rprojroot_1.3-2         
 [3] htmlTable_1.13.1         XVector_0.22.0          
 [5] base64enc_0.1-3          rstudioapi_0.9.0        
 [7] bit64_0.9-7              codetools_0.2-16        
 [9] splines_3.5.0            R.methodsS3_1.7.1       
[11] geneplotter_1.60.0       knitr_1.21              
[13] Formula_1.2-3            workflowr_1.1.1         
[15] Rsamtools_1.34.0         annotate_1.60.0         
[17] cluster_2.0.7-1          R.oo_1.22.0             
[19] compiler_3.5.0           httr_1.4.0              
[21] backports_1.1.3          assertthat_0.2.0        
[23] Matrix_1.2-15            lazyeval_0.2.1          
[25] acepack_1.4.1            htmltools_0.3.6         
[27] prettyunits_1.0.2        tools_3.5.0             
[29] bindrcpp_0.2.2           coda_0.19-2             
[31] gtable_0.2.0             glue_1.3.0              
[33] GenomeInfoDbData_1.2.0   reshape2_1.4.3          
[35] Rcpp_1.0.0               bbmle_1.0.20            
[37] Biostrings_2.50.2        gdata_2.18.0            
[39] rtracklayer_1.42.1       xfun_0.4                
[41] stringr_1.3.1            gtools_3.8.1            
[43] XML_3.98-1.16            zlibbioc_1.28.0         
[45] scales_1.0.0             hms_0.4.2               
[47] ProtGenerics_1.14.0      RColorBrewer_1.1-2      
[49] yaml_2.2.0               curl_3.3                
[51] memoise_1.1.0            gridExtra_2.3           
[53] emdbook_1.3.10           biomaRt_2.38.0          
[55] rpart_4.1-13             latticeExtra_0.6-28     
[57] RSQLite_2.1.1            checkmate_1.9.1         
[59] caTools_1.17.1.1         rlang_0.3.1             
[61] pkgconfig_2.0.2          bitops_1.0-6            
[63] evaluate_0.12            lattice_0.20-38         
[65] purrr_0.2.5              bindr_0.1.1             
[67] labeling_0.3             GenomicAlignments_1.18.1
[69] htmlwidgets_1.3          bit_1.1-14              
[71] tidyselect_0.2.5         plyr_1.8.4              
[73] magrittr_1.5             R6_2.3.0                
[75] Hmisc_4.1-1              DBI_1.0.0               
[77] pillar_1.3.1             whisker_0.3-2           
[79] foreign_0.8-71           withr_2.1.2             
[81] survival_2.43-3          RCurl_1.95-4.11         
[83] nnet_7.3-12              tibble_2.0.1            
[85] crayon_1.3.4             KernSmooth_2.23-15      
[87] rmarkdown_1.11           progress_1.2.0          
[89] locfit_1.5-9.1           grid_3.5.0              
[91] data.table_1.12.0        blob_1.1.1              
[93] git2r_0.24.0             digest_0.6.18           
[95] xtable_1.8-3             numDeriv_2016.8-1       
[97] R.utils_2.7.0            munsell_0.5.0           </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
