<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Padma Sheila Rajagopal, MD MPH" />


<title>Nigerian-TCGA Differential Expression Analysis Using DeSeq2</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Olopade Lab: TCGA-Nigerian-RNAseq Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="NigerianTCGArawcountsDeSeq2.html">Differential Expression Analysis</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/parajago/TCGA-Nigerian-RNAseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Nigerian-TCGA Differential Expression Analysis Using DeSeq2</h1>
<h4 class="author"><em>Padma Sheila Rajagopal, MD MPH</em></h4>
<h4 class="date"><em>1/23/2019</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-01-23</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20190113)</code> </summary></p>
<p>The command <code>set.seed(20190113)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/tree/3ce51dded4a551b7bcf18fbaac17daf9745fceee" target="_blank">3ce51dd</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/NigerianTCGArawcountsDeSeq2-part.Rmd

Unstaged changes:
    Modified:   analysis/NigerianTCGArawcountsDeSeq2.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
<td style="text-align:left;">
batch effect correction implemented
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
<td style="text-align:left;">
batch effect correction implemented
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/616d47012ac13731ce150d005296f7bcdcac52b9/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">616d470</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
<td style="text-align:left;">
Batch effect corrected noted in Rmd file
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/d47a998f13b96ba2dab3394fb2b26251eb015be4/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">d47a998</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-22
</td>
<td style="text-align:left;">
Cleanup of website, no formal changes made
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/d47a998f13b96ba2dab3394fb2b26251eb015be4/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">d47a998</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-22
</td>
<td style="text-align:left;">
Cleanup of website, no formal changes made
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/cd53cd0d2448d0eb970cffed8785f32d085fb4ff/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">cd53cd0</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-16
</td>
<td style="text-align:left;">
initial diff exp pass
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/parajago/TCGA-Nigerian-RNAseq/0ad8ebc0b6566ff025a638847f5624863c92be77/docs/NigerianTCGArawcountsDeSeq2.html" target="_blank">0ad8ebc</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-16
</td>
<td style="text-align:left;">
adding DeSeq data to date
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/c3c1e743aec418ed3c294e6f332dd4e3e580fdf1/analysis/NigerianTCGArawcountsDeSeq2.Rmd" target="_blank">c3c1e74</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-16
</td>
<td style="text-align:left;">
“Updated workflowR with basic information”
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="translation-from-htseq---deseq2-table" class="section level1">
<h1>Translation from HTSeq -&gt; DESeq2 table</h1>
<p>I have 86 TCGA patients with whole-genome sequencing data and RNAseq data as well as 99 Nigerian patients with RNA-seq data. Raw counts were initially processed using HTSeq, so HTSeq data is being formatted for use with DESeq2.</p>
<pre class="r"><code>FOLDER &lt;- &quot;/Users/parajago/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants&quot;
sampleFiles &lt;- grep(&quot;htseq.counts&quot;,list.files(FOLDER),value=TRUE)

#Differential gene expression setup based on race (b/w/other)
sampleConditionrace &lt;- sampleFiles
countVar2=1
for (sample in sampleConditionrace){
  if (stri_detect_fixed(sample,&quot;LIB&quot;)==TRUE){
    sampleConditionrace[countVar2] &lt;- &quot;Nigerian&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;black&quot;)==TRUE){
    sampleConditionrace[countVar2] &lt;- &quot;TCGA_black&quot;
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,&quot;white&quot;)==TRUE){
    sampleConditionrace[countVar2] &lt;- &quot;TCGA_white&quot;
    countVar2=countVar2+1
  } else{
    sampleConditionrace[countVar2] &lt;- &quot;TCGA_other&quot;
    countVar2=countVar2+1
  }
}

sampleTable2 &lt;- data.frame(sampleName=gsub(&quot;.htseq.counts&quot;,&quot;&quot;,sampleFiles),
                          fileName=sampleFiles,
                          condition=sampleConditionrace)

ddsHTSeqMF &lt;- DESeqDataSetFromHTSeqCount(sampleTable=sampleTable2,
                                       directory=FOLDER,
                                       design=~condition)</code></pre>
</div>
<div id="variance-transformation-batch-effect-assessment-and-initial-exploratory-analysis" class="section level1">
<h1>Variance transformation, batch effect assessment and initial exploratory analysis</h1>
<p>I am now performing different visualization approaches to assess the variance of the data, batch effect and expected differences between the two populations.</p>
<pre class="r"><code>ddsHTSeqMF &lt;- ddsHTSeqMF[rowSums(counts(ddsHTSeqMF)) &gt; 1, ] #Pre-filtering the dataset by removing the rows without information about gene expression

dds &lt;- estimateSizeFactors(ddsHTSeqMF) #The size factor is the median ratio of the sample over a &quot;pseudosample&quot;: for each gene, the geometric mean of all samples.
dds$condition &lt;- relevel(dds$condition, ref = &quot;Nigerian&quot;) #explicitng making the Nigerian patients the reference population 

vsd &lt;- vst(ddsHTSeqMF, blind = TRUE) #Variance-stabilizing transformation

df &lt;- bind_rows( #Graphical representation of the variance-stabilizing transformation
          as_data_frame(log2(counts(dds, normalized=TRUE)[, 1:2]+1)) %&gt;%
          mutate(transformation = &quot;log2(x + 1)&quot;),
          as_data_frame(assay(vsd)[, 1:2]) %&gt;% mutate(transformation = &quot;vst&quot;))
          colnames(df)[1:2] &lt;- c(&quot;x&quot;, &quot;y&quot;)

ggplot(df, aes(x = x, y = y)) + labs(subtitle=&quot;Figure 1: Graphical representation of the \nvariance-stabilizing transformation \nrelative to the standard transformation of log2(x+1).\nThis reduces variance purely due to low read counts&quot;) + geom_hex(bins = 80) +
  coord_fixed() + facet_grid( . ~ transformation) </code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance%20transformation%20and%20batch%20effect%20assessment-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Variance transformation and batch effect assessment-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance transformation and batch effect assessment-1.png" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>plotPCA(vsd) + labs(subtitle = &quot;PCA 1: first PCA demonstrates batch effect between\nthe Nigerian patients and TCGA patients,\nwhich would be presumed given that these experiments\nwere done in different settings at different times.\nThe TCGA patients also appear to have internal\nbatch effect that is not race-driven\ngiven distribution into 2 separate groups&quot;) #PCA1: Raw counts</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance%20transformation%20and%20batch%20effect%20assessment-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Variance transformation and batch effect assessment-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance transformation and batch effect assessment-2.png" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>topVarGenes &lt;- head(order(rowVars(assay(vsd)), decreasing = TRUE), 1000)
hm  &lt;- assay(vsd)[ topVarGenes, ]
hm  &lt;- hm - rowMeans(hm)
anno &lt;- as.data.frame(colData(vsd)[, c(&quot;condition&quot;)])
rownames(anno) &lt;- colnames(vsd)
pheatmap(hm, annotation_col = anno, cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=FALSE, main=&quot;Heat map 1: \nRelative VST-transformed counts across samples&quot;)</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance%20transformation%20and%20batch%20effect%20assessment-3.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Variance transformation and batch effect assessment-3.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance transformation and batch effect assessment-3.png" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>mat &lt;- assay(vsd) #PCA2: Removing the batch effect created by the separate batch processing of the TCGA and Nigerian samples (although this may overcorrect for the actual difference between these two populations).  
mat &lt;- limma::removeBatchEffect(mat, vsd$condition)
assay(vsd) &lt;- mat
plotPCA(vsd) + labs(subtitle=&quot;PCA2: The second PCA represents removal of batch effect using the limma removeBatchEffect method.\nThis largely accounts for the batch effect noticed in PCA1. \nThere remain some outliers, primarily from the Nigerian patients.&quot;)</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance%20transformation%20and%20batch%20effect%20assessment-4.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Variance transformation and batch effect assessment-4.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance transformation and batch effect assessment-4.png" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>topVarGenes &lt;- head(order(rowVars(assay(vsd)), decreasing = TRUE), 1000)
hm  &lt;- assay(vsd)[ topVarGenes, ]
hm  &lt;- hm - rowMeans(hm)
anno &lt;- as.data.frame(colData(vsd)[, c(&quot;condition&quot;)])
rownames(anno) &lt;- colnames(vsd)
pheatmap(hm, annotation_col = anno, cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=FALSE, main=&quot;Heat map 2: Batch-effect-corrected relative\n VST-transformed counts across samples&quot;)</code></pre>
<p><img src="figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance%20transformation%20and%20batch%20effect%20assessment-5.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of Variance transformation and batch effect assessment-5.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/parajago/TCGA-Nigerian-RNAseq/blob/3ce51dded4a551b7bcf18fbaac17daf9745fceee/docs/figure/NigerianTCGArawcountsDeSeq2.Rmd/Variance transformation and batch effect assessment-5.png" target="_blank">3ce51dd</a>
</td>
<td style="text-align:left;">
Sheila Rajagopal
</td>
<td style="text-align:left;">
2019-01-23
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="using-sva-to-additionally-account-for-batch-effect-related-to-the-nigeriantcga-samples" class="section level1">
<h1>Using sva to additionally account for batch effect related to the Nigerian/TCGA samples</h1>
<pre class="r"><code>dat  &lt;- counts(dds, normalized = TRUE)
idx  &lt;- rowMeans(dat) &gt; 1
dat  &lt;- dat[idx, ]
mod  &lt;- model.matrix(~condition, colData(dds))
mod0 &lt;- model.matrix(~   1, colData(dds))
svseq &lt;- svaseq(dat, mod, mod0, n.sv = 1)</code></pre>
<pre><code>Number of significant surrogate variables is:  1 
Iteration (out of 5 ):1  2  3  4  5  </code></pre>
<pre class="r"><code>sampleTablebe &lt;- data.frame(sampleName=gsub(&quot;.htseq.counts&quot;,&quot;&quot;,sampleFiles),
                          fileName=sampleFiles,
                          condition=sampleConditionrace, 
                          be=svseq$sv[,1])

ddsHTSeqMFbe &lt;- DESeqDataSetFromHTSeqCount(sampleTable=sampleTablebe,
                                       directory=FOLDER,
                                       design=~condition + be)

ddsHTSeqMFbe &lt;- ddsHTSeqMFbe[rowSums(counts(ddsHTSeqMFbe)) &gt; 1, ]</code></pre>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.5

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] limma_3.38.3                sva_3.30.1                 
 [3] mgcv_1.8-26                 nlme_3.1-137               
 [5] pheatmap_1.0.12             genefilter_1.64.0          
 [7] apeglm_1.4.2                EnsDb.Hsapiens.v75_2.99.0  
 [9] ensembldb_2.6.3             AnnotationFilter_1.6.0     
[11] biomaRt_2.38.0              GenomicFeatures_1.34.1     
[13] org.Hs.eg.db_3.7.0          AnnotationDbi_1.44.0       
[15] hexbin_1.27.2               ggplot2_3.1.0              
[17] dplyr_0.7.8                 stringi_1.2.4              
[19] DESeq2_1.22.2               SummarizedExperiment_1.12.0
[21] DelayedArray_0.8.0          BiocParallel_1.16.5        
[23] matrixStats_0.54.0          Biobase_2.42.0             
[25] GenomicRanges_1.34.0        GenomeInfoDb_1.18.1        
[27] IRanges_2.16.0              S4Vectors_0.20.1           
[29] BiocGenerics_0.28.0        

loaded via a namespace (and not attached):
 [1] colorspace_1.4-0         rprojroot_1.3-2         
 [3] htmlTable_1.13.1         XVector_0.22.0          
 [5] base64enc_0.1-3          rstudioapi_0.9.0        
 [7] bit64_0.9-7              splines_3.5.0           
 [9] R.methodsS3_1.7.1        geneplotter_1.60.0      
[11] knitr_1.21               Formula_1.2-3           
[13] workflowr_1.1.1          Rsamtools_1.34.0        
[15] annotate_1.60.0          cluster_2.0.7-1         
[17] R.oo_1.22.0              compiler_3.5.0          
[19] httr_1.4.0               backports_1.1.3         
[21] assertthat_0.2.0         Matrix_1.2-15           
[23] lazyeval_0.2.1           acepack_1.4.1           
[25] htmltools_0.3.6          prettyunits_1.0.2       
[27] tools_3.5.0              bindrcpp_0.2.2          
[29] coda_0.19-2              gtable_0.2.0            
[31] glue_1.3.0               GenomeInfoDbData_1.2.0  
[33] reshape2_1.4.3           Rcpp_1.0.0              
[35] bbmle_1.0.20             Biostrings_2.50.2       
[37] rtracklayer_1.42.1       xfun_0.4                
[39] stringr_1.3.1            XML_3.98-1.16           
[41] zlibbioc_1.28.0          MASS_7.3-51.1           
[43] scales_1.0.0             hms_0.4.2               
[45] ProtGenerics_1.14.0      RColorBrewer_1.1-2      
[47] yaml_2.2.0               curl_3.3                
[49] memoise_1.1.0            gridExtra_2.3           
[51] emdbook_1.3.10           rpart_4.1-13            
[53] latticeExtra_0.6-28      RSQLite_2.1.1           
[55] checkmate_1.9.1          rlang_0.3.1             
[57] pkgconfig_2.0.2          bitops_1.0-6            
[59] evaluate_0.12            lattice_0.20-38         
[61] purrr_0.2.5              bindr_0.1.1             
[63] labeling_0.3             GenomicAlignments_1.18.1
[65] htmlwidgets_1.3          bit_1.1-14              
[67] tidyselect_0.2.5         plyr_1.8.4              
[69] magrittr_1.5             R6_2.3.0                
[71] Hmisc_4.1-1              DBI_1.0.0               
[73] pillar_1.3.1             whisker_0.3-2           
[75] foreign_0.8-71           withr_2.1.2             
[77] survival_2.43-3          RCurl_1.95-4.11         
[79] nnet_7.3-12              tibble_2.0.1            
[81] crayon_1.3.4             rmarkdown_1.11          
[83] progress_1.2.0           locfit_1.5-9.1          
[85] grid_3.5.0               data.table_1.12.0       
[87] blob_1.1.1               git2r_0.24.0            
[89] digest_0.6.18            xtable_1.8-3            
[91] numDeriv_2016.8-1        R.utils_2.7.0           
[93] munsell_0.5.0           </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
