---
title: "Nigerian-TCGA Differential Expression Analysis Using DeSeq2"
author: "Padma Sheila Rajagopal, MD MPH"
date: "2/26/2019"

site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    df_print: paged
    toc: true
    theme: spacelab
    highlight: textmate
---
```{r Setup, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE)
library("DESeq2")
library(stringi)
library("dplyr")
library("ggplot2")
library("hexbin")
library("AnnotationDbi")
library("org.Hs.eg.db")
library('GenomicFeatures')
library(EnsDb.Hsapiens.v75)
library(apeglm)
library("genefilter")
library(pheatmap)
library("sva")
library(limma)
library(calibrate)
library(gplots)
setwd("~/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants")
```

#Translation from HTSeq -> DESeq2 table
I have 86 TCGA patients with whole-genome sequencing data and RNAseq data as well as 99 Nigerian patients with RNA-seq data. Raw counts were initially processed using HTSeq, so HTSeq data is being formatted for use with DESeq2.
```{r Preparing the sampleTable using HTSeq raw counts}
FOLDER <- "/Users/parajago/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants"
sampleFiles <- grep("htseq.counts",list.files(FOLDER),value=TRUE)

#Differential gene expression setup based on race (b/w/other)
sampleConditionrace <- sampleFiles
countVar2=1
for (sample in sampleConditionrace){
  if (stri_detect_fixed(sample,"LIB")==TRUE){
    sampleConditionrace[countVar2] <- "Nigerian"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"black")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_black"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"white")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_white"
    countVar2=countVar2+1
  } else{
    sampleConditionrace[countVar2] <- "TCGA_other"
    countVar2=countVar2+1
  }
}

#Condition based on PAM50 subtype 
sampleConditionPAM50 <- sampleFiles
countVar2=1
for (sample in sampleConditionPAM50){
  if (stri_detect_fixed(sample,"Her2")==TRUE){
    sampleConditionPAM50[countVar2] <- "Her2"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"Basal")==TRUE){
    sampleConditionPAM50[countVar2] <- "Basal"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"LumA")==TRUE){
    sampleConditionPAM50[countVar2] <- "LumA"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"LumB")==TRUE){
    sampleConditionPAM50[countVar2] <- "LumB"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"PAMNL")==TRUE){
    sampleConditionPAM50[countVar2] <- "Normal"
    countVar2=countVar2+1
  } else{
    sampleConditionPAM50[countVar2] <- "PAM_other"
    countVar2=countVar2+1
  }
}

#Condition based on batch (relevant to the Nigerian patients only; no difference in batch for the TCGA patients)
batchval <- sampleFiles
countVar2=1
for (sample in batchval){
  if (stri_detect_fixed(sample,"batch1")==TRUE){
    batchval[countVar2] <- "batch1"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch23")==TRUE){
    batchval[countVar2] <- "batch23"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch4")==TRUE){
    batchval[countVar2] <- "batch4"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch5")==TRUE){
    batchval[countVar2] <- "batch5"
    countVar2=countVar2+1
  } else{
    batchval[countVar2] <- "batchT"
    countVar2=countVar2+1
  }
}

table(sampleConditionrace, sampleConditionPAM50)

sampleTable2 <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          batch=batchval)

sampleTable2$sampleCondition <- paste(sampleTable2$condition1, sampleTable2$condition2, sep=".")

ddsHTSeqMF <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTable2,
                                       directory=FOLDER,
                                       design=~sampleCondition)
```


#Variance transformation, batch effect assessment and initial exploratory analysis
I am now performing different visualization approaches to assess the variance of the data, batch effect and expected differences between the two populations.
```{r Variance transformation and batch effect assessment}
ddsHTSeqMF <- ddsHTSeqMF[rowSums(counts(ddsHTSeqMF)) > 1, ] #Pre-filtering the dataset by removing the rows without information about gene expression

dds <- estimateSizeFactors(ddsHTSeqMF) #The size factor is the median ratio of the sample over a "pseudosample": for each gene, the geometric mean of all samples.
dds$sampleCondition <- relevel(dds$sampleCondition, ref = "Nigerian.Basal") #explicitly making the Nigerian Basal patients the reference population

vsd <- vst(ddsHTSeqMF, blind = TRUE) #Variance-stabilizing transformation

df <- bind_rows( #Graphical representation of the variance-stabilizing transformation
          as_data_frame(log2(counts(dds, normalized=TRUE)[, 1:2]+1)) %>%
          mutate(transformation = "log2(x + 1)"),
          as_data_frame(assay(vsd)[, 1:2]) %>% mutate(transformation = "vst"))
          colnames(df)[1:2] <- c("x", "y")

ggplot(df, aes(x = x, y = y)) + geom_hex(bins = 80) + coord_fixed() + facet_grid( . ~ transformation)
```

Figure 1: Graphical representation of the variance-stabilizing transformation relative to the standard transformation of log2(x+1).This reduces variance purely due to low read counts

```{r Batch effect assessment-PCAs-1}
#PCAs at this point of the analysis are all of the HTSeq-input raw counts
plotPCA(vsd, intgroup=c("condition1", "batch")) 
```
PCA 1-1:PCA demonstrating the raw difference between Nigerian and TCGA patients, which would be expected given that these experiments were done in different settings at different times. Nigerian patients do not appear to have significant differneces based on their batches. (TCGA patients are known not to have significant difference based on batch performed.) However, there does appear to be some clustering mechanism that is separating these TCGA patients into two groups that is not race dependent. 

```{r Batch effect assessment-PCAs-2}
plotPCA(vsd, intgroup=c("condition2"))
```
PCA 1-2:PCA demonstrating association with breast cancer subtype and gene expression. This is noted in both groups and not likely due to batch effect given known biology. 

```{r Batch effect assessment-PCAs-3}
plotPCA(vsd, intgroup=c("condition1", "condition2"))
```
PCA 1-3: Integrated PCA demonstrating combined effect of breast cancer subtype and race. Subtype appears to account for the clustering seen in the TCGA group. 

```{r Heatmap to assess clusters pre-analysis}
#Heatmap to assess clusters pre-analysis
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 100)
hm  <- assay(vsd)[ topVarGenes, ]
hm  <- hm - rowMeans(hm)
anno <- as.data.frame(colData(vsd)[, c("condition1", "condition2")])
rownames(anno) <- colnames(vsd)
pheatmap(hm, annotation_col = anno, cluster_rows=TRUE, cluster_cols=TRUE, clustering_distance_rows = "euclidean", clustering_distance_cols = "euclidean", clustering_method = "complete", show_rownames=FALSE, show_colnames = FALSE, main="Heat map: Relative VST-transformed\ncounts across samples before batch correction")
```
Heatmap 1-1: This heatmaps is only using the top 100 genes and is competely heirarchically clustered. These clusters appear consistent with the PCA results.  

```{r Batch effect assessment-PCAs-4}
#PCA2: Removing the batch effect created by the separate batch processing of the TCGA and Nigerian samples (although this may overcorrect for the actual difference between these two populations).  
mat <- assay(vsd) 
mat <- limma::removeBatchEffect(mat, vsd$condition1, vsd$batch)
assay(vsd) <- mat
plotPCA(vsd, intgroup=c("condition1", "condition2"))
```
PCA 2-1: We have accounted for batch effect in this PCA only using the limma removeBatchEffect method.This largely accounts for the batch effect noticed in PCA1-Nigerian vs TCGA. There remain some outliers, primarily from the Nigerian patients. 

```{r Heatmap to assess clusters after batch correction}
#Heatmap to assess clusters post-analysis
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 100)
hm  <- assay(vsd)[ topVarGenes, ]
hm  <- hm - rowMeans(hm)
anno <- as.data.frame(colData(vsd)[, c("condition1", "condition2")])
rownames(anno) <- colnames(vsd)
pheatmap(hm, annotation_col = anno, cluster_rows=TRUE, cluster_cols=TRUE, clustering_distance_rows = "euclidean", clustering_distance_cols = "euclidean", clustering_method = "complete", show_rownames=FALSE, show_colnames = FALSE, main="Heat map: Relative VST-transformed\ncounts across samples after batch correction")
```
Heatmap 2-1: This heatmap after batch correction with limma's removeBatchEffect method is only using the top 100 genes and is competely heirarchically clustered. These clusters appear consistent with the PCA results as well with clustering enabled between the breast cancer subtypes and finer interspersing of the Nigerian vs. TCGA characteristic based on breast cancer subtype.

```{r: Boxplots of counts before and after DESeq2 normalization}
boxplot(log2(counts(dds)), las=2, ylab="log2(Raw Counts)", xlab="Samples", main="Boxplot of raw counts before\n batch correction/transformation")

boxplot(log2(counts(dds, normalized=TRUE)), las=2, ylab="log2(Raw Counts)", xlab="Samples", main="Boxplot of batch corrected/transformed counts")
```
These boxplots demonstrate the effect of the DESeq2 normalization process. The normalized DESeq2 values are incorporated into the batch corrected model (as below). 

#Using sva to account for batch effect related to the Nigerian/TCGA samples
```{r Batch effect assessment, message=FALSE}
dat  <- counts(dds, normalized = TRUE)
idx  <- rowMeans(dat) > 1
dat  <- dat[idx, ]
mod  <- model.matrix(~condition1, colData(dds))
mod0 <- model.matrix(~   1, colData(dds))
svseq <- svaseq(dat, mod, mod0, n.sv = 2)

sampleTablebe <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          be1=svseq$sv[,1],
                          be2=svseq$sv[,2],
                          batch=batchval)

sampleTablebe$sampleCondition <- paste(sampleTablebe$condition1, sampleTablebe$condition2, sep=".")

ddsHTSeqMFbe <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTablebe,
                                       directory=FOLDER,
                                       design=~sampleCondition + be1 + be2)

ddsHTSeqMFbe <- ddsHTSeqMFbe[rowSums(counts(ddsHTSeqMFbe)) > 1, ]
```

#Differential expression analysis
This is the actual differential expression analysis with false discovery threshold of 0.05. I am performing this for both the standard and batch-effect-corrected versions of the DESeq2 model to allow comparison. 
```{r Differential expression analysis and visualization, cache=TRUE}
ddsMF <- DESeq(ddsHTSeqMF) #This will refer to a version that does not include batch correction outside of what is accomplished through DESeq2 alone. 
ddsMFbe <-DESeq(ddsHTSeqMFbe) #This will refer to a version that batch-corrects using sva for 2 variables and overtly acknowledges that the TCGA-Nigerian platform is part of this batch effect issue. 

resultsNames(ddsMF)
resultsNames(ddsMFbe)

#MAPlot: log2 fold changes attributable to the condition over the mean of normalized counts for all the samples in the DESeqDataSet. Points will be colored red if the adjusted p value is less than 0.1. Points which fall out of the window are plotted as open triangles pointing either up or down.
cat("MA Plot 1: Original differential expression")
res <- lfcShrink(ddsMF, coef="sampleCondition_TCGA_white.Basal_vs_Nigerian.Basal", type="apeglm")
DESeq2::plotMA(res, ylim=c(-10,10), xlim=c(0.1,200))

cat("MA Plot 2: Batch-corrected differential expression")
res2 <- lfcShrink(ddsMFbe, coef="sampleCondition_TCGA_white.Basal_vs_Nigerian.Basal", type="apeglm")
DESeq2::plotMA(res2, ylim=c(-10,10), xlim=c(0.1,200))
```
This is just an MA-plot demonstration of the efficacy of the batch effect correction (using TCGA-white basal breast cancer patients vs Nigerian basal breast cancer patients as an example). The second MA plot with batch-corrected differential expression looks like it is more appropriately assessing significantly differentially expressed genes between these two populations. 

This is a check between the TCGA black and white patients to verify this differential expression analysis based on prior work by Stewart et al (PLOS One 2013) and Huo et al (JAMA Oncology 2017). This was only performed with the basal patients as this was the largest subsample. 
```{r DE: TCGA black and TCGA white, cache=TRUE}
diffTCGA<- results(ddsMFbe, contrast=c("sampleCondition", "TCGA_black.Basal", "TCGA_white.Basal"), alpha=0.05)

diffTCGA <- subset(diffTCGA, padj < 0.05) #Selecting only the lowest threshold of significant findings
diffTCGA$temp <- row.names(diffTCGA)
diffTCGA$temp <- gsub("[.].+", "", diffTCGA$temp)

diffTCGA$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffTCGA$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

diffTCGA$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

diffTCGA$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

diffTCGA$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTCGA$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

with(diffTCGA, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in TCGA black and \nTCGA white breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(diffTCGA, diffTCGA$biotype=="protein_coding" & padj<0.01 & abs(log2FoldChange)>25), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(diffTCGA, padj<0.01 & abs(log2FoldChange)>25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffTCGAres <- as.data.frame(diffTCGA)[, ]
write.csv(diffTCGAres, file = "TCGAblackwhite-basal-test.csv")
```


This is a comparison of 32 Nigerian basal cases and 17 TCGA white basal cases. 
```{r DE: Basal breast - Nigerian and TCGA white, cache=TRUE}
resBasal<- results(ddsMFbe, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_white.Basal"), alpha=0.05)
resBasal <- subset(resBasal, padj < 0.05) #Selecting only the lowest threshold of significant findings
resBasal$temp <- row.names(resBasal)
resBasal$temp <- gsub("[.].+", "", resBasal$temp)

resBasal$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

resBasal$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

resBasal$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

resBasal$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

resBasal$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

with(resBasal, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and \nTCGA white breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(resBasal, resBasal$biotype=="protein_coding" & padj<0.01 & abs(log2FoldChange)>25), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resBasal, padj<0.01 & abs(log2FoldChange)>25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffNigerianwhitebasal <- as.data.frame(resBasal)[, ]
write.csv(diffNigerianwhitebasal, file = "Nigerian-white-basal.csv")
```

This is a comparison of 32 Nigerian basal cases and 25 TCGA black basal cases. 
```{r DE: Basal breast - Nigerian and TCGA black, cache=TRUE}
resBasalblack<- results(ddsMFbe, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_black.Basal"), alpha=0.05)
resBasalblack <- subset(resBasalblack, padj < 0.05) #Selecting only the lowest threshold of significant findings
resBasalblack$temp <- row.names(resBasalblack)
resBasalblack$temp <- gsub("[.].+", "", resBasalblack$temp)

resBasalblack$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

resBasalblack$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

resBasalblack$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

resBasalblack$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

resBasalblack$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasalblack$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

with(resBasalblack, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and \nTCGA black breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(resBasalblack, resBasalblack$biotype=="protein_coding" & padj<0.01 & abs(log2FoldChange)>25), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resBasalblack, padj<0.01 & abs(log2FoldChange)>25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffNigerianblackbasal <- as.data.frame(resBasalblack)[, ]
write.csv(diffNigerianblackbasal, file = "Nigerian-black-basal.csv")
```

This is a comparison of 26 Nigerian Her2 cases and 5 TCGA white Her2 cases. There are no TCGA black Her2 cases available for this analysis.
```{r DE Her2 breast - Nigerian and TCGA white, cache=TRUE}
resHer2<- results(ddsMFbe, contrast=c("sampleCondition", "Nigerian.Her2", "TCGA_white.Her2"), alpha=0.05)
resHer2 <- subset(resHer2, padj < 0.05) #Selecting only the lowest threshold of significant findings
resHer2$temp <- row.names(resHer2)
resHer2$temp <- gsub("[.].+", "", resHer2$temp)

resHer2$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

resHer2$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

resHer2$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

resHer2$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

resHer2$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resHer2$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

with(resHer2, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between Her2 \nbreast cancers in Nigerian and \nTCGA white breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(resHer2, resHer2$biotype=="protein_coding" & padj<0.01 & abs(log2FoldChange)>25), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resHer2, padj<0.01 & abs(log2FoldChange)>25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffNigerianwhiteHER2 <- as.data.frame(resHer2)[, ]
write.csv(diffNigerianwhiteHER2, file = "Nigerian-white-HER2.csv")
```

This is a comparison of 16 Nigerian luminal A cases and 8 TCGA white luminal A cases. 
```{r DE LuminalA breast - Nigerian and TCGA white, cache=TRUE}
resLumA<- results(ddsMFbe, contrast=c("sampleCondition", "Nigerian.LumA", "TCGA_white.LumA"), alpha=0.05)
resLumA <- subset(resLumA, padj < 0.05) #Selecting only the lowest threshold of significant findings
resLumA$temp <- row.names(resLumA)
resLumA$temp <- gsub("[.].+", "", resLumA$temp)

resLumA$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

resLumA$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

resLumA$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

resLumA$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

resLumA$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumA$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

with(resLumA, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between Luminal A \nbreast cancers in Nigerian and \nTCGA white breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(resLumA, resLumA$biotype=="protein_coding" & padj<0.01 & abs(log2FoldChange)>25), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resLumA, padj<0.01 & abs(log2FoldChange)>25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffNigerianwhiteLumA <- as.data.frame(resLumA)[, ]
write.csv(diffNigerianwhiteLumA, file = "Nigerian-white-LumA.csv")

```

This is a comparison of 16 Nigerian luminal A cases and 4 TCGA black luminal A cases. 
```{r DE LuminalA breast - Nigerian and TCGA black, cache=TRUE}
resLumAblack<- results(ddsMFbe, contrast=c("sampleCondition", "Nigerian.LumA", "TCGA_black.LumA"), alpha=0.05)
resLumAblack <- subset(resLumAblack, padj < 0.05) #Selecting only the lowest threshold of significant findings
resLumAblack$temp <- row.names(resLumAblack)
resLumAblack$temp <- gsub("[.].+", "", resLumAblack$temp)

resLumAblack$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

resLumAblack$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

resLumAblack$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

resLumAblack$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

resLumAblack$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumAblack$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

with(resLumAblack, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between Luminal A \nbreast cancers in Nigerian and \nTCGA black breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(resLumAblack, resLumAblack$biotype=="protein_coding" & padj<0.01 & abs(log2FoldChange)>25), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resLumAblack, padj<0.01 & abs(log2FoldChange)>25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffNigerianblackLumA <- as.data.frame(resLumAblack)[, ]
write.csv(diffNigerianblackLumA, file = "Nigerian-black-LumA.csv")
```

This is a comparison of 18 nigerian luminal B cases and 9 TCGA white luminal B cases. 
```{r DE LuminalB breast - Nigerian and TCGA white, cache=TRUE}
resLumB<- results(ddsMFbe, contrast=c("sampleCondition", "Nigerian.LumB", "TCGA_white.LumB"), alpha=0.05)
resLumB <- subset(resLumB, padj < 0.05) #Selecting only the lowest threshold of significant findings
resLumB$temp <- row.names(resLumB)
resLumB$temp <- gsub("[.].+", "", resLumB$temp)

resLumB$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

resLumB$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

resLumB$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

resLumB$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

resLumB$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumB$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

with(resLumB, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between Luminal B \nbreast cancers in Nigerian and \nTCGA white breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(resLumB, resLumB$biotype=="protein_coding" & padj<0.01 & abs(log2FoldChange)>25), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resLumB, padj<0.01 & abs(log2FoldChange)>25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffNigerianwhiteLumB <- as.data.frame(resLumB)[, ]
write.csv(diffNigerianwhiteLumB, file = "Nigerian-white-LumB.csv")
```

This is a comparison of 18 Nigerian luminal B cases and 4 TCGA black luminal B cases. 
```{r DE LuminalB breast - Nigerian and TCGA black, cache=TRUE}
resLumBblack<- results(ddsMFbe, contrast=c("sampleCondition", "Nigerian.LumB", "TCGA_black.LumB"), alpha=0.05)
resLumBblack <- subset(resLumBblack, padj < 0.05) #Selecting only the lowest threshold of significant findings
resLumBblack$temp <- row.names(resLumBblack)
resLumBblack$temp <- gsub("[.].+", "", resLumBblack$temp)

resLumBblack$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

resLumBblack$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

resLumBblack$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

resLumBblack$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

resLumBblack$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resLumBblack$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

with(resLumBblack, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between Luminal B \nbreast cancers in Nigerian and \nTCGA black breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(resLumBblack, resLumBblack$biotype=="protein_coding" & padj<0.01 & abs(log2FoldChange)>25), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resLumBblack, padj<0.01 & abs(log2FoldChange)>25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffNigerianblackLumB <- as.data.frame(resLumBblack)[, ]
write.csv(diffNigerianblackLumB, file = "Nigerian-black-LumB.csv")
```

#Validation of results
```{r Checking distribution of all results by chromosome}
chromecheck <- factor(diffNigerianblackbasal$chr)
plot(table(chromecheck))
chromecheck2 <- factor(diffNigerianwhitebasal$chr)
plot(table(chromecheck2))
chromecheck3 <- factor(diffNigerianwhiteHER2$chr)
plot(table(chromecheck3))
chromecheck4 <- factor(diffNigerianwhiteLumA$chr)
plot(table(chromecheck4))
chromecheck5 <- factor(diffNigerianblackLumA$chr)
plot(table(chromecheck5))
chromecheck6 <- factor(diffNigerianwhiteLumB$chr)
plot(table(chromecheck6))
chromecheck7 <- factor(diffNigerianblackLumB$chr)
plot(table(chromecheck7))
```
*****Given that each of the differential expression analyses produce a similar distribution along the chromosomes, this is concerning for a problem with how the differential expression is run.*****


#Repeat validation of results without batch effect
```{r Repeat analysis without batch effect}
resregBasal<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_white.Basal"), alpha=0.05)
resregBasal <- subset(resregBasal, padj < 0.05) #Selecting only the lowest threshold of significant findings
resregBasal$temp <- row.names(resregBasal)
resregBasal$temp <- gsub("[.].+", "", resregBasal$temp)

resregBasal$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregBasal$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")
chrome2check <- factor(resregBasal$chr)
plot(table(chrome2check))

resregBasalblack<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_black.Basal"), alpha=0.05)
resregBasalblack <- subset(resregBasalblack, padj < 0.05) #Selecting only the lowest threshold of significant findings
resregBasalblack$temp <- row.names(resregBasalblack)
resregBasalblack$temp <- gsub("[.].+", "", resregBasalblack$temp)

resregBasalblack$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregBasalblack$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")
chrome2check2 <- factor(resregBasalblack$chr)
plot(table(chrome2check2))

resregHer2<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.Her2", "TCGA_white.Her2"), alpha=0.05)
resregHer2 <- subset(resregHer2, padj < 0.05) #Selecting only the lowest threshold of significant findings
resregHer2$temp <- row.names(resregHer2)
resregHer2$temp <- gsub("[.].+", "", resregHer2$temp)

resregHer2$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregHer2$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")
chrome2check3 <- factor(resregHer2$chr)
plot(table(chrome2check3))

resregLumA<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.LumA", "TCGA_white.LumA"), alpha=0.05)
resregLumA <- subset(resregLumA, padj < 0.05) #Selecting only the lowest threshold of significant findings
resregLumA$temp <- row.names(resregLumA)
resregLumA$temp <- gsub("[.].+", "", resregLumA$temp)

resregLumA$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregLumA$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")
chrome2check4 <- factor(resregLumA$chr)
plot(table(chrome2check4))

resregLumAblack<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.LumA", "TCGA_black.LumA"), alpha=0.05)
resregLumAblack <- subset(resregLumAblack, padj < 0.05) #Selecting only the lowest threshold of significant findings
resregLumAblack$temp <- row.names(resregLumAblack)
resregLumAblack$temp <- gsub("[.].+", "", resregLumAblack$temp)

resregLumAblack$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregLumAblack$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")
chrome2check5 <- factor(resregLumAblack$chr)
plot(table(chrome2check5))

resregLumB<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.LumB", "TCGA_white.LumB"), alpha=0.05)
resregLumB <- subset(resregLumB, padj < 0.05) #Selecting only the lowest threshold of significant findings
resregLumB$temp <- row.names(resregLumB)
resregLumB$temp <- gsub("[.].+", "", resregLumB$temp)

resregLumB$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregLumB$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")
chrome2check6 <- factor(resregLumB$chr)
plot(table(chrome2check6))

resregLumBblack<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.LumB", "TCGA_black.LumB"), alpha=0.05)
resregLumBblack <- subset(resregLumBblack, padj < 0.05) #Selecting only the lowest threshold of significant findings
resregLumBblack$temp <- row.names(resregLumBblack)
resregLumBblack$temp <- gsub("[.].+", "", resregLumBblack$temp)

resregLumBblack$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resregLumBblack$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")
chrome2check7 <- factor(resregLumBblack$chr)
plot(table(chrome2check7))
```
*****The concerning, persistent distribution in RNA results across chromosomes is persistent across differential expression results regardless of batch effect correction.*****


#Protein-coding assessment -> pending resolution of validation issues
```{r Visualizing protein-coding differences specifically}
proteincoding <- (subset(resBasal, biotype=="protein_coding"))
proteincoding$analysis <- "Nigerian-TCGAwhite.Basal"

proteincoding2 <- (subset(resBasalblack, biotype=="protein_coding"))
proteincoding2$analysis <- "Nigerian-TCGAblack.Basal"

proteincoding3 <- (subset(resHer2, biotype=="protein_coding"))
proteincoding3$analysis <- "Nigerian-TCGAwhite.Her2"

proteincoding4 <- (subset(resLumA, biotype=="protein_coding"))
proteincoding4$analysis <- "Nigerian-TCGAwhite.LuminalA"

proteincoding5 <- (subset(resLumAblack, biotype=="protein_coding"))
proteincoding5$analysis <- "Nigerian-TCGAblack.LuminalA"

proteincoding6 <- (subset(resLumB, biotype=="protein_coding"))
proteincoding6$analysis <- "Nigerian-TCGAwhite.LuminalB"

proteincoding7 <- (subset(resLumBblack, biotype=="protein_coding"))
proteincoding7$analysis <- "Nigerian-TCGAblack.LuminalB"

proteincoding=rbind(proteincoding, proteincoding2, proteincoding3, proteincoding4, proteincoding5, proteincoding6, proteincoding7)

table(proteincoding$analysis)
summary(proteincoding)

proteincodingsig <- head(proteincoding[order(proteincoding$log2FoldChange, proteincoding$padj), ],100)
proteincodingsig2 <- tail(proteincoding[ order(proteincoding$log2FoldChange, proteincoding$padj), ],100)
head(proteincodingsig)
tail(proteincodingsig2)
write.table(proteincodingsig,"all-analysis-protein-coding-genes-head",sep="\t",row.names=TRUE)
write.table(proteincodingsig2,"all-analysis-protein-coding-genes-tail",sep="\t",row.names=TRUE)
```
These are the strongest differentially-expressed protein-coding genes across all analyses.
