---
title: "Nigerian-TCGA Differential Expression Analysis Using DeSeq2 - Protein Coding Genes Only for whole-genome samples"
author: "Padma Sheila Rajagopal, MD MPH"
date: "11/17/2019"

site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    df_print: paged
    toc: true
    theme: spacelab
    highlight: textmate
---
```{r Setup, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE)
library(gplots)
library(ggbiplot)
library(ggplot2)
library(edgeR)
library("DESeq2")
library("RUVSeq")
library(vsn)
library("genefilter")
library(pheatmap)
library(clusterProfiler)
library(pathview)
library("AnnotationDbi")
library(checkmate)
library(affy)
library("dplyr")
library(stringi)
library("hexbin")
library("org.Hs.eg.db")
library('GenomicFeatures')
library(EnsDb.Hsapiens.v75)
library("sva")
library(limma)
library(calibrate)
library(ggfortify)
library("ashr")
library(preprocessCore)
library("BiocParallel")
library(RColorBrewer)
library(Glimma)
library(clusterProfiler)
library(pathview)
library(AnnotationHub)

register(MulticoreParam(4))
setwd("~/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants-proteincoding-IHC")
```

#Translation from HTSeq raw counts -> Count Matrix
I have 66 TCGA patients with whole-genome sequencing data and RNAseq data as well as 48 Nigerian patients with RNA-seq data. Raw counts were initially processed using HTSeq, so HTSeq data is being formatted for use with DESeq2 and limma-voom. 
```{r Preparing the sampleTable using HTSeq raw counts}
FOLDER <- "/Users/parajago/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants-proteincoding-IHC"
sampleFiles <- grep("htseq.counts",list.files(FOLDER),value=TRUE)

#Differential gene expression setup based on ethnicity
sampleConditionrace <- sampleFiles
countVar2=1
for (sample in sampleConditionrace){
  if (stri_detect_fixed(sample,"LIB")==TRUE){
    sampleConditionrace[countVar2] <- "Nigerian"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"black")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_black"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"white")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_white"
    countVar2=countVar2+1
  } else{
    sampleConditionrace[countVar2] <- "TCGA_other"
    countVar2=countVar2+1
  }
}

#Condition based on IHC subtype 
sampleConditionIHC <- sampleFiles
countVar2=1
for (sample in sampleConditionIHC){
  if (stri_detect_fixed(sample,"Her2")==TRUE){
    sampleConditionIHC[countVar2] <- "Her2"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"TNBC")==TRUE){
    sampleConditionIHC[countVar2] <- "TNBC"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"HRpos")==TRUE){
    sampleConditionIHC[countVar2] <- "HRpos"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"TPBC")==TRUE){
    sampleConditionIHC[countVar2] <- "TPBC"
    countVar2=countVar2+1
  } else{
    sampleConditionIHC[countVar2] <- "Other"
    countVar2=countVar2+1
  }
}

#Condition based on batch (relevant to the Nigerian patients only; no difference in batch for the TCGA patients)
batchval <- sampleFiles
countVar2=1
for (sample in batchval){
  if (stri_detect_fixed(sample,"batch1")==TRUE){
    batchval[countVar2] <- "batch1"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch23")==TRUE){
    batchval[countVar2] <- "batch23"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch4")==TRUE){
    batchval[countVar2] <- "batch4"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch5")==TRUE){
    batchval[countVar2] <- "batch5"
    countVar2=countVar2+1
  } else{
    batchval[countVar2] <- "batchT"
    countVar2=countVar2+1
  }
}

#Conditions based on DNA characteristics including chromosome14 LOH, GATA3 mutation status and TP53 mutation status
sampleLOH <-sampleFiles
countVar2=1
for (sample in sampleLOH){
  if (stri_detect_fixed(sample,"-noLOH-")==TRUE){
    sampleLOH[countVar2] <- "noLOH"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"-LOH-")==TRUE){
    sampleLOH[countVar2] <- "LOH"
    countVar2=countVar2+1
  } else{
    sampleLOH[countVar2] <- NA
    countVar2=countVar2+1
  }
}

sampleGATA3 <-sampleFiles
countVar2=1
for (sample in sampleGATA3){
  if (stri_detect_fixed(sample,"GATA3LOH")==TRUE){
    sampleGATA3[countVar2] <- "GATA3LOH"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"GATA3mut")==TRUE){
    sampleGATA3[countVar2] <- "GATA3mut"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"GATA3wt")==TRUE){
    sampleGATA3[countVar2] <- "GATA3wt"
    countVar2=countVar2+1
  } else{
    sampleGATA3[countVar2] <- "NA"
    countVar2=countVar2+1
  }
}

sampleTP53 <-sampleFiles
countVar2=1
for (sample in sampleTP53){
  if (stri_detect_fixed(sample,"TP53wt")==TRUE){
    sampleTP53[countVar2] <- "TP53wt"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"TP53single")==TRUE){
    sampleTP53[countVar2] <- "TP53single"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"TP53double")==TRUE){
    sampleTP53[countVar2] <- "TP53double"
    countVar2=countVar2+1
  } else{
    sampleTP53[countVar2] <- "NA"
    countVar2=countVar2+1
  }
}

table(sampleConditionrace, sampleConditionIHC)

sampleTable <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionIHC,
                          condition3=sampleLOH,
                          condition4=sampleGATA3,
                          condition5=sampleTP53,
                          batch=batchval)

sampleTable$sampleCondition <- paste(sampleTable$condition1, sampleTable$condition2, sep=".")

ddsHTSeqMF <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTable,
                                       directory=FOLDER,
                                       design=~sampleCondition)

ddsHTSeqMF <- ddsHTSeqMF[rowSums(counts(ddsHTSeqMF)) > 0, ] #Pre-filtering the dataset by removing the rows without any information about gene expression -> this removes 731 genes
```

#Quantile normalization
Please refer to: 
https://parajago.github.io/TCGA-Nigerian-RNAseq/NigerianTCGArawcountsDeSeq2-pc2.html
regarding comparison between the Nigerian and TCGA data sets and why quantile normalization under the limma-voom approach was chosen for primary differential expression analysis in any comparisons across populations. 

##Data visualization
```{r Quantile normalization setup and visualization}
countmatrix <- assay(ddsHTSeqMF) #Raw counts organized into matrix format from individual files
countmatrix2 <- log2(countmatrix + 1) #Basic transformation of the count data 

plot(density(countmatrix2[,1]),lwd=3,ylim=c(0,.30), main="Density of counts with log2[count]+1 transformation ONLY") 
for(i in 1:114){lines(density(countmatrix2[,i]),lwd=3)} #This demonstrates that there is a difference in distributions between the Nigerian and TCGA data with basic log transformation normalization 

norm_countmatrix <- as.matrix(countmatrix2) 
norm_countmatrix = normalize.quantiles(norm_countmatrix)
plot(density(norm_countmatrix[,1]),lwd=3,ylim=c(0,.3), main="Density of counts with quantile normalization")
for(i in 1:114){lines(density(norm_countmatrix[,i]),lwd=3)} #This demonstrates the effect of comparative quantile normalization

colnames (norm_countmatrix) <- colnames (countmatrix2)
rownames (norm_countmatrix) <- rownames (countmatrix2)

norm_countmatrix <- as.data.frame(norm_countmatrix)
countmatrixNigerian <- dplyr::select(norm_countmatrix, contains("LIB"))
plot(density(countmatrixNigerian[,1]),lwd=3,ylim=c(0,.3), main="Density of counts with quantile normalization - Nigerian")
for(i in 1:48){lines(density(countmatrixNigerian[,i]),lwd=3)} #This demonstrates the result of the normalized Nigerian counts separately

tcgacolnames <- colnames(countmatrix)
tcgacolnames <- setdiff(tcgacolnames, colnames(countmatrixNigerian))
countmatrixTCGA <- norm_countmatrix[ , tcgacolnames]
plot(density(countmatrixTCGA[,1]),lwd=3,ylim=c(0,.3), main="Density of counts with quantile normalization - TCGA")
for(i in 1:66){
  lines(density(countmatrixTCGA[,i]),lwd=3);
#  print(colnames(countmatrix)[i])
#  invisible(readline(prompt=i))
  } #This demonstrates the result of the normalized TCGA counts separately

norm_countmatrix <- as.data.frame(norm_countmatrix)
t_norm_countmatrix <- t(norm_countmatrix)

t_norm_countmatrix <- cbind (t_norm_countmatrix, sampleTable) #This binds the characteristics of the original patients to the quantile normalized counts. CBinding was checked to make sure that patients were correctly aligned to characteristics. 

quant.pca <- prcomp(t_norm_countmatrix[,1:19596])
autoplot(quant.pca, data=t_norm_countmatrix, colour='sampleCondition', main="PCA of quantile normalization results prior to DE analysis")
```
In the raw data with log transformation only, we are able to see that there are two peaks corresponding to the two datasets (Nigerian and TCGA). The quantile normalization demonstrates a PCA that has similar clustering. Only ~20% of the distribution of the data set is explained by the PCA1, 2 of the variables.

##Differential expression setup
```{r Differential expression setup, cache=TRUE}
annotation <- as.data.frame(row.names(countmatrix))
colnames(annotation) <- c("GeneID")
annotation$temp <- gsub("[.].+", "", annotation[,1])

annotation$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

annotation$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

annotation$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

annotation$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

annotation$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")
annotation$temp <- NULL

design <- t_norm_countmatrix %>% dplyr::select(sampleCondition)

fc = 1.5 #Subsequent threshold of signifcance for log2 fold change -> 0.58 = log2(1.5)
fdr = 0.05 #Subsequent threshold of significance for p-value (adjusted by FDR)

```

#DE: HER2 14q LOH (Nigerian)
```{r Limma-voom differential expression results for Nigerian HER2+ 14q LOH, cache=TRUE}
designNvsTLOH <- t_norm_countmatrix %>% dplyr::select(condition3)

designNvsTLOH$condition3 <- ifelse (designNvsTLOH$condition3=="noLOH", 0, as.character(designNvsTLOH$condition3))
designNvsTLOH$condition3 <- ifelse (designNvsTLOH$condition3=="LOH", 1, as.character(designNvsTLOH$condition3))

designNvsTLOH$condition3 <- ifelse (designNvsTLOH$condition3==0 | designNvsTLOH$condition3==1, designNvsTLOH$condition3, NA)

designNvsTLOH <- designNvsTLOH %>% subset(is.na(condition3)==FALSE)

designNvsTLOH$noLOH <- ifelse (designNvsTLOH$condition3==0, 1, 0)
designNvsTLOH$LOH <- ifelse (designNvsTLOH$condition3==1, 1, 0)

designNvsTLOH$condition3 <- NULL

mm <- model.matrix(~0+designNvsTLOH$noLOH+designNvsTLOH$LOH)

quantids <- rownames(designNvsTLOH)
rownames(mm) <- quantids
colnames(mm) <- c("noLOH", "LOH")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

v=voom(d,designNvsTLOH,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(noLOH-LOH, levels=colnames(designNvsTLOH))

fit <- lmFit(v, designNvsTLOH)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)
```

No significant differential expression was identified via quantile normalization/voom, which is expected as this method can be overly convservative for inter-group differential expression estimation. We have previously validated DESeq2 for inter-Nigerian comparison. 

```{r DESeq2 differential expression results for Nigerian HER2+ 14q LOH, cache=TRUE}
sampleTableLOH <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          sampleCondition=sampleLOH)

sampleTableLOH <- sampleTableLOH %>% subset(is.na(sampleCondition)==FALSE)

ddsHTSeqMFLOH <- DESeqDataSetFromHTSeqCount(
                                      sampleTable=sampleTableLOH,
                                      directory=FOLDER,
                                      design=~sampleCondition)

ddsHTSeqMFLOH <- ddsHTSeqMFLOH[rowSums(counts(ddsHTSeqMFLOH)) > 0, ] #Pre-filtering the dataset by removing the rows without any  information about gene expression

ddsLOH <- estimateSizeFactors(ddsHTSeqMFLOH) #The size factor is the median ratio of the sample over a "pseudosample": for each gene, the geometric mean of all samples. This accounts for sequencing depth. 

vsd <- vst(ddsHTSeqMFLOH, blind=FALSE) #Variance-stabilizing transformation, only using this since >50 samples

plotPCA(vsd, intgroup=c("sampleCondition"))

ddsLOH <- DESeq(ddsHTSeqMFLOH)
resultsNames(ddsLOH)

cat("MA Plot: Differential expression in Nigerian breast cancer patients based on 14q LOH")
resLOH <- lfcShrink(ddsLOH, coef="sampleCondition_noLOH_vs_LOH", type="ashr", optmethod = "mixEM")
DESeq2::plotMA(resLOH, ylim=c(-10,10), xlim=c(0.1,200))

diffLOH<- results(ddsLOH, contrast=c("sampleCondition", "noLOH", "LOH"), pAdjustMethod ="fdr", alpha=fdr)

diffLOH$foldChange <- NA
row.pos <- which(! is.na(diffLOH$log2FoldChange) & 
                diffLOH$log2FoldChange >= 0)
row.neg <- which(! is.na(diffLOH$log2FoldChange) & 
                diffLOH$log2FoldChange < 0)
diffLOH$foldChange[row.pos] <- 2^diffLOH$log2FoldChange[row.pos]
diffLOH$foldChange[row.neg] <- -2^((-1) * diffLOH$log2FoldChange[row.neg])

nrow(diffLOH)
diffLOH <- diffLOH[(diffLOH$foldChange > fc | diffLOH$foldChange < -fc),] 
diffLOH <- subset(diffLOH, padj < fdr)
nrow(diffLOH)

restemp <- lfcShrink(ddsLOH, contrast=c("sampleCondition", "noLOH", "LOH"), res = diffLOH, type="ashr", optmethod = "mixEM")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between \nbreast cancers in Nigerian patients with and without 14q LOH", xlim=c(-60,60), ylim=c(0,60)))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffLOHtable <- as.data.frame(diffLOH)
diffLOHtable$temp <- row.names(diffLOH)
diffLOHtable$temp <- gsub("[.].+", "", diffLOHtable$temp)

diffLOHtable$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffLOHtable$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffLOHtable$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffLOHtable$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

diffLOHtable$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffLOHtable$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

diffLOHtable$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffLOHtable$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

diffLOHtable$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffLOHtable$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

diffLOHtable$temp <- NULL

diffLOHtable <- diffLOHtable %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(diffLOHtable, 10, foldChange)
top_n(diffLOHtable, -10, foldChange)

diffLOHtable$baseMean <- NULL
write.csv(diffLOHtable, file = "RNA-DE-Nigerian-14q-LOH.csv", row.names =FALSE)


#Pathway analysis based on 14q LOH in Nigerian samples
diffLOHpath <- results(ddsLOH, contrast=c("sampleCondition", "noLOH", "LOH"), pAdjustMethod ="fdr", alpha=fdr)
diffLOHpath <- as.data.frame(diffLOHpath)

diffLOHpath$foldChange <- NA
row.pos <- which(! is.na(diffLOHpath$log2FoldChange) & 
                diffLOHpath$log2FoldChange >= 0)
row.neg <- which(! is.na(diffLOHpath$log2FoldChange) & 
                diffLOHpath$log2FoldChange < 0)
diffLOHpath$foldChange[row.pos] <- 2^diffLOHpath$log2FoldChange[row.pos]
diffLOHpath$foldChange[row.neg] <- -2^((-1) * diffLOHpath$log2FoldChange[row.neg])

diffLOHpath$ENSEMBL <- row.names(diffLOHpath)
diffLOHpath$temp <- row.names(diffLOHpath)
diffLOHpath$temp <- gsub("[.].+", "", diffLOHpath$temp)

diffLOHpath$SYMBOL <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffLOHpath$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffLOHpath$baseMean <- NULL
diffLOHpath$log2FoldChange <- NULL
diffLOHpath$temp <- NULL

diffLOHpath.flt <- diffLOHpath[(diffLOHpath$foldChange > fc | diffLOHpath$foldChange < -fc),] 
diffLOHpath.flt <- subset(diffLOHpath, padj < fdr)

genes.all <- diffLOHpath
genes.sig <- diffLOHpath.flt
genes.all$ENSEMBL <- gsub('[.]\\d+', '', genes.all$ENSEMBL, perl = TRUE)
genes.sig$ENSEMBL <- gsub('[.]\\d+', '', genes.sig$ENSEMBL, perl = TRUE)

genes.all.anno <- bitr(geneID   =  genes.all$ENSEMBL, 
                      fromType = 'GENEID', 
                      toType   = c('ENTREZID', 'SYMBOL'), 
                      OrgDb    = 'EnsDb.Hsapiens.v75', 
                      drop     = TRUE)

genes.all.anno <- genes.all.anno[
    which(! duplicated(genes.all.anno$ENTREZID)), ]
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.all.anno$ENSEMBL <- genes.all.anno$GENEID
genes.all.anno$GENEID <- NULL

genes.all.anno <- merge(genes.all.anno, genes.all, by = 'ENSEMBL')
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.sig.anno <- genes.all.anno[genes.all.anno$ENSEMBL %in% 
                                genes.sig$ENSEMBL,]

gene.list <- genes.all.anno$foldChange
names(gene.list) <- genes.all.anno$ENTREZID
gene.list <- sort(gene.list, decreasing = TRUE)

ego <- enrichGO(gene          = genes.sig.anno$ENTREZID,
                universe      = as.character(genes.all.anno$ENTREZID),
                OrgDb         = 'org.Hs.eg.db',
                ont           = "BP",
                pAdjustMethod = "fdr",
                pvalueCutoff  = 0.05,
                readable      = TRUE)

as.data.frame(ego)
save(ego, file="GO-LOH.significantgenes.fdr0.05.fc1.5.enrichGO.RData")
write.csv(ego, file="GO-LOH.significantgenes.fdr0.05.fc1.5.enrichGO.csv")

options(jupyter.plot_mimetypes = "image/svg+xml") 
options(repr.plot.width = 10, repr.plot.height = 5)

egokegg <- ego
for(i in 1:5) { 
  egokegg <- dropGO(egokegg, level = i)
}

p1 <- barplot(egokegg)
p2 <- dotplot(egokegg)

kk <- enrichKEGG(gene          = genes.sig.anno$ENTREZID,
                 universe      = as.character(genes.all.anno$ENTREZID),
                 organism      = "hsa",
                 pAdjustMethod = "fdr",
                 pvalueCutoff  = 0.05)

plot(p1)
plot(p2)

save(kk, file="GO-LOH.significantgenes.fdr0.05.fc1.5.enrichKEGG.RData")
write.csv(kk@result, file="GO-LOH.significantgenes.fdr0.05.fc1.5.enrichKEGG.csv")
```


#DE: GATA3 (Nigerian)
Among the Nigerian patients, 9 demonstated GATA3 loss of heterozygosity and 9 demonstrated GATA3 mutations, vs. 30 patients who had GATA3 WT. We therefore performed differential expression analysis within this group to see the comparative effects. 

```{r DESeq2 differential expression results for Nigerian GATA3 mutation status, cache=TRUE}
sampleTableGATA <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleGATA3,
                          batch=sampleConditionIHC)

sampleTableGATA$condition1 <- ifelse (grepl("Nigerian",sampleTableGATA$condition1), "Nigerian", NA)
sampleTableGATA <- sampleTableGATA %>% subset(is.na(sampleTableGATA$condition1)==FALSE)

sampleTableGATA$sampleCondition <- sampleTableGATA$condition2
sampleTableGATA$condition1 <- NULL
sampleTableGATA$condition2 <- NULL

ddsHTSeqMFGATA <- DESeqDataSetFromHTSeqCount(
                                      sampleTable=sampleTableGATA,
                                      directory=FOLDER,
                                      design=~sampleCondition+batch)


ddsHTSeqMFGATA <- ddsHTSeqMFGATA[rowSums(counts(ddsHTSeqMFGATA)) > 0, ] #Pre-filtering the dataset by removing the rows without any information about gene expression
ddsHTSeqMFGATA$sampleCondition <- relevel(ddsHTSeqMFGATA$sampleCondition, ref = "GATA3wt")

ddsGATA <- estimateSizeFactors(ddsHTSeqMFGATA)
vsdGATA <- vst(ddsHTSeqMFGATA, blind=FALSE)

plotPCA(vsdGATA, intgroup=c("sampleCondition", "batch"))
ddsGATA <- DESeq(ddsHTSeqMFGATA)
resultsNames(ddsGATA)


cat("MA Plot: Differential expression based on GATA3 LOH status in Nigerians")
resGATALOH <- lfcShrink(ddsGATA, coef="sampleCondition_GATA3LOH_vs_GATA3wt", type="ashr")
DESeq2::plotMA(resGATALOH, ylim=c(-10,10), xlim=c(0.1,200))

cat("MA Plot: Differential expression based on GATA3 mutation status in Nigerians")
resGATAmut <- lfcShrink(ddsGATA, coef="sampleCondition_GATA3mut_vs_GATA3wt", type="ashr")
DESeq2::plotMA(resGATAmut, ylim=c(-10,10), xlim=c(0.1,200))

plotCounts(ddsGATA, gene="ENSG00000107485.11", intgroup=c("sampleCondition", "batch"), main="Distribution of GATA3 expression by breast cancer subtype / mutation status in Nigerian patients")

diffGATALOH<- results(ddsGATA, contrast=c("sampleCondition", "GATA3LOH", "GATA3wt"), pAdjustMethod ="fdr", alpha=fdr)

diffGATALOH$foldChange <- NA
row.pos <- which(! is.na(diffGATALOH$log2FoldChange) & 
                diffGATALOH$log2FoldChange >= 0)
row.neg <- which(! is.na(diffGATALOH$log2FoldChange) & 
                diffGATALOH$log2FoldChange < 0)
diffGATALOH$foldChange[row.pos] <- 2^diffGATALOH$log2FoldChange[row.pos]
diffGATALOH$foldChange[row.neg] <- -2^((-1) * diffGATALOH$log2FoldChange[row.neg])

nrow(diffGATALOH)
diffGATALOH <- diffGATALOH[(diffGATALOH$foldChange > fc | diffGATALOH$foldChange < -fc),] 
diffGATALOH <- subset(diffGATALOH, padj < fdr)
nrow(diffGATALOH)

restemp <- lfcShrink(ddsGATA, contrast=c("sampleCondition", "GATA3LOH", "GATA3wt"), res = diffGATALOH, type="ashr")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between \nbreast cancers in Nigerian patients with and without GATA3 LOH", xlim=c(-60,60), ylim=c(0,60)))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffGATALOHtable <- as.data.frame(diffGATALOH)
diffGATALOHtable$temp <- row.names(diffGATALOH)
diffGATALOHtable$temp <- gsub("[.].+", "", diffGATALOHtable$temp)

diffGATALOHtable$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATALOHtable$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffGATALOHtable$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATALOHtable$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

diffGATALOHtable$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATALOHtable$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

diffGATALOHtable$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATALOHtable$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

diffGATALOHtable$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATALOHtable$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

diffGATALOHtable$temp <- NULL

diffGATALOHtable <- diffGATALOHtable %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(diffGATALOHtable, 10, foldChange)
top_n(diffGATALOHtable, -10, foldChange)

diffGATALOHtable$baseMean <- NULL
write.csv(diffGATALOHtable, file = "RNA-DE-Nigerian-GATA3-LOH.csv", row.names =FALSE)


diffGATAmut<- results(ddsGATA, contrast=c("sampleCondition", "GATA3mut", "GATA3wt"), pAdjustMethod ="fdr", alpha=fdr)

diffGATAmut$foldChange <- NA
row.pos <- which(! is.na(diffGATAmut$log2FoldChange) & 
                diffGATAmut$log2FoldChange >= 0)
row.neg <- which(! is.na(diffGATAmut$log2FoldChange) & 
                diffGATAmut$log2FoldChange < 0)
diffGATAmut$foldChange[row.pos] <- 2^diffGATAmut$log2FoldChange[row.pos]
diffGATAmut$foldChange[row.neg] <- -2^((-1) * diffGATAmut$log2FoldChange[row.neg])

nrow(diffGATAmut)
diffGATAmut <- diffGATAmut[(diffGATAmut$foldChange > fc | diffGATAmut$foldChange < -fc),] 
diffGATAmut <- subset(diffGATAmut, padj < fdr)
nrow(diffGATAmut)

restemp <- lfcShrink(ddsGATA, contrast=c("sampleCondition", "GATA3mut", "GATA3wt"), res = diffGATAmut, type="ashr")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between \nbreast cancers in Nigerian patients with and without GATA3 mutations", xlim=c(-60,60), ylim=c(0,60)))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffGATAmuttable <- as.data.frame(diffGATAmut)
diffGATAmuttable$temp <- row.names(diffGATAmut)
diffGATAmuttable$temp <- gsub("[.].+", "", diffGATAmuttable$temp)

diffGATAmuttable$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATAmuttable$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffGATAmuttable$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATAmuttable$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

diffGATAmuttable$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATAmuttable$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

diffGATAmuttable$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATAmuttable$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

diffGATAmuttable$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffGATAmuttable$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

diffGATAmuttable$temp <- NULL

diffGATAmuttable <- diffGATAmuttable %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(diffGATAmuttable, 10, foldChange)
top_n(diffGATAmuttable, -10, foldChange)

diffGATAmuttable$baseMean <- NULL
write.csv(diffGATAmuttable, file = "RNA-DE-Nigerian-GATA3-mutations.csv", row.names =FALSE)
```

All of the TCGA black patients in this sample were GATA3 wild type, so a differential expression analysis could not be performed. Similarly, there were only 4 patients with LOH and 2 with mutations in the TCGA white patients, so any differential expression analysis would be difficult to be performed reliably. 

#DE: TP53 (Nigerian)
The majority of Nigerian patients had TP53 double hits in their tumors; 10 patients had single hits and 6 patients had WT TP53. We therefore performed differential expression analysis to determine if there were any significant additional expression changes between these groups within the Nigerian patients. 
```{r DESeq2 differential expression results for Nigerian TP53 mutation status, cache=TRUE}
sampleTableTP53 <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleTP53,
                          batch=sampleConditionIHC)

sampleTableTP53$condition1 <- ifelse (grepl("Nigerian",sampleTableTP53$condition1), "Nigerian", NA)
sampleTableTP53 <- sampleTableTP53 %>% subset(is.na(sampleTableTP53$condition1)==FALSE)

sampleTableTP53$sampleCondition <- sampleTableTP53$condition2
sampleTableTP53$condition1 <- NULL
sampleTableTP53$condition2 <- NULL

ddsHTSeqMFTP53 <- DESeqDataSetFromHTSeqCount(
                                      sampleTable=sampleTableTP53,
                                      directory=FOLDER,
                                      design=~sampleCondition+batch)


ddsHTSeqMFTP53 <- ddsHTSeqMFTP53[rowSums(counts(ddsHTSeqMFTP53)) > 0, ] #Pre-filtering the dataset by removing the rows without any information about gene expression
ddsHTSeqMFTP53$sampleCondition <- relevel(ddsHTSeqMFTP53$sampleCondition, ref = "TP53wt")

ddsTP53 <- estimateSizeFactors(ddsHTSeqMFTP53)
vsdTP53 <- vst(ddsHTSeqMFTP53, blind=FALSE)

plotPCA(vsdTP53, intgroup=c("sampleCondition", "batch"))
ddsTP53 <- DESeq(ddsHTSeqMFTP53)
resultsNames(ddsTP53)


cat("MA Plot: Differential expression based on TP53 single-hit status in Nigerians")
resTP53singlehit <- lfcShrink(ddsTP53, coef="sampleCondition_TP53single_vs_TP53wt", type="ashr")
DESeq2::plotMA(resTP53singlehit, ylim=c(-10,10), xlim=c(0.1,200))

cat("MA Plot: Differential expression based on TP53 double-hit status in Nigerians")
resTP53doublehit <- lfcShrink(ddsTP53, coef="sampleCondition_TP53double_vs_TP53wt", type="ashr")
DESeq2::plotMA(resTP53doublehit, ylim=c(-10,10), xlim=c(0.1,200))

plotCounts(ddsTP53, gene="ENSG00000141510.11", intgroup=c("sampleCondition", "batch"), main="Distribution of TP53 expression by breast cancer subtype / hit status in Nigerian patients")

diffTP53single<- results(ddsTP53, contrast=c("sampleCondition", "TP53single", "TP53wt"), pAdjustMethod ="fdr", alpha=fdr)

diffTP53single$foldChange <- NA
row.pos <- which(! is.na(diffTP53single$log2FoldChange) & 
                diffTP53single$log2FoldChange >= 0)
row.neg <- which(! is.na(diffTP53single$log2FoldChange) & 
                diffTP53single$log2FoldChange < 0)
diffTP53single$foldChange[row.pos] <- 2^diffTP53single$log2FoldChange[row.pos]
diffTP53single$foldChange[row.neg] <- -2^((-1) * diffTP53single$log2FoldChange[row.neg])

nrow(diffTP53single)
diffTP53single <- diffTP53single[(diffTP53single$foldChange > fc | diffTP53single$foldChange < -fc),] 
diffTP53single <- subset(diffTP53single, padj < fdr)
nrow(diffTP53single)

restemp <- lfcShrink(ddsTP53, contrast=c("sampleCondition", "TP53single", "TP53wt"), res = diffTP53single, type="ashr")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between \nbreast cancers in Nigerian patients with and without TP53 single-hit mutations", xlim=c(-60,60), ylim=c(0,60)))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffTP53singletable <- as.data.frame(diffTP53single)
diffTP53singletable$temp <- row.names(diffTP53single)
diffTP53singletable$temp <- gsub("[.].+", "", diffTP53singletable$temp)

diffTP53singletable$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53singletable$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53singletable$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53singletable$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

diffTP53singletable$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

diffTP53singletable$temp <- NULL

diffTP53singletable <- diffTP53singletable %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(diffTP53singletable, 10, foldChange)
top_n(diffTP53singletable, -10, foldChange)

diffTP53singletable$baseMean <- NULL
write.csv(diffTP53singletable, file = "RNA-DE-Nigerian-TP53-singlehit.csv", row.names =FALSE)


diffTP53double<- results(ddsTP53, contrast=c("sampleCondition", "TP53double", "TP53wt"), pAdjustMethod ="fdr", alpha=fdr)

diffTP53double$foldChange <- NA
row.pos <- which(! is.na(diffTP53double$log2FoldChange) & 
                diffTP53double$log2FoldChange >= 0)
row.neg <- which(! is.na(diffTP53double$log2FoldChange) & 
                diffTP53double$log2FoldChange < 0)
diffTP53double$foldChange[row.pos] <- 2^diffTP53double$log2FoldChange[row.pos]
diffTP53double$foldChange[row.neg] <- -2^((-1) * diffTP53double$log2FoldChange[row.neg])

nrow(diffTP53double)
diffTP53double <- diffTP53double[(diffTP53double$foldChange > fc | diffTP53double$foldChange < -fc),] 
diffTP53double <- subset(diffTP53double, padj < fdr)
nrow(diffTP53double)

restemp <- lfcShrink(ddsTP53, contrast=c("sampleCondition", "TP53double", "TP53wt"), res = diffTP53double, type="ashr")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between \nbreast cancers in Nigerian patients with and without TP53 double hit mutations", xlim=c(-60,60), ylim=c(0,60)))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffTP53doubletable <- as.data.frame(diffTP53double)
diffTP53doubletable$temp <- row.names(diffTP53double)
diffTP53doubletable$temp <- gsub("[.].+", "", diffTP53doubletable$temp)

diffTP53doubletable$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53doubletable$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53doubletable$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53doubletable$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

diffTP53doubletable$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

diffTP53doubletable$temp <- NULL

diffTP53doubletable <- diffTP53doubletable %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(diffTP53doubletable, 10, foldChange)
top_n(diffTP53doubletable, -10, foldChange)

diffTP53doubletable$baseMean <- NULL
write.csv(diffTP53doubletable, file = "RNA-DE-Nigerian-TP53-doublehit.csv", row.names =FALSE)
```

Among the TCGA white patients, 19 were TP53 double-hit, 8 were TP53 single-hit and 9 were TP53 WT. Among the TCGA black patients, 14 were TP53 WT and 16 were TP53 single-hit. 

#DE: TP53 (TCGA White)
```{r DESeq2 differential expression results for TCGA TP53 mutation status, cache=TRUE}
sampleTableTP53 <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleTP53,
                          batch=sampleConditionIHC)

sampleTableTP53$condition1 <- ifelse (grepl("TCGA_white",sampleTableTP53$condition1), "TCGA_white", NA)
sampleTableTP53 <- sampleTableTP53 %>% subset(is.na(sampleTableTP53$condition1)==FALSE)

sampleTableTP53$sampleCondition <- sampleTableTP53$condition2
sampleTableTP53$condition1 <- NULL
sampleTableTP53$condition2 <- NULL

ddsHTSeqMFTP53 <- DESeqDataSetFromHTSeqCount(
                                      sampleTable=sampleTableTP53,
                                      directory=FOLDER,
                                      design=~sampleCondition+batch)


ddsHTSeqMFTP53 <- ddsHTSeqMFTP53[rowSums(counts(ddsHTSeqMFTP53)) > 0, ] #Pre-filtering the dataset by removing the rows without any information about gene expression
ddsHTSeqMFTP53$sampleCondition <- relevel(ddsHTSeqMFTP53$sampleCondition, ref = "TP53wt")

ddsTP53 <- estimateSizeFactors(ddsHTSeqMFTP53)
vsdTP53 <- vst(ddsHTSeqMFTP53, blind=FALSE)

plotPCA(vsdTP53, intgroup=c("sampleCondition", "batch"))
ddsTP53 <- DESeq(ddsHTSeqMFTP53)
resultsNames(ddsTP53)


cat("MA Plot: Differential expression based on TP53 single-hit status in TCGA white patients")
resTP53singlehit <- lfcShrink(ddsTP53, coef="sampleCondition_TP53single_vs_TP53wt", type="ashr")
DESeq2::plotMA(resTP53singlehit, ylim=c(-10,10), xlim=c(0.1,200))

cat("MA Plot: Differential expression based on TP53 double-hit status in TCGA white patients")
resTP53doublehit <- lfcShrink(ddsTP53, coef="sampleCondition_TP53double_vs_TP53wt", type="ashr")
DESeq2::plotMA(resTP53doublehit, ylim=c(-10,10), xlim=c(0.1,200))

plotCounts(ddsTP53, gene="ENSG00000141510.11", intgroup=c("sampleCondition", "batch"), main="Distribution of TP53 expression by breast cancer subtype / hit status in TCGA white patients")

diffTP53single<- results(ddsTP53, contrast=c("sampleCondition", "TP53single", "TP53wt"), pAdjustMethod ="fdr", alpha=fdr)

diffTP53single$foldChange <- NA
row.pos <- which(! is.na(diffTP53single$log2FoldChange) & 
                diffTP53single$log2FoldChange >= 0)
row.neg <- which(! is.na(diffTP53single$log2FoldChange) & 
                diffTP53single$log2FoldChange < 0)
diffTP53single$foldChange[row.pos] <- 2^diffTP53single$log2FoldChange[row.pos]
diffTP53single$foldChange[row.neg] <- -2^((-1) * diffTP53single$log2FoldChange[row.neg])

nrow(diffTP53single)
diffTP53single <- diffTP53single[(diffTP53single$foldChange > fc | diffTP53single$foldChange < -fc),] 
diffTP53single <- subset(diffTP53single, padj < fdr)
nrow(diffTP53single)

restemp <- lfcShrink(ddsTP53, contrast=c("sampleCondition", "TP53single", "TP53wt"), res = diffTP53single, type="ashr")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between \nbreast cancers in TCGA white patients with and without TP53 single-hit mutations", xlim=c(-60,60), ylim=c(0,60)))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffTP53singletable <- as.data.frame(diffTP53single)
diffTP53singletable$temp <- row.names(diffTP53single)
diffTP53singletable$temp <- gsub("[.].+", "", diffTP53singletable$temp)

diffTP53singletable$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53singletable$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53singletable$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53singletable$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

diffTP53singletable$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

diffTP53singletable$temp <- NULL

diffTP53singletable <- diffTP53singletable %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(diffTP53singletable, 10, foldChange)
top_n(diffTP53singletable, -10, foldChange)

diffTP53singletable$baseMean <- NULL
write.csv(diffTP53singletable, file = "RNA-DE-TCGA_white-TP53-singlehit.csv", row.names =FALSE)


diffTP53double<- results(ddsTP53, contrast=c("sampleCondition", "TP53double", "TP53wt"), pAdjustMethod ="fdr", alpha=fdr)

diffTP53double$foldChange <- NA
row.pos <- which(! is.na(diffTP53double$log2FoldChange) & 
                diffTP53double$log2FoldChange >= 0)
row.neg <- which(! is.na(diffTP53double$log2FoldChange) & 
                diffTP53double$log2FoldChange < 0)
diffTP53double$foldChange[row.pos] <- 2^diffTP53double$log2FoldChange[row.pos]
diffTP53double$foldChange[row.neg] <- -2^((-1) * diffTP53double$log2FoldChange[row.neg])

nrow(diffTP53double)
diffTP53double <- diffTP53double[(diffTP53double$foldChange > fc | diffTP53double$foldChange < -fc),] 
diffTP53double <- subset(diffTP53double, padj < fdr)
nrow(diffTP53double)

restemp <- lfcShrink(ddsTP53, contrast=c("sampleCondition", "TP53double", "TP53wt"), res = diffTP53double, type="ashr")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between \nbreast cancers in TCGA white patients with and without TP53 double hit mutations", xlim=c(-60,60), ylim=c(0,60)))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffTP53doubletable <- as.data.frame(diffTP53double)
diffTP53doubletable$temp <- row.names(diffTP53double)
diffTP53doubletable$temp <- gsub("[.].+", "", diffTP53doubletable$temp)

diffTP53doubletable$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53doubletable$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53doubletable$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53doubletable$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

diffTP53doubletable$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53doubletable$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

diffTP53doubletable$temp <- NULL

diffTP53doubletable <- diffTP53doubletable %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(diffTP53doubletable, 10, foldChange)
top_n(diffTP53doubletable, -10, foldChange)

diffTP53doubletable$baseMean <- NULL
write.csv(diffTP53doubletable, file = "RNA-DE-TCGA_white-TP53-doublehit.csv", row.names =FALSE)
```

#DE: TP53 (TCGA Black)
```{r DESeq2 differential expression results for TCGA black TP53 mutation status, cache=TRUE}
sampleTableTP53 <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleTP53,
                          batch=sampleConditionIHC)

sampleTableTP53$condition1 <- ifelse (grepl("TCGA_black",sampleTableTP53$condition1), "TCGA_black", NA)
sampleTableTP53 <- sampleTableTP53 %>% subset(is.na(sampleTableTP53$condition1)==FALSE)

sampleTableTP53$sampleCondition <- sampleTableTP53$condition2
sampleTableTP53$condition1 <- NULL
sampleTableTP53$condition2 <- NULL

ddsHTSeqMFTP53 <- DESeqDataSetFromHTSeqCount(
                                      sampleTable=sampleTableTP53,
                                      directory=FOLDER,
                                      design=~sampleCondition+batch)

ddsHTSeqMFTP53 <- ddsHTSeqMFTP53[rowSums(counts(ddsHTSeqMFTP53)) > 0, ] #Pre-filtering the dataset by removing the rows without any information about gene expression
ddsHTSeqMFTP53$sampleCondition <- relevel(ddsHTSeqMFTP53$sampleCondition, ref = "TP53wt")

ddsTP53 <- estimateSizeFactors(ddsHTSeqMFTP53)
vsdTP53 <- vst(ddsHTSeqMFTP53, blind=FALSE)

plotPCA(vsdTP53, intgroup=c("sampleCondition", "batch"))
ddsTP53 <- DESeq(ddsHTSeqMFTP53)
resultsNames(ddsTP53)

cat("MA Plot: Differential expression based on TP53 single-hit status in TCGA black patients")
resTP53singlehit <- lfcShrink(ddsTP53, coef="sampleCondition_TP53single_vs_TP53wt", type="ashr")
DESeq2::plotMA(resTP53singlehit, ylim=c(-10,10), xlim=c(0.1,200))

plotCounts(ddsTP53, gene="ENSG00000141510.11", intgroup=c("sampleCondition", "batch"), main="Distribution of TP53 expression by breast cancer subtype / hit status in TCGA black patients")

diffTP53single<- results(ddsTP53, contrast=c("sampleCondition", "TP53single", "TP53wt"), pAdjustMethod ="fdr", alpha=fdr)

diffTP53single$foldChange <- NA
row.pos <- which(! is.na(diffTP53single$log2FoldChange) & 
                diffTP53single$log2FoldChange >= 0)
row.neg <- which(! is.na(diffTP53single$log2FoldChange) & 
                diffTP53single$log2FoldChange < 0)
diffTP53single$foldChange[row.pos] <- 2^diffTP53single$log2FoldChange[row.pos]
diffTP53single$foldChange[row.neg] <- -2^((-1) * diffTP53single$log2FoldChange[row.neg])

nrow(diffTP53single)
diffTP53single <- diffTP53single[(diffTP53single$foldChange > fc | diffTP53single$foldChange < -fc),] 
diffTP53single <- subset(diffTP53single, padj < fdr)
nrow(diffTP53single)

restemp <- lfcShrink(ddsTP53, contrast=c("sampleCondition", "TP53single", "TP53wt"), res = diffTP53single, type="ashr")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between \nbreast cancers in TCGA black patients with and without TP53 single-hit mutations", xlim=c(-60,60), ylim=c(0,60)))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

diffTP53singletable <- as.data.frame(diffTP53single)
diffTP53singletable$temp <- row.names(diffTP53single)
diffTP53singletable$temp <- gsub("[.].+", "", diffTP53singletable$temp)

diffTP53singletable$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53singletable$biotype <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="GENEBIOTYPE",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53singletable$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

diffTP53singletable$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

diffTP53singletable$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=diffTP53singletable$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")

diffTP53singletable$temp <- NULL

diffTP53singletable <- diffTP53singletable %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(diffTP53singletable, 10, foldChange)
top_n(diffTP53singletable, -10, foldChange)

diffTP53singletable$baseMean <- NULL
write.csv(diffTP53singletable, file = "RNA-DE-TCGA_black-TP53-singlehit.csv", row.names =FALSE)
```




#DE: Nigerian/TCGA White - TNBC
```{r Limma-voom differential expression results for Nigerian/TCGA white TNBC patients, eval=FALSE, include=FALSE, cache=TRUE}
designNvsW <- design
designNvsW$sampleCondition <- ifelse (designNvsW$sampleCondition=="TCGA_white.TNBC", 0, as.character(designNvsW$sampleCondition))
designNvsW$sampleCondition <- ifelse (designNvsW$sampleCondition=="Nigerian.TNBC", 1, as.character(designNvsW$sampleCondition))

designNvsW$sampleCondition <- ifelse (designNvsW$sampleCondition==0 | designNvsW$sampleCondition==1, designNvsW$sampleCondition, NA)

designNvsW <- designNvsW %>% subset(is.na(sampleCondition)==FALSE)

designNvsW$TCGA_white.TNBC <- ifelse (designNvsW$sampleCondition==0, 1, 0)
designNvsW$Nigerian.TNBC <- ifelse (designNvsW$sampleCondition==1, 1, 0)

designNvsW$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsW$TCGA_white.TNBC+designNvsW$Nigerian.TNBC)

quantids <- rownames(designNvsW)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

v=voom(d,designNvsW,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.TNBC-Nigerian.TNBC, levels=colnames(designNvsW))

fit <- lmFit(v, designNvsW)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between TNBC breast cancers in Nigerian and TCGA white patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between TNBC \nbreast cancers in Nigerian and TCGA white breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

write.csv(df_limmaprint, file = "Nigerian-TCGAwhite-TNBC.csv", row.names = FALSE)
```

#DE: Nigerian/TCGA Black - TNBC
```{r Limma-voom differential expression results for Nigerian/TCGA black TNBC patients, eval=FALSE, include=FALSE, cache=TRUE}
designNvsB <- design
designNvsB$sampleCondition <- ifelse (designNvsB$sampleCondition=="TCGA_black.TNBC", 0, as.character(designNvsB$sampleCondition))
designNvsB$sampleCondition <- ifelse (designNvsB$sampleCondition=="Nigerian.TNBC", 1, as.character(designNvsB$sampleCondition))

designNvsB$sampleCondition <- ifelse (designNvsB$sampleCondition==0 | designNvsB$sampleCondition==1, designNvsB$sampleCondition, NA)

designNvsB <- designNvsB %>% subset(is.na(sampleCondition)==FALSE)

designNvsB$TCGA_black.TNBC <- ifelse (designNvsB$sampleCondition==0, 1, 0)
designNvsB$Nigerian.TNBC <- ifelse (designNvsB$sampleCondition==1, 1, 0)

designNvsB$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsB$TCGA_black.TNBC+designNvsB$Nigerian.TNBC)

quantids <- rownames(designNvsB)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_black", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

v=voom(d,designNvsB,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_black.TNBC-Nigerian.TNBC, levels=colnames(designNvsB))

fit <- lmFit(v, designNvsB)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between TNBC breast cancers in Nigerian and TCGA black patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between TNBC \nbreast cancers in Nigerian and TCGA black breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

write.csv(df_limmaprint, file = "Nigerian-TCGAblack-TNBC.csv", row.names = FALSE)
```

#DE: Nigerian/TCGA White - HR+\HER2-
```{r Limma-voom differential expression results for Nigerian/TCGA white HR+HER2-, eval=FALSE, include=FALSE, cache=TRUE}
designNvsWHR <- design
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="TCGA_white.HRpos", 0, as.character(designNvsWHR$sampleCondition))
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="Nigerian.HRpos", 1, as.character(designNvsWHR$sampleCondition))

designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition==0 | designNvsWHR$sampleCondition==1, designNvsWHR$sampleCondition, NA)

designNvsWHR <- designNvsWHR %>% subset(is.na(sampleCondition)==FALSE)

designNvsWHR$TCGA_white.HRpos <- ifelse (designNvsWHR$sampleCondition==0, 1, 0)
designNvsWHR$Nigerian.HRpos <- ifelse (designNvsWHR$sampleCondition==1, 1, 0)

designNvsWHR$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsWHR$TCGA_white.HRpos+designNvsWHR$Nigerian.HRpos)

quantids <- rownames(designNvsWHR)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

v=voom(d,designNvsWHR,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.HRpos-Nigerian.HRpos, levels=colnames(designNvsWHR))

fit <- lmFit(v, designNvsWHR)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between HR-positive breast cancers in Nigerian and TCGA white patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between HR-postive \nbreast cancers in Nigerian and TCGA white breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)
write.csv(df_limmaprint, file = "TCGA_white-Nigerian-HRpos.csv", row.names = FALSE)
```

#DE: Nigerian/TCGA Black - HR+\HER2-
```{r Limma-voom differential expression results for Nigerian/TCGA black HR+HER2-, eval=FALSE, include=FALSE, cache=TRUE}
designNvsBHR <- design
designNvsBHR$sampleCondition <- ifelse (designNvsBHR$sampleCondition=="TCGA_black.HRpos", 0, as.character(designNvsBHR$sampleCondition))
designNvsBHR$sampleCondition <- ifelse (designNvsBHR$sampleCondition=="Nigerian.HRpos", 1, as.character(designNvsBHR$sampleCondition))

designNvsBHR$sampleCondition <- ifelse (designNvsBHR$sampleCondition==0 | designNvsBHR$sampleCondition==1, designNvsBHR$sampleCondition, NA)

designNvsBHR <- designNvsBHR %>% subset(is.na(sampleCondition)==FALSE)

designNvsBHR$TCGA_black.HRpos <- ifelse (designNvsBHR$sampleCondition==0, 1, 0)
designNvsBHR$Nigerian.HRpos <- ifelse (designNvsBHR$sampleCondition==1, 1, 0)

designNvsBHR$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsBHR$TCGA_black.HRpos+designNvsBHR$Nigerian.HRpos)

quantids <- rownames(designNvsBHR)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_black", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

v=voom(d,designNvsBHR,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_black.HRpos-Nigerian.HRpos, levels=colnames(designNvsBHR))

fit <- lmFit(v, designNvsBHR)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between HR-positive breast cancers in Nigerian and TCGA black patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between HR-positive \nbreast cancers in Nigerian and TCGA black breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

write.csv(df_limmaprint, file = "TCGAblack-Nigerian-HRpos.csv", row.names = FALSE)
```

#DE: Nigerian/TCGA White - HER2 (no TCGA Black HER2+ patients)
```{r Limma-voom differential expression results for Nigerian/TCGA white HER2+, eval=FALSE, include=FALSE, cache=TRUE}
designNvsWHER2 <- design
designNvsWHER2$sampleCondition <- ifelse (designNvsWHER2$sampleCondition=="TCGA_white.Her2", 0, as.character(designNvsWHER2$sampleCondition))
designNvsWHER2$sampleCondition <- ifelse (designNvsWHER2$sampleCondition=="Nigerian.Her2", 1, as.character(designNvsWHER2$sampleCondition))

designNvsWHER2$sampleCondition <- ifelse (designNvsWHER2$sampleCondition==0 | designNvsWHER2$sampleCondition==1, designNvsWHER2$sampleCondition, NA)

designNvsWHER2 <- designNvsWHER2 %>% subset(is.na(sampleCondition)==FALSE)

designNvsWHER2$TCGA_white.Her2 <- ifelse (designNvsWHER2$sampleCondition==0, 1, 0)
designNvsWHER2$Nigerian.Her2 <- ifelse (designNvsWHER2$sampleCondition==1, 1, 0)

designNvsWHER2$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsWHER2$TCGA_white.Her2+designNvsWHER2$Nigerian.Her2)

quantids <- rownames(designNvsWHER2)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

v=voom(d,designNvsWHER2,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.Her2-Nigerian.Her2, levels=colnames(designNvsWHER2))

fit <- lmFit(v, designNvsWHER2)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between Her2+ breast cancers in Nigerian and TCGA white patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between Her2+ \nbreast cancers in Nigerian and TCGA white breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)
write.csv(df_limmaprint, file = "TCGAwhite-Nigerian-Her2.csv", row.names = FALSE)
```
