---
title: "Nigerian-TCGA Differential Expression Analysis Using DeSeq2 - Protein Coding Genes Only for whole-genome samples (PAM50) without ribosomal proteins"
author: "Padma Sheila Rajagopal, MD MPH"
date: "12/4/2019"

site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    df_print: paged
    toc: true
    theme: spacelab
    highlight: textmate
---
```{r Setup, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE)
library(gplots)
library(ggbiplot)
library(ggplot2)
library(edgeR)
library("DESeq2")
library("RUVSeq")
library(vsn)
library("genefilter")
library(pheatmap)
library(clusterProfiler)
library(pathview)
library("AnnotationDbi")
library(checkmate)
library(affy)
library("dplyr")
library(stringi)
library("hexbin")
library("org.Hs.eg.db")
library('GenomicFeatures')
library(EnsDb.Hsapiens.v75)
library("sva")
library(limma)
library(calibrate)
library(ggfortify)
library("ashr")
library(preprocessCore)
library("BiocParallel")
library(RColorBrewer)
library(Glimma)
library(clusterProfiler)
library(pathview)
library(AnnotationHub)

register(MulticoreParam(4))
setwd("~/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants-proteincoding")
```

#Translation from HTSeq raw counts -> Count Matrix
I have 84 TCGA patients with whole-genome sequencing data and RNAseq data as well as 96 Nigerian patients with RNA-seq data. Raw counts were initially processed using HTSeq, so HTSeq data is being formatted for use with DESeq2 and limma-voom. 
```{r Preparing the sampleTable using HTSeq raw counts}
FOLDER <- "/Users/parajago/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants-proteincoding"
sampleFiles <- grep("htseq.counts",list.files(FOLDER),value=TRUE)

#Differential gene expression setup based on race (b/w/other)
sampleConditionrace <- sampleFiles
countVar2=1
for (sample in sampleConditionrace){
  if (stri_detect_fixed(sample,"LIB")==TRUE){
    sampleConditionrace[countVar2] <- "Nigerian"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"black")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_black"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"white")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_white"
    countVar2=countVar2+1
  } else{
    sampleConditionrace[countVar2] <- "TCGA_other"
    countVar2=countVar2+1
  }
}

#Condition based on PAM50 subtype 
sampleConditionPAM50 <- sampleFiles
countVar2=1
for (sample in sampleConditionPAM50){
  if (stri_detect_fixed(sample,"Her2")==TRUE){
    sampleConditionPAM50[countVar2] <- "Her2"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"Basal")==TRUE){
    sampleConditionPAM50[countVar2] <- "Basal"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"LumA")==TRUE){
    sampleConditionPAM50[countVar2] <- "LumA"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"LumB")==TRUE){
    sampleConditionPAM50[countVar2] <- "LumB"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"PAMNL")==TRUE){
    sampleConditionPAM50[countVar2] <- "Normal"
    countVar2=countVar2+1
  } else{
    sampleConditionPAM50[countVar2] <- "PAM_other"
    countVar2=countVar2+1
  }
}

#Condition based on batch (relevant to the Nigerian patients only; no difference in batch for the TCGA patients)
batchval <- sampleFiles
countVar2=1
for (sample in batchval){
  if (stri_detect_fixed(sample,"batch1")==TRUE){
    batchval[countVar2] <- "batch1"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch23")==TRUE){
    batchval[countVar2] <- "batch23"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch4")==TRUE){
    batchval[countVar2] <- "batch4"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch5")==TRUE){
    batchval[countVar2] <- "batch5"
    countVar2=countVar2+1
  } else{
    batchval[countVar2] <- "batchT"
    countVar2=countVar2+1
  }
}

table(sampleConditionrace, sampleConditionPAM50)

sampleTable <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          batch=batchval)

sampleTable$sampleCondition <- paste(sampleTable$condition1, sampleTable$condition2, sep=".")

ddsHTSeqMF <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTable,
                                       directory=FOLDER,
                                       design=~sampleCondition)

ddsHTSeqMF <- ddsHTSeqMF[rowSums(counts(ddsHTSeqMF)) > 0, ] #Pre-filtering the dataset by removing the rows without any information about gene expression -> this removes 603 genes
```

#Quantile normalization
Please refer to: 
https://parajago.github.io/TCGA-Nigerian-RNAseq/NigerianTCGArawcountsDeSeq2-pc2.html
regarding comparison between the Nigerian and TCGA data sets and why quantile normalization under the limma-voom approach was chosen for primary differential expression analysis. 

##Data visualization
```{r Quantile normalization setup and visualization}
countmatrix <- assay(ddsHTSeqMF) #Raw counts organized into matrix format from individual files
countmatrix2 <- log2(countmatrix + 1) #Basic transformation of the count data 

plot(density(countmatrix2[,1]),lwd=3,ylim=c(0,.30), main="Density of counts with log2[count]+1 transformation ONLY") 
for(i in 1:180){lines(density(countmatrix2[,i]),lwd=3)} #This demonstrates that there is a difference in distributions between the Nigerian and TCGA data with basic log transformation normalization 

norm_countmatrix <- as.matrix(countmatrix2) 
norm_countmatrix = normalize.quantiles(norm_countmatrix)
plot(density(norm_countmatrix[,1]),lwd=3,ylim=c(0,.3), main="Density of counts with quantile normalization")
for(i in 1:180){lines(density(norm_countmatrix[,i]),lwd=3)} #This demonstrates the effect of comparative quantile normalization

colnames (norm_countmatrix) <- colnames (countmatrix2)
rownames (norm_countmatrix) <- rownames (countmatrix2)

norm_countmatrix <- as.data.frame(norm_countmatrix)
countmatrixNigerian <- dplyr::select(norm_countmatrix, contains("LIB"))
plot(density(countmatrixNigerian[,1]),lwd=3,ylim=c(0,.3), main="Density of counts with quantile normalization - Nigerian")
for(i in 1:96){lines(density(countmatrixNigerian[,i]),lwd=3)} #This demonstrates the result of the normalized Nigerian counts separately

tcgacolnames <- colnames(countmatrix)
tcgacolnames <- setdiff(tcgacolnames, colnames(countmatrixNigerian))
countmatrixTCGA <- norm_countmatrix[ , tcgacolnames]
plot(density(countmatrixTCGA[,1]),lwd=3,ylim=c(0,.3), main="Density of counts with quantile normalization - TCGA")
for(i in 1:84){
  lines(density(countmatrixTCGA[,i]),lwd=3);
#  print(colnames(countmatrix)[i])
#  invisible(readline(prompt=i))
  } #This demonstrates the result of the normalized TCGA counts separately

norm_countmatrix <- as.data.frame(norm_countmatrix)
t_norm_countmatrix <- t(norm_countmatrix)

t_norm_countmatrix <- cbind (t_norm_countmatrix, sampleTable) #This binds the characteristics of the original patients to the quantile normalized counts. CBinding was checked to make sure that patients were correctly aligned to characteristics. 

quant.pca <- prcomp(t_norm_countmatrix[,1:19724])
autoplot(quant.pca, data=t_norm_countmatrix, colour='sampleCondition', shape='condition1', main="PCA of quantile normalization results prior to DE analysis")
```
In the raw data with log transformation only, we are able to see that there are two peaks corresponding to the two datasets (Nigerian and TCGA). The quantile normalization demonstrates a PCA that has similar clustering. Only ~20% of the distribution of the data set is explained by the PCA1, 2 of the variables.

#Distribution of counts across groups
```{r Distributions of counts across groups}

countmatrixNigerian.test <- countmatrixNigerian
Nigeriancounts <- as.data.frame(rowSums(countmatrixNigerian.test))
names(Nigeriancounts) <- c("sum")
Nigeriancounts$mean <- rowMeans(countmatrixNigerian.test)
Nigeriancounts$tfx <- log(Nigeriancounts$mean)+1
Nigeriancounts$gene <- rownames(Nigeriancounts)

countmatrixTCGA.test <- countmatrixTCGA
TCGAcounts <- as.data.frame(rowSums(countmatrixTCGA.test))
names(TCGAcounts) <- c("sum")
TCGAcounts$mean <- rowMeans(countmatrixTCGA.test)
TCGAcounts$tfx <- log(TCGAcounts$mean)+1
TCGAcounts$gene <- rownames(TCGAcounts)

ggplot(data=TCGAcounts, aes(tfx)) + 
     geom_histogram()

ggplot(data=Nigeriancounts, aes(tfx)) + 
     geom_histogram()

jointcounts <- merge(Nigeriancounts,TCGAcounts, by="gene")
corr <- cor.test(x=jointcounts$mean.x, y=jointcounts$mean.y, method = 'spearman')
```

#Differential expression setup
```{r Differential expression setup, cache=TRUE}
annotation <- as.data.frame(row.names(countmatrix))
colnames(annotation) <- c("GeneID")
annotation$temp <- gsub("[.].+", "", annotation[,1])

annotation$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

annotation$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

annotation$chr <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="SEQNAME",
                     keytype="GENEID",           
                     multiVals="first")

annotation$locstart <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="GENESEQSTART",
                     keytype="GENEID",
                     multiVals="first")

annotation$locend <- mapIds(EnsDb.Hsapiens.v75,
                     keys=annotation$temp,
                     column="GENESEQEND",
                     keytype="GENEID",
                     multiVals="first")
annotation$temp <- NULL

design <- t_norm_countmatrix
design <- design %>% dplyr::select(sampleCondition)
```

##DE: Nigerian/TCGA White - Basal
```{r Limma-voom differential expression results for Nigerian/TCGA white Basal patients, cache=TRUE}
designNvsW <- design
designNvsW$sampleCondition <- ifelse (designNvsW$sampleCondition=="TCGA_white.Basal", 0, as.character(designNvsW$sampleCondition))
designNvsW$sampleCondition <- ifelse (designNvsW$sampleCondition=="Nigerian.Basal", 1, as.character(designNvsW$sampleCondition))

designNvsW$sampleCondition <- ifelse (designNvsW$sampleCondition==0 | designNvsW$sampleCondition==1, designNvsW$sampleCondition, NA)

designNvsW <- designNvsW %>% subset(is.na(sampleCondition)==FALSE)

designNvsW$TCGA_white.Basal <- ifelse (designNvsW$sampleCondition==0, 1, 0)
designNvsW$Nigerian.Basal <- ifelse (designNvsW$sampleCondition==1, 1, 0)

designNvsW$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsW$TCGA_white.Basal+designNvsW$Nigerian.Basal)

quantids <- rownames(designNvsW)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]

v=voom(d,designNvsW,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.Basal-Nigerian.Basal, levels=colnames(designNvsW))

fit <- lmFit(v, designNvsW)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between basal breast cancers in Nigerian and TCGA white patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and TCGA white breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,50)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

fit$genes$status <- ifelse(fit$F.p.value<0.05,"red","black")
limma::plotMA(fit, xlab = "Average log-expression",
       ylab = "Expression log-ratio (this sample vs others)", status=fit$genes$status, 
       main = "MA Plot of differential gene expression between basal \nbreast cancers in Nigerian and TCGA white breast cancer patients")

write.csv(df_limmaprint, file = "TCGA_white-Nigerian-Basal-woribosomes.csv", row.names = FALSE)
```

##DE: Nigerian/TCGA Black - Basal
```{r Limma-voom differential expression results for Nigerian/TCGA black Basal patients, cache=TRUE}
designNvsB <- design
designNvsB$sampleCondition <- ifelse (designNvsB$sampleCondition=="TCGA_black.Basal", 0, as.character(designNvsB$sampleCondition))
designNvsB$sampleCondition <- ifelse (designNvsB$sampleCondition=="Nigerian.Basal", 1, as.character(designNvsB$sampleCondition))

designNvsB$sampleCondition <- ifelse (designNvsB$sampleCondition==0 | designNvsB$sampleCondition==1, designNvsB$sampleCondition, NA)

designNvsB <- designNvsB %>% subset(is.na(sampleCondition)==FALSE)

designNvsB$TCGA_black.Basal <- ifelse (designNvsB$sampleCondition==0, 1, 0)
designNvsB$Nigerian.Basal <- ifelse (designNvsB$sampleCondition==1, 1, 0)

designNvsB$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsB$TCGA_black.Basal+designNvsB$Nigerian.Basal)

quantids <- rownames(designNvsB)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_black", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]

v=voom(d,designNvsB,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_black.Basal-Nigerian.Basal, levels=colnames(designNvsB))

fit <- lmFit(v, designNvsB)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between TNBC breast cancers in Nigerian and TCGA black patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between Basal \nbreast cancers in Nigerian and TCGA black breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,50)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

fit$genes$status <- ifelse(fit$F.p.value<0.05,"red","black")
limma::plotMA(fit, xlab = "Average log-expression",
       ylab = "Expression log-ratio (this sample vs others)", status=fit$genes$status, 
       main = "MA Plot of differential gene expression between basal \nbreast cancers in Nigerian and TCGA black breast cancer patients")

write.csv(df_limmaprint, file = "TCGA_black-Nigerian-Basal-woribosomes.csv", row.names = FALSE)
```

```{r Pathway analysis for Nigerian-TCGA white-Basal, cache=TRUE}
mm <- model.matrix(~0+designNvsW$TCGA_white.Basal+designNvsW$Nigerian.Basal)

quantids <- rownames(designNvsW)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]

v=voom(d,designNvsW,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.Basal-Nigerian.Basal, levels=colnames(designNvsW))

fit <- lmFit(v, designNvsW)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = p.adjust(fit$p.value[,1], method='fdr'),
                       anno = fit$genes)

pathway.Nigerian.TCGAwhite.Basal <- as.data.frame(df_limma)

pathway.Nigerian.TCGAwhite.Basal$foldChange <- NA
row.pos <- which(! is.na(pathway.Nigerian.TCGAwhite.Basal$log2FoldChange) & 
                pathway.Nigerian.TCGAwhite.Basal$log2FoldChange >= 0)
row.neg <- which(! is.na(pathway.Nigerian.TCGAwhite.Basal$log2FoldChange) & 
                pathway.Nigerian.TCGAwhite.Basal$log2FoldChange < 0)
pathway.Nigerian.TCGAwhite.Basal$foldChange[row.pos] <- 2^pathway.Nigerian.TCGAwhite.Basal$log2FoldChange[row.pos]
pathway.Nigerian.TCGAwhite.Basal$foldChange[row.neg] <- -2^((-1) * pathway.Nigerian.TCGAwhite.Basal$log2FoldChange[row.neg])

pathway.Nigerian.TCGAwhite.Basal$log2FoldChange <- NULL
pathway.Nigerian.TCGAwhite.Basal$ENSEMBL <- pathway.Nigerian.TCGAwhite.Basal$anno$GeneID
pathway.Nigerian.TCGAwhite.Basal$SYMBOL <- pathway.Nigerian.TCGAwhite.Basal$anno$symbol

pathway.Nigerian.TCGAwhite.Basal$anno$GeneID <- NULL
pathway.Nigerian.TCGAwhite.Basal$anno$symbol <- NULL

pathway.Nigerian.TCGAwhite.Basal.flt <- pathway.Nigerian.TCGAwhite.Basal %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

genes.all <- pathway.Nigerian.TCGAwhite.Basal
genes.sig <- pathway.Nigerian.TCGAwhite.Basal.flt

genes.all$ENSEMBL <- gsub('[.]\\d+', '', genes.all$ENSEMBL, perl = TRUE)
genes.sig$ENSEMBL <- gsub('[.]\\d+', '', genes.sig$ENSEMBL, perl = TRUE)

genes.all.anno <- bitr(geneID   =  genes.all$ENSEMBL, 
                      fromType = 'GENEID', 
                      toType   = c('ENTREZID', 'SYMBOL'), 
                      OrgDb    = 'EnsDb.Hsapiens.v75', 
                      drop     = TRUE)

genes.all.anno <- genes.all.anno[which(!duplicated(genes.all.anno$ENTREZID)), ]
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.all.anno$ENSEMBL <- genes.all.anno$GENEID
genes.all.anno$GENEID <- NULL

genes.all.anno <- merge(genes.all.anno, genes.all, by = 'ENSEMBL')
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.sig.anno <- genes.all.anno[genes.all.anno$ENSEMBL %in% 
                                genes.sig$ENSEMBL,]

gene.list <- genes.all.anno$foldChange
names(gene.list) <- genes.all.anno$ENTREZID
gene.list <- sort(gene.list, decreasing = TRUE)

ego <- enrichGO(gene          = genes.sig.anno$ENTREZID,
                universe      = as.character(genes.all.anno$ENTREZID),
                OrgDb         = 'org.Hs.eg.db',
                ont           = "BP",
                pAdjustMethod = "fdr",
                pvalueCutoff  = 0.05,
                readable      = TRUE)

as.data.frame(ego)
save(ego, file="GO-Nigerian-TCGAwhite-Basal.significantgenes.fdr0.05.fc1.5.enrichGO.woribosomes.RData")
write.csv(ego, file="GO-Nigerian-TCGAwhite-Basal.significantgenes.fdr0.05.fc1.5.enrichGO.woribosomes.csv")

options(jupyter.plot_mimetypes = "image/svg+xml") 
options(repr.plot.width = 10, repr.plot.height = 5)

egokegg <- ego
for(i in 1:5) { 
  egokegg <- dropGO(egokegg, level = i)
}

p1 <- barplot(egokegg)
p2 <- dotplot(egokegg)

plot(p1)
plot(p2)
```

##DE: Nigerian/TCGA White - HER2 (no TCGA Black HER2+ patients)
```{r Limma-voom differential expression results for Nigerian/TCGA white HER2+, cache=TRUE}
designNvsWHER2 <- design
designNvsWHER2$sampleCondition <- ifelse (designNvsWHER2$sampleCondition=="TCGA_white.Her2", 0, as.character(designNvsWHER2$sampleCondition))
designNvsWHER2$sampleCondition <- ifelse (designNvsWHER2$sampleCondition=="Nigerian.Her2", 1, as.character(designNvsWHER2$sampleCondition))

designNvsWHER2$sampleCondition <- ifelse (designNvsWHER2$sampleCondition==0 | designNvsWHER2$sampleCondition==1, designNvsWHER2$sampleCondition, NA)

designNvsWHER2 <- designNvsWHER2 %>% subset(is.na(sampleCondition)==FALSE)

designNvsWHER2$TCGA_white.Her2 <- ifelse (designNvsWHER2$sampleCondition==0, 1, 0)
designNvsWHER2$Nigerian.Her2 <- ifelse (designNvsWHER2$sampleCondition==1, 1, 0)

designNvsWHER2$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsWHER2$TCGA_white.Her2+designNvsWHER2$Nigerian.Her2)

quantids <- rownames(designNvsWHER2)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]

v=voom(d,designNvsWHER2,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.Her2-Nigerian.Her2, levels=colnames(designNvsWHER2))

fit <- lmFit(v, designNvsWHER2)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between Her2+ breast cancers in Nigerian and TCGA white patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between Her2+ \nbreast cancers in Nigerian and TCGA white breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,20)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

fit$genes$status <- ifelse(fit$F.p.value<0.05,"red","black")
limma::plotMA(fit, xlab = "Average log-expression",
       ylab = "Expression log-ratio (this sample vs others)", status=fit$genes$status, 
       main = "MA Plot of differential gene expression between Her2 \nbreast cancers in Nigerian and TCGA white breast cancer patients")

write.csv(df_limmaprint, file = "TCGAwhite-Nigerian-Her2.woribosomes.csv", row.names = FALSE)
```

```{r Pathway analysis for Nigerian-TCGA white-Her2, cache=TRUE}
mm <- model.matrix(~0+designNvsWHER2$TCGA_white.Her2+designNvsWHER2$Nigerian.Her2)

quantids <- rownames(designNvsWHER2)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]

v=voom(d,designNvsWHER2,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.Her2-Nigerian.Her2, levels=colnames(designNvsWHER2))

fit <- lmFit(v, designNvsWHER2)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

pathway.Nigerian.TCGAwhite.Her2 <- as.data.frame(df_limma)

pathway.Nigerian.TCGAwhite.Her2$foldChange <- NA
row.pos <- which(! is.na(pathway.Nigerian.TCGAwhite.Her2$log2FoldChange) & 
                pathway.Nigerian.TCGAwhite.Her2$log2FoldChange >= 0)
row.neg <- which(! is.na(pathway.Nigerian.TCGAwhite.Her2$log2FoldChange) & 
                pathway.Nigerian.TCGAwhite.Her2$log2FoldChange < 0)
pathway.Nigerian.TCGAwhite.Her2$foldChange[row.pos] <- 2^pathway.Nigerian.TCGAwhite.Her2$log2FoldChange[row.pos]
pathway.Nigerian.TCGAwhite.Her2$foldChange[row.neg] <- -2^((-1) * pathway.Nigerian.TCGAwhite.Her2$log2FoldChange[row.neg])

pathway.Nigerian.TCGAwhite.Her2$log2FoldChange <- NULL
pathway.Nigerian.TCGAwhite.Her2$ENSEMBL <- pathway.Nigerian.TCGAwhite.Her2$anno$GeneID
pathway.Nigerian.TCGAwhite.Her2$SYMBOL <- pathway.Nigerian.TCGAwhite.Her2$anno$symbol

pathway.Nigerian.TCGAwhite.Her2$anno$GeneID <- NULL
pathway.Nigerian.TCGAwhite.Her2$anno$symbol <- NULL

pathway.Nigerian.TCGAwhite.Her2.flt <- pathway.Nigerian.TCGAwhite.Her2 %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

genes.all <- pathway.Nigerian.TCGAwhite.Her2
genes.sig <- pathway.Nigerian.TCGAwhite.Her2.flt

genes.all$ENSEMBL <- gsub('[.]\\d+', '', genes.all$ENSEMBL, perl = TRUE)
genes.sig$ENSEMBL <- gsub('[.]\\d+', '', genes.sig$ENSEMBL, perl = TRUE)

genes.all.anno <- bitr(geneID   =  genes.all$ENSEMBL, 
                      fromType = 'GENEID', 
                      toType   = c('ENTREZID', 'SYMBOL'), 
                      OrgDb    = 'EnsDb.Hsapiens.v75', 
                      drop     = TRUE)

genes.all.anno <- genes.all.anno[which(!duplicated(genes.all.anno$ENTREZID)), ]
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.all.anno$ENSEMBL <- genes.all.anno$GENEID
genes.all.anno$GENEID <- NULL

genes.all.anno <- merge(genes.all.anno, genes.all, by = 'ENSEMBL')
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.sig.anno <- genes.all.anno[genes.all.anno$ENSEMBL %in% 
                                genes.sig$ENSEMBL,]

gene.list <- genes.all.anno$foldChange
names(gene.list) <- genes.all.anno$ENTREZID
gene.list <- sort(gene.list, decreasing = TRUE)

ego <- enrichGO(gene          = genes.sig.anno$ENTREZID,
                universe      = as.character(genes.all.anno$ENTREZID),
                OrgDb         = 'org.Hs.eg.db',
                ont           = "BP",
                pAdjustMethod = "fdr",
                pvalueCutoff  = 0.05,
                readable      = TRUE)

as.data.frame(ego)
save(ego, file="GO-Nigerian-TCGAwhite-Her2.significantgenes.fdr0.05.fc1.5.enrichGO.woribosomes.RData")
write.csv(ego, file="GO-Nigerian-TCGAwhite-Her2.significantgenes.fdr0.05.fc1.5.enrichGO.woribosomes.csv")

options(jupyter.plot_mimetypes = "image/svg+xml") 
options(repr.plot.width = 10, repr.plot.height = 5)

egokegg <- ego
for(i in 1:5) { 
  egokegg <- dropGO(egokegg, level = i)
}

p1 <- barplot(egokegg)
p2 <- dotplot(egokegg)

plot(p1)
plot(p2)
```

##DE: Nigerian/TCGA White - LumA
```{r Limma-voom differential expression results for Nigerian/TCGA white LumA, cache=TRUE}
designNvsWHR <- design
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="TCGA_white.LumA", 0, as.character(designNvsWHR$sampleCondition))
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="Nigerian.LumA", 1, as.character(designNvsWHR$sampleCondition))

designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition==0 | designNvsWHR$sampleCondition==1, designNvsWHR$sampleCondition, NA)

designNvsWHR <- designNvsWHR %>% subset(is.na(sampleCondition)==FALSE)

designNvsWHR$TCGA_white.LumA <- ifelse (designNvsWHR$sampleCondition==0, 1, 0)
designNvsWHR$Nigerian.LumA <- ifelse (designNvsWHR$sampleCondition==1, 1, 0)

designNvsWHR$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsWHR$TCGA_white.LumA+designNvsWHR$Nigerian.LumA)

quantids <- rownames(designNvsWHR)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]

v=voom(d,designNvsWHR,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.LumA-Nigerian.LumA, levels=colnames(designNvsWHR))

fit <- lmFit(v, designNvsWHR)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between LumA breast cancers in Nigerian and TCGA white patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between LumA \nbreast cancers in Nigerian and TCGA white breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,20)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

fit$genes$status <- ifelse(fit$F.p.value<0.05,"red","black")
limma::plotMA(fit, xlab = "Average log-expression",
       ylab = "Expression log-ratio (this sample vs others)", status=fit$genes$status, 
       main = "MA Plot of differential gene expression between LumA \nbreast cancers in Nigerian and TCGA white breast cancer patients")

write.csv(df_limmaprint, file = "TCGA_white-Nigerian-LumA-woribosomes.csv", row.names = FALSE)
```

##DE: Nigerian/TCGA Black - LumA
```{r Limma-voom differential expression results for Nigerian/TCGA black LumA, cache=TRUE}
designNvsBHR <- design
designNvsBHR$sampleCondition <- ifelse (designNvsBHR$sampleCondition=="TCGA_black.LumA", 0, as.character(designNvsBHR$sampleCondition))
designNvsBHR$sampleCondition <- ifelse (designNvsBHR$sampleCondition=="Nigerian.LumA", 1, as.character(designNvsBHR$sampleCondition))

designNvsBHR$sampleCondition <- ifelse (designNvsBHR$sampleCondition==0 | designNvsBHR$sampleCondition==1, designNvsBHR$sampleCondition, NA)

designNvsBHR <- designNvsBHR %>% subset(is.na(sampleCondition)==FALSE)

designNvsBHR$TCGA_black.LumA <- ifelse (designNvsBHR$sampleCondition==0, 1, 0)
designNvsBHR$Nigerian.LumA <- ifelse (designNvsBHR$sampleCondition==1, 1, 0)

designNvsBHR$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsBHR$TCGA_black.LumA+designNvsBHR$Nigerian.LumA)

quantids <- rownames(designNvsBHR)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_black", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]

v=voom(d,designNvsBHR,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_black.LumA-Nigerian.LumA, levels=colnames(designNvsBHR))

fit <- lmFit(v, designNvsBHR)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between LumA breast cancers in Nigerian and TCGA black patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between LumA \nbreast cancers in Nigerian and TCGA black breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,20)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

fit$genes$status <- ifelse(fit$F.p.value<0.05,"red","black")
limma::plotMA(fit, xlab = "Average log-expression",
       ylab = "Expression log-ratio (this sample vs others)", status=fit$genes$status, 
       main = "MA Plot of differential gene expression between LumA \nbreast cancers in Nigerian and TCGA white breast cancer patients")

write.csv(df_limmaprint, file = "TCGA_black-Nigerian-LumA-woribosomes.csv", row.names = FALSE)
```

```{r Pathway analysis for Nigerian-TCGA white-LumA, cache=TRUE}
designNvsWHR <- design
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="TCGA_white.LumA", 0, as.character(designNvsWHR$sampleCondition))
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="Nigerian.LumA", 1, as.character(designNvsWHR$sampleCondition))

designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition==0 | designNvsWHR$sampleCondition==1, designNvsWHR$sampleCondition, NA)

designNvsWHR <- designNvsWHR %>% subset(is.na(sampleCondition)==FALSE)

designNvsWHR$TCGA_white.LumA <- ifelse (designNvsWHR$sampleCondition==0, 1, 0)
designNvsWHR$Nigerian.LumA <- ifelse (designNvsWHR$sampleCondition==1, 1, 0)

designNvsWHR$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsWHR$TCGA_white.LumA+designNvsWHR$Nigerian.LumA)

quantids <- rownames(designNvsWHR)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]


v=voom(d,designNvsWHR,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.LumA-Nigerian.LumA, levels=colnames(designNvsWHR))

fit <- lmFit(v, designNvsWHR)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = p.adjust(fit$p.value[,1], method='fdr'),
                       anno = fit$genes)

pathway.Nigerian.TCGAwhite.LumA <- as.data.frame(df_limma)

pathway.Nigerian.TCGAwhite.LumA$foldChange <- NA
row.pos <- which(! is.na(pathway.Nigerian.TCGAwhite.LumA$log2FoldChange) & 
                pathway.Nigerian.TCGAwhite.LumA$log2FoldChange >= 0)
row.neg <- which(! is.na(pathway.Nigerian.TCGAwhite.LumA$log2FoldChange) & 
                pathway.Nigerian.TCGAwhite.LumA$log2FoldChange < 0)
pathway.Nigerian.TCGAwhite.LumA$foldChange[row.pos] <- 2^pathway.Nigerian.TCGAwhite.LumA$log2FoldChange[row.pos]
pathway.Nigerian.TCGAwhite.LumA$foldChange[row.neg] <- -2^((-1) * pathway.Nigerian.TCGAwhite.LumA$log2FoldChange[row.neg])

pathway.Nigerian.TCGAwhite.LumA$log2FoldChange <- NULL
pathway.Nigerian.TCGAwhite.LumA$ENSEMBL <- pathway.Nigerian.TCGAwhite.LumA$anno$GeneID
pathway.Nigerian.TCGAwhite.LumA$SYMBOL <- pathway.Nigerian.TCGAwhite.LumA$anno$symbol

pathway.Nigerian.TCGAwhite.LumA$anno$GeneID <- NULL
pathway.Nigerian.TCGAwhite.LumA$anno$symbol <- NULL

pathway.Nigerian.TCGAwhite.LumA.flt <- pathway.Nigerian.TCGAwhite.LumA %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

genes.all <- pathway.Nigerian.TCGAwhite.LumA
genes.sig <- pathway.Nigerian.TCGAwhite.LumA.flt 

genes.all$ENSEMBL <- gsub('[.]\\d+', '', genes.all$ENSEMBL, perl = TRUE)
genes.sig$ENSEMBL <- gsub('[.]\\d+', '', genes.sig$ENSEMBL, perl = TRUE)

genes.all.anno <- bitr(geneID   =  genes.all$ENSEMBL, 
                      fromType = 'GENEID', 
                      toType   = c('ENTREZID', 'SYMBOL'), 
                      OrgDb    = 'EnsDb.Hsapiens.v75', 
                      drop     = TRUE)

genes.all.anno <- genes.all.anno[which(!duplicated(genes.all.anno$ENTREZID)), ]
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.all.anno$ENSEMBL <- genes.all.anno$GENEID
genes.all.anno$GENEID <- NULL

genes.all.anno <- merge(genes.all.anno, genes.all, by = 'ENSEMBL')
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.sig.anno <- genes.all.anno[genes.all.anno$ENSEMBL %in% 
                                genes.sig$ENSEMBL,]

gene.list <- genes.all.anno$foldChange
names(gene.list) <- genes.all.anno$ENTREZID
gene.list <- sort(gene.list, decreasing = TRUE)

ego <- enrichGO(gene          = genes.sig.anno$ENTREZID,
                universe      = as.character(genes.all.anno$ENTREZID),
                OrgDb         = 'org.Hs.eg.db',
                ont           = "BP",
                pAdjustMethod = "fdr",
                pvalueCutoff  = 0.05,
                readable      = TRUE)

as.data.frame(ego)
save(ego, file="GO-Nigerian-TCGAwhite-LumA.significantgenes.fdr0.05.fc1.5.enrichGO.woribosomes.RData")
write.csv(ego, file="GO-Nigerian-TCGAwhite-LumA.significantgenes.fdr0.05.fc1.5.enrichGO.woribosomes.csv")

options(jupyter.plot_mimetypes = "image/svg+xml") 
options(repr.plot.width = 10, repr.plot.height = 5)

egokegg <- ego
for(i in 1:5) { 
  egokegg <- dropGO(egokegg, level = i)
}

p1 <- barplot(egokegg)
p2 <- dotplot(egokegg)

plot(p1)
plot(p2)
```

##DE: Nigerian/TCGA White - LumB
```{r Limma-voom differential expression results for Nigerian/TCGA white LumB, cache=TRUE}
designNvsWHR <- design
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="TCGA_white.LumB", 0, as.character(designNvsWHR$sampleCondition))
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="Nigerian.LumB", 1, as.character(designNvsWHR$sampleCondition))

designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition==0 | designNvsWHR$sampleCondition==1, designNvsWHR$sampleCondition, NA)

designNvsWHR <- designNvsWHR %>% subset(is.na(sampleCondition)==FALSE)

designNvsWHR$TCGA_white.LumB <- ifelse (designNvsWHR$sampleCondition==0, 1, 0)
designNvsWHR$Nigerian.LumB <- ifelse (designNvsWHR$sampleCondition==1, 1, 0)

designNvsWHR$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsWHR$TCGA_white.LumB+designNvsWHR$Nigerian.LumB)

quantids <- rownames(designNvsWHR)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]

v=voom(d,designNvsWHR,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.LumB-Nigerian.LumB, levels=colnames(designNvsWHR))

fit <- lmFit(v, designNvsWHR)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between LumB breast cancers in Nigerian and TCGA white patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between LumB \nbreast cancers in Nigerian and TCGA white breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,20)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

fit$genes$status <- ifelse(fit$F.p.value<0.05,"red","black")
limma::plotMA(fit, xlab = "Average log-expression",
       ylab = "Expression log-ratio (this sample vs others)", status=fit$genes$status, 
       main = "MA Plot of differential gene expression between LumB \nbreast cancers in Nigerian and TCGA white breast cancer patients")

write.csv(df_limmaprint, file = "TCGA_white-Nigerian-LumB-woribosomes.csv", row.names = FALSE)
```

##DE: Nigerian/TCGA Black - LumB
```{r Limma-voom differential expression results for Nigerian/TCGA black LumB, cache=TRUE}
designNvsBHR <- design
designNvsBHR$sampleCondition <- ifelse (designNvsBHR$sampleCondition=="TCGA_black.LumB", 0, as.character(designNvsBHR$sampleCondition))
designNvsBHR$sampleCondition <- ifelse (designNvsBHR$sampleCondition=="Nigerian.LumB", 1, as.character(designNvsBHR$sampleCondition))

designNvsBHR$sampleCondition <- ifelse (designNvsBHR$sampleCondition==0 | designNvsBHR$sampleCondition==1, designNvsBHR$sampleCondition, NA)

designNvsBHR <- designNvsBHR %>% subset(is.na(sampleCondition)==FALSE)

designNvsBHR$TCGA_black.LumB <- ifelse (designNvsBHR$sampleCondition==0, 1, 0)
designNvsBHR$Nigerian.LumB <- ifelse (designNvsBHR$sampleCondition==1, 1, 0)

designNvsBHR$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsBHR$TCGA_black.LumB+designNvsBHR$Nigerian.LumB)

quantids <- rownames(designNvsBHR)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_black", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]

v=voom(d,designNvsBHR,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_black.LumB-Nigerian.LumB, levels=colnames(designNvsBHR))

fit <- lmFit(v, designNvsBHR)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential gene expression\n between LumB breast cancers in Nigerian and TCGA black patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       anno = fit$genes)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between LumB \nbreast cancers in Nigerian and TCGA black breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,20)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=anno$symbol, cex=.5))

df_limmaprint <- as.data.frame(df_limma)

df_limmaprint$foldChange <- NA
row.pos <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange >= 0)
row.neg <- which(! is.na(df_limmaprint$log2FoldChange) & 
                df_limmaprint$log2FoldChange < 0)
df_limmaprint$foldChange[row.pos] <- 2^df_limmaprint$log2FoldChange[row.pos]
df_limmaprint$foldChange[row.neg] <- -2^((-1) * df_limmaprint$log2FoldChange[row.neg])

df_limmaprint <- df_limmaprint %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

top_n(df_limmaprint, 10, foldChange)
top_n(df_limmaprint, -10, foldChange)

fit$genes$status <- ifelse(fit$F.p.value<0.05,"red","black")
limma::plotMA(fit, xlab = "Average log-expression",
       ylab = "Expression log-ratio (this sample vs others)", status=fit$genes$status, 
       main = "MA Plot of differential gene expression between LumB \nbreast cancers in Nigerian and TCGA black breast cancer patients")

write.csv(df_limmaprint, file = "TCGA_black-Nigerian-LumB-woribosomes.csv", row.names = FALSE)
```


```{r Pathway analysis for Nigerian-TCGA white-LumB, cache=TRUE}
designNvsWHR <- design
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="TCGA_white.LumB", 0, as.character(designNvsWHR$sampleCondition))
designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition=="Nigerian.LumB", 1, as.character(designNvsWHR$sampleCondition))

designNvsWHR$sampleCondition <- ifelse (designNvsWHR$sampleCondition==0 | designNvsWHR$sampleCondition==1, designNvsWHR$sampleCondition, NA)

designNvsWHR <- designNvsWHR %>% subset(is.na(sampleCondition)==FALSE)

designNvsWHR$TCGA_white.LumB <- ifelse (designNvsWHR$sampleCondition==0, 1, 0)
designNvsWHR$Nigerian.LumB <- ifelse (designNvsWHR$sampleCondition==1, 1, 0)

designNvsWHR$sampleCondition <- NULL

mm <- model.matrix(~0+designNvsWHR$TCGA_white.LumB+designNvsWHR$Nigerian.LumB)

quantids <- rownames(designNvsWHR)
rownames(mm) <- quantids
colnames(mm) <- c("TCGA_white", "Nigerian")

quantdata <- as.data.frame(t(counts(ddsHTSeqMF)))
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)

d0 <- DGEList(counts=quantdata, genes=annotation)

cutoff <- 10
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # Number of genes after taking out low expressed genes

drop2 <- which(d$genes$symbol=="RPS25")
d <- d[-drop2,]
drop3 <- which(d$genes$symbol=="RPL18A")
d <- d[-drop3,]
drop4 <- which(d$genes$symbol=="RPL7")
d <- d[-drop4,]
drop5 <- which(d$genes$symbol=="RPL21")
d <- d[-drop5,]
drop6 <- which(d$genes$symbol=="RPS3A")
d <- d[-drop6,]
drop7 <- which(d$genes$symbol=="RPL39")
d <- d[-drop7,]
drop8 <- which(d$genes$symbol=="RPL7A")
d <- d[-drop8,]
drop9 <- which(d$genes$symbol=="RPL10")
d <- d[-drop9,]


v=voom(d,designNvsWHR,plot=T, normalize="quantile")

contr.matrix <- makeContrasts(TCGA_white.LumB-Nigerian.LumB, levels=colnames(designNvsWHR))

fit <- lmFit(v, designNvsWHR)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = p.adjust(fit$p.value[,1], method='fdr'),
                       anno = fit$genes)

pathway.Nigerian.TCGAwhite.LumB <- as.data.frame(df_limma)

pathway.Nigerian.TCGAwhite.LumB$foldChange <- NA
row.pos <- which(! is.na(pathway.Nigerian.TCGAwhite.LumB$log2FoldChange) & 
                pathway.Nigerian.TCGAwhite.LumB$log2FoldChange >= 0)
row.neg <- which(! is.na(pathway.Nigerian.TCGAwhite.LumB$log2FoldChange) & 
                pathway.Nigerian.TCGAwhite.LumB$log2FoldChange < 0)
pathway.Nigerian.TCGAwhite.LumB$foldChange[row.pos] <- 2^pathway.Nigerian.TCGAwhite.LumB$log2FoldChange[row.pos]
pathway.Nigerian.TCGAwhite.LumB$foldChange[row.neg] <- -2^((-1) * pathway.Nigerian.TCGAwhite.LumB$log2FoldChange[row.neg])

pathway.Nigerian.TCGAwhite.LumB$log2FoldChange <- NULL
pathway.Nigerian.TCGAwhite.LumB$ENSEMBL <- pathway.Nigerian.TCGAwhite.LumB$anno$GeneID
pathway.Nigerian.TCGAwhite.LumB$SYMBOL <- pathway.Nigerian.TCGAwhite.LumB$anno$symbol

pathway.Nigerian.TCGAwhite.LumB$anno$GeneID <- NULL
pathway.Nigerian.TCGAwhite.LumB$anno$symbol <- NULL

pathway.Nigerian.TCGAwhite.LumB.flt <- pathway.Nigerian.TCGAwhite.LumB %>% arrange(foldChange) %>% dplyr::filter(padj < 0.05) %>% dplyr::filter(abs(foldChange)>1.5)

genes.all <- pathway.Nigerian.TCGAwhite.LumB
genes.sig <- pathway.Nigerian.TCGAwhite.LumB.flt 

genes.all$ENSEMBL <- gsub('[.]\\d+', '', genes.all$ENSEMBL, perl = TRUE)
genes.sig$ENSEMBL <- gsub('[.]\\d+', '', genes.sig$ENSEMBL, perl = TRUE)

genes.all.anno <- bitr(geneID   =  genes.all$ENSEMBL, 
                      fromType = 'GENEID', 
                      toType   = c('ENTREZID', 'SYMBOL'), 
                      OrgDb    = 'EnsDb.Hsapiens.v75', 
                      drop     = TRUE)

genes.all.anno <- genes.all.anno[which(!duplicated(genes.all.anno$ENTREZID)), ]
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.all.anno$ENSEMBL <- genes.all.anno$GENEID
genes.all.anno$GENEID <- NULL

genes.all.anno <- merge(genes.all.anno, genes.all, by = 'ENSEMBL')
row.names(genes.all.anno) <- genes.all.anno$ENTREZID

genes.sig.anno <- genes.all.anno[genes.all.anno$ENSEMBL %in% 
                                genes.sig$ENSEMBL,]

gene.list <- genes.all.anno$foldChange
names(gene.list) <- genes.all.anno$ENTREZID
gene.list <- sort(gene.list, decreasing = TRUE)

ego <- enrichGO(gene          = genes.sig.anno$ENTREZID,
                universe      = as.character(genes.all.anno$ENTREZID),
                OrgDb         = 'org.Hs.eg.db',
                ont           = "BP",
                pAdjustMethod = "fdr",
                pvalueCutoff  = 0.05,
                readable      = TRUE)

as.data.frame(ego)
save(ego, file="GO-Nigerian-TCGAwhite-LumB.significantgenes.fdr0.05.fc1.5.enrichGO.woribosomes.RData")
write.csv(ego, file="GO-Nigerian-TCGAwhite-LumB.significantgenes.fdr0.05.fc1.5.enrichGO.woribosomes.csv")

options(jupyter.plot_mimetypes = "image/svg+xml") 
options(repr.plot.width = 10, repr.plot.height = 5)

egokegg <- ego
for(i in 1:5) { 
  egokegg <- dropGO(egokegg, level = i)
}

p1 <- barplot(egokegg)
p2 <- dotplot(egokegg)

plot(p1)
plot(p2)
```
