---
title: "Nigerian-TCGA Differential Expression Analysis Using DeSeq2"
author: "Padma Sheila Rajagopal, MD MPH"
date: "1/23/2019"

site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    df_print: paged
    toc: true
    theme: spacelab
    highlight: textmate
---
```{r Setup, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE)
library("DESeq2")
library(stringi)
library("dplyr")
library("ggplot2")
library("hexbin")
library("AnnotationDbi")
library("org.Hs.eg.db")
library('GenomicFeatures')
library(biomaRt)
library(EnsDb.Hsapiens.v75)
library(apeglm)
library("genefilter")
library(pheatmap)
library("sva")
library(limma)
setwd("~/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants")
```

#Translation from HTSeq -> DESeq2 table
I have 86 TCGA patients with whole-genome sequencing data and RNAseq data as well as 99 Nigerian patients with RNA-seq data. Raw counts were initially processed using HTSeq, so HTSeq data is being formatted for use with DESeq2.
```{r Preparing the sampleTable using HTSeq raw counts}
FOLDER <- "/Users/parajago/Research-Local/RNAseq-Local/Inputs/NigerianTCGA_quants"
sampleFiles <- grep("htseq.counts",list.files(FOLDER),value=TRUE)

#Differential gene expression setup based on race (b/w/other)
sampleConditionrace <- sampleFiles
countVar2=1
for (sample in sampleConditionrace){
  if (stri_detect_fixed(sample,"LIB")==TRUE){
    sampleConditionrace[countVar2] <- "Nigerian"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"black")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_black"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"white")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_white"
    countVar2=countVar2+1
  } else{
    sampleConditionrace[countVar2] <- "TCGA_other"
    countVar2=countVar2+1
  }
}

#Condition based on PAM50 subtype 
sampleConditionPAM50 <- sampleFiles
countVar2=1
for (sample in sampleConditionPAM50){
  if (stri_detect_fixed(sample,"Her2")==TRUE){
    sampleConditionPAM50[countVar2] <- "Her2"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"Basal")==TRUE){
    sampleConditionPAM50[countVar2] <- "Basal"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"LumA")==TRUE){
    sampleConditionPAM50[countVar2] <- "LumA"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"LumB")==TRUE){
    sampleConditionPAM50[countVar2] <- "LumB"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"PAMNL")==TRUE){
    sampleConditionPAM50[countVar2] <- "Normal"
    countVar2=countVar2+1
  } else{
    sampleConditionPAM50[countVar2] <- "PAM_other"
    countVar2=countVar2+1
  }
}

#Condition based on PAM50 subtype 
batchval <- sampleFiles
countVar2=1
for (sample in batchval){
  if (stri_detect_fixed(sample,"batch1")==TRUE){
    batchval[countVar2] <- "batch1"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch23")==TRUE){
    batchval[countVar2] <- "batch23"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch4")==TRUE){
    batchval[countVar2] <- "batch4"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch5")==TRUE){
    batchval[countVar2] <- "batch5"
    countVar2=countVar2+1
  } else{
    batchval[countVar2] <- "batchT"
    countVar2=countVar2+1
  }
}

table(sampleConditionrace, sampleConditionPAM50)

sampleTable2 <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          batch=batchval)

sampleTable2$sampleCondition <- paste(sampleTable2$condition1, sampleTable2$condition2, sep=".")

ddsHTSeqMF <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTable2,
                                       directory=FOLDER,
                                       design=~sampleCondition)
```


#Variance transformation, batch effect assessment and initial exploratory analysis
I am now performing different visualization approaches to assess the variance of the data, batch effect and expected differences between the two populations.
```{r Variance transformation and batch effect assessment}
ddsHTSeqMF <- ddsHTSeqMF[rowSums(counts(ddsHTSeqMF)) > 1, ] #Pre-filtering the dataset by removing the rows without information about gene expression

dds <- estimateSizeFactors(ddsHTSeqMF) #The size factor is the median ratio of the sample over a "pseudosample": for each gene, the geometric mean of all samples.
dds$condition1 <- relevel(dds$condition1, ref = "Nigerian") #explicitng making the Nigerian patients the reference population 

vsd <- vst(ddsHTSeqMF, blind = TRUE) #Variance-stabilizing transformation

df <- bind_rows( #Graphical representation of the variance-stabilizing transformation
          as_data_frame(log2(counts(dds, normalized=TRUE)[, 1:2]+1)) %>%
          mutate(transformation = "log2(x + 1)"),
          as_data_frame(assay(vsd)[, 1:2]) %>% mutate(transformation = "vst"))
          colnames(df)[1:2] <- c("x", "y")

ggplot(df, aes(x = x, y = y)) + labs(subtitle="Figure 1: Graphical representation of the \nvariance-stabilizing transformation \nrelative to the standard transformation of log2(x+1).\nThis reduces variance purely due to low read counts") + geom_hex(bins = 80) +
  coord_fixed() + facet_grid( . ~ transformation) 

#PCA1: Raw counts
plotPCA(vsd, intgroup=c("condition1", "batch")) + labs(subtitle = "PCA 1-1:PCA demonstrating batch effect between\nthe Nigerian patients and TCGA patients,\nwhich would be expected given that these experiments\nwere done in different settings at different times.\nThe TCGA patients also appear to have internal\nbatch effect that is not completely race-consistent.\nNo clear effect from internal batches of Nigerian samples") 

plotPCA(vsd, intgroup=c("condition2")) + labs(subtitle = "PCA 1-3:PCA demonstrating association\nwith breast cancer subtype and gene expression.\nThis is noted in both groups and not likely due to batch effect.") 

plotPCA(vsd, intgroup=c("condition1", "condition2")) + labs(subtitle = "PCA 1-3: Integrated PCA demonstrating combined effect of\n breast cancer subtype and race.\n There is likely some contributor of batch effect to this presentation.") 

#PCA2: Removing the batch effect created by the separate batch processing of the TCGA and Nigerian samples (although this may overcorrect for the actual difference between these two populations).  
mat <- assay(vsd) 
mat <- limma::removeBatchEffect(mat, vsd$condition1, vsd$batch)
assay(vsd) <- mat
plotPCA(vsd, intgroup=c("condition1", "condition2")) + labs(subtitle="PCA2-1: We have accounted for batch effect using the limma removeBatchEffect method.\nThis largely accounts for the batch effect noticed in PCA1-Nigerian vs TCGA. \nThere remain some outliers, primarily from the Nigerian patients.")

topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 500)
hm  <- assay(vsd)[ topVarGenes, ]
hm  <- hm - rowMeans(hm)
anno <- as.data.frame(colData(vsd)[, c("condition1", "condition2")])
rownames(anno) <- colnames(vsd)
pheatmap(hm, annotation_col = anno, cluster_rows=FALSE, show_rownames=FALSE, show_colnames = FALSE,
         cluster_cols=FALSE, main="Heat map: \nRelative VST-transformed counts across samples")
```

#Using sva to additionally account for batch effect related to the Nigerian/TCGA samples
```{r Batch effect assessment, message=FALSE}
dat  <- counts(dds, normalized = TRUE)
idx  <- rowMeans(dat) > 1
dat  <- dat[idx, ]
mod  <- model.matrix(~condition1, colData(dds))
mod0 <- model.matrix(~   1, colData(dds))
svseq <- svaseq(dat, mod, mod0, n.sv = 2)

sampleTablebe <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          be1=svseq$sv[,1],
                          be2=svseq$sv[,2],
                          batch=batchval)

sampleTablebe$sampleCondition <- paste(sampleTablebe$condition1, sampleTablebe$condition2, sep=".")

ddsHTSeqMFbe <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTablebe,
                                       directory=FOLDER,
                                       design=~sampleCondition + be1 + be2)

ddsHTSeqMFbe <- ddsHTSeqMFbe[rowSums(counts(ddsHTSeqMFbe)) > 1, ]
```

