---
title: "Nigerian-TCGA Differential Expression Analysis - Exploration of solutions to batch effect normalization"
author: "Padma Sheila Rajagopal, MD MPH"
date: "7/24/2019"

site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    df_print: paged
    toc: true
    theme: spacelab
    highlight: textmate
---

```{r Setup, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE)
library(gplots)
library(ggbiplot)
library(ggplot2)
library(edgeR)
library("DESeq2")
library("RUVSeq")
library(vsn)
library("genefilter")
library(pheatmap)
library(clusterProfiler)
library(pathview)
library("AnnotationDbi")
library(checkmate)
library(affy)
library("dplyr")
library(stringi)
library("hexbin")
library("org.Hs.eg.db")
library('GenomicFeatures')
library(EnsDb.Hsapiens.v75)
library("sva")
library(limma)
library(calibrate)
library(ggfortify)
library("ashr")
library(preprocessCore)
library("BiocParallel")
library(RColorBrewer)
library(Glimma)
register(MulticoreParam(4))
setwd("~/Research-Local/2019-rnaseq/Inputs/NigerianTCGA_quants-proteincoding")
```

#Translation from HTSeq raw counts -> DESeq2 table of raw counts
I have 84 TCGA patients (70 white or black) with whole-genome sequencing data and RNAseq data as well as 96 Nigerian patients with RNA-seq data. Raw counts were initially processed using HTSeq, so HTSeq raw counts (with one file per patient) are being formatted for use with DESeq2.
```{r Preparing the sampleTable using HTSeq raw counts, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
FOLDER <- "/Users/parajago/Research-Local/2019-rnaseq/Inputs/NigerianTCGA_quants-proteincoding"
sampleFiles <- grep("htseq.counts",list.files(FOLDER),value=TRUE)

#Differential gene expression setup based on race (b/w/other)
sampleConditionrace <- sampleFiles
countVar2=1
for (sample in sampleConditionrace){
  if (stri_detect_fixed(sample,"LIB")==TRUE){
    sampleConditionrace[countVar2] <- "Nigerian"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"black")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_black"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"white")==TRUE){
    sampleConditionrace[countVar2] <- "TCGA_white"
    countVar2=countVar2+1
  } else{
    sampleConditionrace[countVar2] <- "TCGA_other"
    countVar2=countVar2+1
  }
}

#Condition based on PAM50 subtype 
sampleConditionPAM50 <- sampleFiles
countVar2=1
for (sample in sampleConditionPAM50){
  if (stri_detect_fixed(sample,"Her2")==TRUE){
    sampleConditionPAM50[countVar2] <- "Her2"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"Basal")==TRUE){
    sampleConditionPAM50[countVar2] <- "Basal"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"LumA")==TRUE){
    sampleConditionPAM50[countVar2] <- "LumA"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"LumB")==TRUE){
    sampleConditionPAM50[countVar2] <- "LumB"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"PAMNL")==TRUE){
    sampleConditionPAM50[countVar2] <- "Normal"
    countVar2=countVar2+1
  } else{
    sampleConditionPAM50[countVar2] <- "PAM_other"
    countVar2=countVar2+1
  }
}

#Condition based on batch (relevant to the Nigerian patients only; no difference in batch for the TCGA patients)
batchval <- sampleFiles
countVar2=1
for (sample in batchval){
  if (stri_detect_fixed(sample,"batch1")==TRUE){
    batchval[countVar2] <- "batch1"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch23")==TRUE){
    batchval[countVar2] <- "batch23"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch4")==TRUE){
    batchval[countVar2] <- "batch4"
    countVar2=countVar2+1
  } else if (stri_detect_fixed(sample,"batch5")==TRUE){
    batchval[countVar2] <- "batch5"
    countVar2=countVar2+1
  } else{
    batchval[countVar2] <- "batchT"
    countVar2=countVar2+1
  }
}

table(sampleConditionrace, sampleConditionPAM50)

sampleTable2 <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          batch=batchval)

sampleTable2$sampleCondition <- paste(sampleTable2$condition1, sampleTable2$condition2, sep=".")

ddsHTSeqMF <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTable2,
                                       directory=FOLDER,
                                       design=~sampleCondition)

ddsHTSeqMF <- ddsHTSeqMF[rowSums(counts(ddsHTSeqMF)) > 0, ] #Pre-filtering the dataset by removing the rows without information about gene expression

ddsHTSeqMF$sampleCondition <- relevel(ddsHTSeqMF$sampleCondition, ref = "Nigerian.Basal") #explicitly making the Nigerian Basal patients the reference population
```

#Initial exploratory analysis
These are initial visualizations of our data sets to assess for batch effect between the Nigerian and TCGA sample sets. 
```{r Initial visualization of the data: VST, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
dds <- estimateSizeFactors(ddsHTSeqMF) #The size factor is the median ratio of the sample over a "pseudosample": for each gene, the geometric mean of all samples. This accounts for sequencing depth. 

vsd <- vst(ddsHTSeqMF, blind=FALSE) #Variance-stabilizing transformation, only using this since >50 samples

df <- bind_rows( #Graphical representation of the variance-stabilizing transformation
          as_data_frame(log2(counts(dds, normalized=TRUE)[, 1:2]+1)) %>%
          mutate(transformation = "log2(x + 1)"),
          as_data_frame(assay(vsd)[, 1:2]) %>% mutate(transformation = "vst"))
          colnames(df)[1:2] <- c("x", "y")

ggplot(df, aes(x = x, y = y)) + geom_hex(bins = 80) + coord_fixed() + facet_grid( . ~ transformation)
```

Figure 1: Graphical representation of the variance-stabilizing transformation relative to the standard transformation of log2(x+1). This reduces variance purely due to low read counts. VST is effectively employed in DESeq2. This transformation will be the input for subsequent visualizations (ONLY VISUALIZATIONS) in this section. 

```{r Initial visualization of the data: mean SD plots, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
meanSdPlot(log2(counts(estimateSizeFactors(ddsHTSeqMF),normalized=TRUE)[rowSums(counts(ddsHTSeqMF)) > 0,] + 1))
meanSdPlot(assay(vsd[rowSums(counts(ddsHTSeqMF)) > 0,]))
```

Figure 2: Plot of mean to variance to confirm that variance shrinking does work. This is another visualization of the approach noted in Figure 1.

```{r Batch effect assessment-PCAs-1, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
#PCAs at this point of the analysis are the VST transformed counts, using the top 500 variance genes
plotPCA(vsd, intgroup=c("condition1", "batch"), ntop=500)
```
PCA 1-1:PCA demonstrating the raw difference between Nigerian and TCGA patients with within-batch demonstration. Nigerian patients do not appear to have significant differneces based on their batches. (TCGA patients are known not to have significant difference based on batch performed.) 

There is a factor separating the TCGA patients into 2 groups that is not wholly ancestry dependent. 

Of note, the PCA as performed by only has 33% variance explained with two variables when transformation is performed. 

```{r Batch effect assessment-PCAs-2, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
plotPCA(vsd, intgroup=c("condition2"), ntop=1000)
```
PCA 1-2:PCA demonstrating association with breast cancer subtype and gene expression. This is noted in both groups and consistent with known biology. Again, only 32% variance is explained with 2 principal components. 

```{r Batch effect assessment-PCAs-3, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
plotPCA(vsd, intgroup=c("condition1", "condition2"), ntop=1000)
```
PCA 1-3: Integrated PCA demonstrating combined effect of breast cancer subtype and race. Subtype appears to account for the clustering seen in the TCGA group. Again, only 32% variance is explained with 2 principal components (with top 1000 genes). 

```{r Heatmap to assess clusters pre-analysis, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
#Heatmap to assess clusters pre-analysis
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 250)
hm  <- assay(vsd)[ topVarGenes, ]
hm  <- hm - rowMeans(hm)
anno <- as.data.frame(colData(vsd)[, c("condition1", "condition2")])
colnames(anno) <- c("Ethnicity", "PAM50-subtype")
rownames(anno) <- colnames(vsd)
pheatmap(hm, annotation_col = anno, cluster_rows=TRUE, cluster_cols=TRUE, clustering_distance_rows = "correlation", clustering_distance_cols = "correlation", clustering_method = "complete", show_rownames=FALSE, show_colnames = FALSE, main="Heat map: Relative VST-transformed\ncounts across samples without batch correction")
```
Heatmap 1-1: This heatmap is only using the top 250 genes and is competely heirarchically clustered. Results from this are consistent with the PCA. 

#Specific issues associated with our batches
-We do not know the extent to which general RNA expression varies within the Nigerian population compared to within TCGA populations because of ethnicity / common ancestral variants
-Biological differences are potentially confounded by batch effect (as Nigerians are a separate batch from TCGA); although no evidence of within-batch effect

We will discuss the following methods to manage the batch effect and our reasoning for our chosen approach.
(1) sva (1 principal component)
(2) RUVseq of housekeeping genes (k=1, with 2 and 6 housekeeping genes)
(3) Quantile normalization
(4) Limma removeBatchEffect

#Method 1: sva to account for batch effect related to the Nigerian/TCGA samples
We are using SVA to create an unsupervised hypothetical "batch effect" variable that is then used to account for testing differences between the two groups. 
```{r Batch effect assessment, message=FALSE, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
dat  <- counts(dds, normalized = TRUE)
idx  <- rowMeans(dat) > 1
dat  <- dat[idx, ]
mod  <- model.matrix(~condition1, colData(dds))
mod0 <- model.matrix(~   1, colData(dds))
svseq <- svaseq(dat, mod, mod0, n.sv = 1)

sampleTablebe <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          be1=svseq$sv[,1],
                          batch=batchval)

sampleTablebe$sampleCondition <- paste(sampleTablebe$condition1, sampleTablebe$condition2, sep=".")

ddsHTSeqMFbe <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTablebe,
                                       directory=FOLDER,
                                       design= ~be1+sampleCondition)

ddsHTSeqMFbe <- ddsHTSeqMFbe[rowSums(counts(ddsHTSeqMFbe)) > 0, ]

ddsHTSeqMFbe$sampleCondition <- relevel(ddsHTSeqMFbe$sampleCondition, ref = "Nigerian.Basal") #explicitly making the Nigerian Basal patients the reference population
```

Use of sva to create an unsupervised "batch effect" variable that effectively has full interaction with the Nigerian vs. TCGA variable does cause issues -- performing differential expression analysis under these conditions does lead to not converging in beta. 

I am performing an intial differential expression analysis with a batch-uncorrected version of the counts as well as the version that incorporates SVA-based batch correction.  

```{r Differential expression analysis -> batch effect comparison, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
ddsMF <- DESeq(ddsHTSeqMF) #This will refer to a version that does not include batch correction outside of what is accomplished through DESeq2 alone. 

ddsMFbe <-DESeq(ddsHTSeqMFbe) 
#4 rows did not converge in beta, labelled in mcols(object)$betaConv

resultsNames(ddsMF)
resultsNames(ddsMFbe)

fc = 0.58 #Subsequent threshold of signifcance for log2 fold change -> 0.58 = log2(1.5)
fdr = 0.05 #Subsequent threshold of significance for p-value (adjusted by FDR)
```

##sva -> differential expression analysis overall results
```{r MA plot of TCGA white Basal vs Nigerian Basal, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
cat("MA Plot 1: Original differential expression")
res <- lfcShrink(ddsMF, coef="sampleCondition_TCGA_white.Basal_vs_Nigerian.Basal", type="ashr", optmethod = "mixEM")
DESeq2::plotMA(res, ylim=c(-10,10), xlim=c(0.1,200))

cat("MA Plot 2: Batch-corrected differential expression")
res2 <- lfcShrink(ddsMFbe, coef="sampleCondition_TCGA_white.Basal_vs_Nigerian.Basal", type="ashr" , optmethod = "mixEM")
DESeq2::plotMA(res2, ylim=c(-10,10), xlim=c(0.1,200))
```

An MAPlot demonstrates log2 fold changes attributable to the condition over the mean of normalized counts for all the samples in the DESeqDataSet. Points will be colored red if the adjusted p value is less than 0.1. Points which fall out of the window are plotted as open triangles pointing either up or down.

We see an overrepresentation of significant negative log fold changes of expression in the Nigerian patients relative to the TCGA white patients, but this is exacerbated after the sva correction.

##sva -> Inter-TCGA test
```{r DE test: TCGA black and TCGA white, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
#Without batch correction
diffTCGA<- results(ddsMF, contrast=c("sampleCondition", "TCGA_black.Basal", "TCGA_white.Basal"), alpha=0.05)
nrow(diffTCGA)

#diffTCGA$foldChange <- NA
#row.pos <- which(! is.na(diffTCGA$log2FoldChange) & 
#                diffTCGA$log2FoldChange >= 0)
#row.neg <- which(! is.na(diffTCGA$log2FoldChange) & 
#                diffTCGA$log2FoldChange < 0)
#diffTCGA$foldChange[row.pos] <- 2^diffTCGA$log2FoldChange[row.pos]
#diffTCGA$foldChange[row.neg] <- -2^((-1) * diffTCGA$log2FoldChange[row.neg])
#diffTCGA <- data.frame(id = row.names(diffTCGA), diffTCGA)

#before <- nrow(diffTCGA)
#diffTCGA <- diffTCGA[!is.na(diffTCGA$foldChange) & ! is.na(diffTCGA$padj),];
#after <- nrow(diffTCGA)
#print(paste0('Genes removed = ', (before - after), 
#             ' (fold change is NA)'))

diffTCGA <- diffTCGA[(diffTCGA$log2FoldChange >= fc | diffTCGA$log2FoldChange <= -fc),] 
diffTCGA <- subset(diffTCGA, padj < fdr)
nrow(diffTCGA)

restemp <- lfcShrink(ddsMF, contrast=c("sampleCondition", "TCGA_black.Basal", "TCGA_white.Basal"), res = diffTCGA, type="ashr", optmethod = "mixEM")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

hist(restemp$pvalue, main="Histogram of unadjusted p-values of differential\n gene expression between basal breast cancers \nin TCGA black and \nTCGA white breast cancer patients")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in TCGA black and \nTCGA white breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(restemp, padj<0.01 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.01 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

#With batch correction
diffTCGA2<- results(ddsMFbe, contrast=c("sampleCondition", "TCGA_black.Basal", "TCGA_white.Basal"), alpha=0.05)
nrow(diffTCGA2)

#diffTCGA2$foldChange <- NA
#row.pos <- which(! is.na(diffTCGA2$log2FoldChange) & 
#                diffTCGA2$log2FoldChange >= 0)
#row.neg <- which(! is.na(diffTCGA2$log2FoldChange) & 
#                diffTCGA2$log2FoldChange < 0)
#diffTCGA2$foldChange[row.pos] <- 2^diffTCGA2$log2FoldChange[row.pos]
#diffTCGA2$foldChange[row.neg] <- -2^((-1) * diffTCGA2$log2FoldChange[row.neg])
#diffTCGA2 <- data.frame(id = row.names(diffTCGA2), diffTCGA2)

#before <- nrow(diffTCGA2)
#diffTCGA2 <- diffTCGA2[!is.na(diffTCGA2$foldChange) & ! is.na(diffTCGA2$padj),];
#after <- nrow(diffTCGA2)
#print(paste0('Genes removed = ', (before - after), 
#             ' (fold change is NA)'))

#diffTCGA2 <- diffTCGA2[(diffTCGA2$foldChange >= fc | diffTCGA2$foldChange <= -fc) & 
#              diffTCGA2$padj < fdr,]

diffTCGA2 <- diffTCGA2[(diffTCGA2$log2FoldChange >= fc | diffTCGA2$log2FoldChange <= -fc),] 
diffTCGA2 <- subset(diffTCGA2, padj < fdr)
nrow(diffTCGA2)

restemp2 <- lfcShrink(ddsMFbe, contrast=c("sampleCondition", "TCGA_black.Basal", "TCGA_white.Basal"), res = diffTCGA2, type="ashr", optmethod = "mixEM")

restemp2$temp <- row.names(restemp2)
restemp2$temp <- gsub("[.].+", "", restemp2$temp)

restemp2$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp2$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

hist(restemp2$pvalue, main="Histogram of unadjusted p-values of differential\n gene expression between basal breast cancers \nin TCGA black and white \n breast cancer patients\nBatch corrected")

with(restemp2, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in TCGA black and \nTCGA white breast cancer patients\nBatch corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(restemp2, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp2, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))
```

The histogram of unadjusted p-values for both uncorrected and SVA corrected appears anticonservative. Both volcano plots demonstrate a tendency for increased log fold change in TCGA white relative to TCGA black, but the batch corrected volcano plot shows greater dispersion / more significance in findings that is concerning for inappropriate exaggeration of effect of positive differential expression. 

The following is a demonstration example of comparing 32 Nigerian basal cases and 17 TCGA white basal cases. 

##sva -> Basal Nigerian-TCGA white (cross population) test
```{r DE test: Nigerian basal vs TCGA white basal, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
#Without batch correction
resBasal<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_white.Basal"), alpha=0.05)
nrow(resBasal)

#resBasal$foldChange <- NA
#row.pos <- which(! is.na(resBasal$log2FoldChange) & 
#                resBasal$log2FoldChange >= 0)
#row.neg <- which(! is.na(resBasal$log2FoldChange) & 
#                resBasal$log2FoldChange < 0)
#resBasal$foldChange[row.pos] <- 2^resBasal$log2FoldChange[row.pos]
#resBasal$foldChange[row.neg] <- -2^((-1) * resBasal$log2FoldChange[row.neg])
#resBasal <- data.frame(id = row.names(resBasal), resBasal)

#before <- nrow(resBasal)
#resBasal <- resBasal[!is.na(resBasal$foldChange) & ! is.na(resBasal$padj),];
#after <- nrow(resBasal)
#print(paste0('Genes removed = ', (before - after), 
#             ' (fold change is NA)'))

resBasal <- resBasal[(resBasal$log2FoldChange >= fc | resBasal$log2FoldChange <= -fc),]
resBasal <- subset(resBasal, padj < fdr)
nrow(resBasal)

restemp <- NULL
restemp <- lfcShrink(ddsMF, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_white.Basal"), res = resBasal, type="ashr", optmethod = "mixEM")

restemp$temp <- row.names(restemp)
restemp$temp <- gsub("[.].+", "", restemp$temp)

restemp$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

hist(restemp$pvalue, main="Histogram of unadjusted p-values of differential\n gene expression between basal breast cancers \nin Nigerian and \nTCGA white breast cancer patients")

with(restemp, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and \nTCGA white breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

#With batch correction
resBasal2<- results(ddsMFbe, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_white.Basal"), alpha=0.05)
nrow(resBasal2)

#resBasal2$foldChange <- NA
#row.pos <- which(! is.na(resBasal2$log2FoldChange) & 
#                resBasal2$log2FoldChange >= 0)
#row.neg <- which(! is.na(resBasal2$log2FoldChange) & 
#                resBasal2$log2FoldChange < 0)
#resBasal2$foldChange[row.pos] <- 2^resBasal2$log2FoldChange[row.pos]
#resBasal2$foldChange[row.neg] <- -2^((-1) * resBasal2$log2FoldChange[row.neg])
#resBasal2 <- data.frame(id = row.names(resBasal2), resBasal2)

#before <- nrow(resBasal2)
#resBasal2 <- resBasal2[!is.na(resBasal2$foldChange) & ! is.na(resBasal2$padj),];
#after <- nrow(resBasal2)
#print(paste0('Genes removed = ', (before - after), 
#             ' (fold change is NA)'))


resBasal2 <- resBasal2[(resBasal2$log2FoldChange >= fc | resBasal2$log2FoldChange <= -fc),]
resBasal2 <- subset(resBasal2, padj < fdr)
nrow(resBasal2) #More genes are considered significant with the batch correction method

resBasal2$temp <- row.names(resBasal2)
resBasal2$temp <- gsub("[.].+", "", resBasal2$temp)

resBasal2$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal2$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

restemp2 <- NULL
restemp2 <- lfcShrink(ddsMFbe, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_white.Basal"), res = resBasal2, type="ashr", optmethod = "mixEM")

restemp2$temp <- row.names(restemp2)
restemp2$temp <- gsub("[.].+", "", restemp2$temp)

restemp2$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp2$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

hist(restemp2$pvalue, main="Histogram of unadjusted p-values of differential\n gene expression between basal breast cancers \nin Nigerian and \nTCGA white breast cancer patients\n batch corrected")

with(restemp2, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and \nTCGA white breast cancer patients\n batch corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(restemp2, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp2, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

resBasal2print <- as.data.frame(resBasal2)
resBasal2print <- resBasal2print %>% arrange(log2FoldChange)
top_n(resBasal2print, 10, log2FoldChange)
top_n(resBasal2print, -10, log2FoldChange)
```

Although unadjusted p-value histogram appears anti-conservative again for both non batch corrected and sva corrected, volcano plot with batch correction appears to have an exaggered effect with numerous genes demonstrating >50x log fold change with significance. The volcano plot without batch correction is more consistent with expected findings, making the sva method concerning for use. 

#Method 2: RUVseq of housekeeping genes
Based on the paper by Krasnov et al. (https://www.frontiersin.org/articles/10.3389/fgene.2019.00097/full), we examined the housekeeping genes UBC and PUM1 across the Nigerian and TCGA samples to assess their differential expression. We also used the paper by Tilli et al. (https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-2946-1) to assess the efficacy of adding CCSER2, SYMPK and ANKRD17 for housekeeping comparison. 

UBC: ENSG00000150991.10
PUM1: ENSG00000134644.11
CCSER2: ENSG00000107771.11
SYMPK:  ENSG00000125755.14
ANKRD17: ENSG00000132466.13 -> Given the broad distribution of expression of this gene, this did not appear to be a good standard for comparison in our cohort. 

```{r UBC and PUM1 cross-ethnic comparisons, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
resBasal<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_white.Basal"), alpha=0.05)

genename <- "ENSG00000150991.10"
plotCounts(dds, gene = genename, intgroup=c("condition1"), main="Distribution of UBC expression across groups (before DE analysis)")

genename <- "ENSG00000134644.11"
plotCounts(dds, gene = genename, intgroup=c("condition1"), main="Distribution of PUM1 expression across groups (before DE analysis)")

genename <- "ENSG00000107771.11"
plotCounts(dds, gene = genename, intgroup=c("condition1"), main="Distribution of CCSER2 expression across groups (before DE analysis)")

genename <- "ENSG00000125755.14"
plotCounts(dds, gene = genename, intgroup=c("condition1"), main="Distribution of SYMPK expression across groups (before DE analysis)")

genename <- "ENSG00000132466.13"
plotCounts(dds, gene = genename, intgroup=c("condition1"), main="Distribution of ANKRD17 expression across groups (before DE analysis)")

ubctempassay <- subset(resBasal, rownames(resBasal)=="ENSG00000150991.10")
ubctempassay[,1:6]
pum1tempassay <- subset(resBasal, rownames(resBasal)=="ENSG00000134644.11")
pum1tempassay[,1:6]
ccser2tempassay <- subset(resBasal, rownames(resBasal)=="ENSG00000107771.11")
ccser2tempassay[,1:6]
sympktempassay <- subset(resBasal, rownames(resBasal)=="ENSG00000125755.14")
sympktempassay[,1:6]
```

UBC and PUM1 (as well as CCSER2) demonstrate signifcantly different distribution between the Nigerian and TCGA samples (although the log2 fold change is less for the PUM1 gene). SYMPK does not have significantly different distribution. This suggests that normalization of the housekeeping genes can be used as a means of comparing the RNA read counts. 

```{r RUVSeq use of housekeeping genes for normalization, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
genenames <- c("ENSG00000150991.10", "ENSG00000134644.11", "ENSG00000107771.11", "ENSG00000125755.14")

set1 <- RUVg(counts(ddsHTSeqMF), genenames, k=1)

plotRLE(counts(ddsHTSeqMF), outline=FALSE, ylim=c(-4, 4))
plotRLE(set1$normalizedCounts, outline=FALSE, ylim=c(-4, 4))

set1pca <- set1$normalizedCounts
colnames(set1pca) <- sampleTable2$sampleCondition
plotPCA(set1pca, main="PCA of RUV normalization results prior to DE analysis")

sampleTableRUV <- data.frame(sampleName=gsub(".htseq.counts","",sampleFiles),
                          fileName=sampleFiles,
                          condition1=sampleConditionrace,
                          condition2=sampleConditionPAM50,
                          W_1 = set1$W)

sampleTableRUV$sampleCondition <- paste(sampleTableRUV$condition1, sampleTableRUV$condition2, sep=".")

ddsHTSeqMFRUV <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTableRUV,
                                       directory=FOLDER,
                                       design= ~W_1+sampleCondition)

ddsHTSeqMFRUV <- ddsHTSeqMFRUV[rowSums(counts(ddsHTSeqMFRUV)) > 0, ]

ddsHTSeqMFRUV$sampleCondition <- relevel(ddsHTSeqMFRUV$sampleCondition, ref = "Nigerian.Basal") #explicitly making the Nigerian Basal patients the reference population

ddsMFRUV <- DESeq(ddsHTSeqMFRUV) #Running differential expression with the two housekeeping genes for normalization
```
The RUV normalization demonstrates a different PCA with some clustering based on Nigerian vs. TCGA and some loss of clustering based on subtype relative to VSD normalization or SVA batch effect correction. 

##RUVseq -> differential expression analysis overall results
```{r MA plot of TCGA white Basal vs Nigerian Basal for RUV, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
cat("MA Plot: RUVSeq-corrected differential expression")
res2 <- lfcShrink(ddsMFRUV, coef="sampleCondition_TCGA_white.Basal_vs_Nigerian.Basal", type="ashr" , optmethod = "mixEM")
DESeq2::plotMA(res2, ylim=c(-10,10), xlim=c(0.1,200))
```

Again, we see an overrepresentation of significant negative log fold changes of expression in the Nigerian patients relative to the TCGA white patients that is exacerbated by the RUV correction.

##RUVseq -> Inter-TCGA test
```{r DE test: TCGA black and TCGA white for RUV, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
#With RUV correction
diffTCGA3<- results(ddsMFRUV, contrast=c("sampleCondition", "TCGA_black.Basal", "TCGA_white.Basal"), alpha=0.05)
nrow(diffTCGA3)
diffTCGA3 <- diffTCGA3[(diffTCGA3$log2FoldChange >= fc | diffTCGA3$log2FoldChange <= -fc),] 
diffTCGA3 <- subset(diffTCGA3, padj < fdr)
nrow(diffTCGA3)

restemp3 <- NULL
restemp3 <- lfcShrink(ddsMFRUV, contrast=c("sampleCondition", "TCGA_black.Basal", "TCGA_white.Basal"), res = diffTCGA3, type="ashr", optmethod = "mixEM")

restemp3$temp <- row.names(restemp3)
restemp3$temp <- gsub("[.].+", "", restemp3$temp)

restemp3$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp3$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

hist(restemp3$pvalue, main="Histogram of unadjusted p-values of differential\n gene expression between basal breast cancers \nin TCGA black and white \n breast cancer patients\nRUV corrected")

with(restemp3, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in TCGA black and \nTCGA white breast cancer patients\nRUV corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(restemp3, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp3, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))
```

Again, we see an overrepresentation of differentially expressed genes on the positive side relative to the negative, concerning for the same issue as seen in SVA batch correction. 

##RUVSeq -> Basal Nigerian-TCGA white and TCGA black (cross population) tests
```{r DE test: Nigerian basal vs TCGA white basal and TCGA black basal for RUV, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
#With RUV correction
resBasal3<- results(ddsMFRUV, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_white.Basal"), alpha=0.05)
nrow(resBasal3)
resBasal3 <- resBasal3[(resBasal3$log2FoldChange >= fc | resBasal3$log2FoldChange <= -fc),]
resBasal3 <- subset(resBasal3, padj < fdr)
nrow(resBasal3)

resBasal3$temp <- row.names(resBasal3)
resBasal3$temp <- gsub("[.].+", "", resBasal3$temp)

resBasal3$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=resBasal3$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")
resBasal3$foldchange <- (2^abs(resBasal3$log2FoldChange))

restemp3 <- NULL
restemp3 <- lfcShrink(ddsMFRUV, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_white.Basal"), res = resBasal3, type="ashr", optmethod = "mixEM")

restemp3$temp <- row.names(restemp3)
restemp3$temp <- gsub("[.].+", "", restemp3$temp)

restemp3$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp3$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

hist(restemp3$pvalue, main="Histogram of unadjusted p-values of differential\n gene expression between basal breast cancers \nin Nigerian and \nTCGA white breast cancer patients\n RUV corrected")

with(restemp3, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and \nTCGA white breast cancer patients\nRUV corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(restemp3, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp3, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))

resBasal3print <- as.data.frame(resBasal3)
resBasal3print <- resBasal3print %>% arrange(log2FoldChange)

resBasal4<- results(ddsMFRUV, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_black.Basal"), alpha=0.05)
nrow(resBasal4)
resBasal4 <- resBasal4[(resBasal4$log2FoldChange >= fc | resBasal4$log2FoldChange <= -fc),]
resBasal4 <- subset(resBasal4, padj < fdr)
nrow(resBasal4)

restemp4 <- NULL
restemp4 <- lfcShrink(ddsMFRUV, contrast=c("sampleCondition", "Nigerian.Basal", "TCGA_black.Basal"), res = resBasal4, type="ashr", optmethod = "mixEM")

restemp4$temp <- row.names(restemp4)
restemp4$temp <- gsub("[.].+", "", restemp4$temp)

restemp4$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restemp4$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

hist(restemp4$pvalue, main="Histogram of unadjusted p-values of differential\n gene expression between basal breast cancers \nin Nigerian and \nTCGA black breast cancer patients\n RUV corrected")

with(restemp4, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and \nTCGA black breast cancer patients\nRUV corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(restemp4, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restemp4, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))
```

Although unadjusted p-value histogram appears anti-conservative again for RUV data, volcano plot with RUV correction appears to have the same exaggered effect as seen with the SVA batch effect. 

There is some reassurance that both methods are finding genes in common.

#Method 3: Quantile normalization
Quantile normalization is another technique that can be used to make distributions comparable. To quantile normalize two or more distributions to each other, without a reference distribution, one must sort the values within the distribution, then set to the average (usually, arithmetic mean) of the distributions. So the highest value in all cases becomes the mean of the highest values, the second highest value becomes the mean of the second highest values, and so on.

```{r Quantile normalization setup and visualization, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
countmatrix <- assay(ddsHTSeqMF) #Raw counts organized into matrix format from individual files
countmatrix2 <- log2(countmatrix + 1) #Basic transformation of the count data 
countmatrix2 <- countmatrix2[rowSums(countmatrix2) > 0, ] #Remove if there is no data re: counts

plot(density(countmatrix2[,1]),lwd=3,ylim=c(0,.30), main="Density of counts with log2[count]+1 transformation ONLY") 
for(i in 1:180){lines(density(countmatrix2[,i]),lwd=3)} #This demonstrates that there is a difference in distributions between the Nigerian and TCGA data with basic log transformation normalization 

norm_countmatrix <- as.matrix(countmatrix2) 
norm_countmatrix = normalize.quantiles(norm_countmatrix)
plot(density(norm_countmatrix[,1]),lwd=3,ylim=c(0,.3), main="Density of counts with quantile normalization")
for(i in 1:180){lines(density(norm_countmatrix[,i]),lwd=3)} #This demonstrates the effect of comparative quantile normalization

colnames (norm_countmatrix) <- colnames (countmatrix2)
rownames (norm_countmatrix) <- rownames (countmatrix2)

norm_countmatrix <- as.data.frame(norm_countmatrix)
countmatrixNigerian <- dplyr::select(norm_countmatrix, contains("LIB"))
plot(density(countmatrixNigerian[,1]),lwd=3,ylim=c(0,.3), main="Density of counts with quantile normalization - Nigerian")
for(i in 1:96){lines(density(countmatrixNigerian[,i]),lwd=3)} #This demonstrates the result of the normalized Nigerian counts separately

tcgacolnames <- colnames(countmatrix)
tcgacolnames <- setdiff(tcgacolnames, colnames(countmatrixNigerian))
countmatrixTCGA <- norm_countmatrix[ , tcgacolnames]
plot(density(countmatrixTCGA[,1]),lwd=3,ylim=c(0,.3), main="Density of counts with quantile normalization - TCGA")
for(i in 1:84){lines(density(countmatrixTCGA[,i]),lwd=3)} #This demonstrates the result of the normalized TCGA counts separately

norm_countmatrix <- as.data.frame(norm_countmatrix)
t_norm_countmatrix <- t(norm_countmatrix)

t_norm_countmatrix <- cbind (t_norm_countmatrix, sampleTable2) #This binds the characteristics of the original patients to the quantile normalized counts. CBinding was checked to make sure that patients were correctly aligned to characteristics. 

quant.pca <- prcomp(t_norm_countmatrix[,1:19724])
autoplot(quant.pca, data=t_norm_countmatrix, colour='sampleCondition', main="PCA of quantile normalization results prior to DE analysis")
```
The quantile normalization demonstrates a PCA that has similar clustering and % explanations relative to VSD normalization, but the differences between groups have been narrowed. 

##Quantile normalization -> DE analysis and Inter-TCGA test

```{r Limma-voom differential expression results for TCGA black/TCGA white Basal, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
design <- t_norm_countmatrix
design <- design %>% dplyr::select(sampleCondition)

design$sampleCondition <- ifelse (design$sampleCondition=="TCGA_black.Basal", 0, as.character(design$sampleCondition))
design$sampleCondition <- ifelse (design$sampleCondition=="TCGA_white.Basal", 1, as.character(design$sampleCondition))

design$sampleCondition <- ifelse (design$sampleCondition==0 | design$sampleCondition==1, design$sampleCondition, NA)

design <- design %>% subset(is.na(sampleCondition)==FALSE)

design$TCGA_black.Basal <- ifelse (design$sampleCondition==0, 1, 0)
design$TCGA_white.Basal <- ifelse (design$sampleCondition==1, 1, 0)

design$sampleCondition <- NULL

quantids <- rownames(design)

design <- model.matrix(~0+design$TCGA_black.Basal+design$TCGA_white.Basal)
rownames(design) <- quantids
colnames(design) <- c("TCGA_black", "TCGA_white")

quantdata <- t(counts(ddsHTSeqMF))
quantdata <- as.data.frame(quantdata)
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)
quantdata <- quantdata[rowSums(quantdata) > 0, ] 

contr.matrix <- makeContrasts(TCGAblackvsTCGAwhite = TCGA_black-TCGA_white, levels=colnames(design))

dge <- DGEList(counts=quantdata)
dge <- calcNormFactors(dge)

v=voom(dge,design,plot=T, normalize="quantile")

fit <- lmFit(v, design)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential\n gene expression between basal breast cancers \nin TCGA black and \nTCGA white breast cancer patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       temp = gsub("[.].+", "", rownames(fit)))

df_limma$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=df_limma$temp,
                     column="SYMBOL",
                     keytype="GENEID",        
                     multiVals="first")

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in TCGA black and \nTCGA white breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))
```

It has been previously demonstrated that there are expression difference between ethnic groups within the TCGA cohort, and this technique appears to drastically minimize this effect. 

##Quantile normalization -> Basal Nigerian-TCGA white (cross population) test

```{r Limma-voom differential expression results for Nigerian/TCGA Basal, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
design <- t_norm_countmatrix
design <- design %>% dplyr::select(sampleCondition)

design$sampleCondition <- ifelse (design$sampleCondition=="Nigerian.Basal", 0, as.character(design$sampleCondition))
design$sampleCondition <- ifelse (design$sampleCondition=="TCGA_white.Basal", 1, as.character(design$sampleCondition))

design$sampleCondition <- ifelse (design$sampleCondition==0 | design$sampleCondition==1, design$sampleCondition, NA)

design <- design %>% subset(is.na(sampleCondition)==FALSE)

design$Nigerian.Basal <- ifelse (design$sampleCondition==0, 1, 0)
design$TCGA_white.Basal <- ifelse (design$sampleCondition==1, 1, 0)

design$sampleCondition <- NULL

quantids <- rownames(design)

design <- model.matrix(~0+design$Nigerian.Basal+design$TCGA_white.Basal)
rownames(design) <- quantids
colnames(design) <- c("Nigerian", "TCGA_white")

quantdata <- t(counts(ddsHTSeqMF))
quantdata <- as.data.frame(quantdata)
quantdata <- quantdata[quantids,]
quantdata <- t(quantdata)
quantdata <- quantdata[rowSums(quantdata) > 0, ] 

contr.matrix <- makeContrasts( NigerianvsTCGA = Nigerian-TCGA_white, levels=colnames(design))

dge <- DGEList(counts=quantdata)
dge <- calcNormFactors(dge)

v=voom(dge,design,plot=T, normalize="quantile")

fit <- lmFit(v, design)
fit <- contrasts.fit(fit, contrasts=contr.matrix)
fit <- eBayes(fit)
dt <- decideTests(fit)
summary(dt)

hist(fit$p.value, ylim=c(0,3000), main="Histogram of unadjusted p-values of differential\n gene expression between basal breast cancers \nin Nigerian and \nTCGA white breast cancer patients\n quantile corrected")

qvals<-p.adjust(fit$p.value[,1], method='fdr')

df_limma <- data_frame(log2FoldChange = fit$coefficients[,1], 
                       pval = fit$p.value[,1],
                       padj = qvals,
                       temp = gsub("[.].+", "", rownames(fit)))

df_limma$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=df_limma$temp,
                     column="SYMBOL",
                     keytype="GENEID",        
                     multiVals="first")

df_limmaprint <- as.data.frame(df_limma)
df_limmaprint <- df_limmaprint %>% arrange(log2FoldChange)
top_n(df_limmaprint, 10, log2FoldChange)
top_n(df_limmaprint, -10, log2FoldChange)

with(df_limma, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential gene expression between basal \nbreast cancers in Nigerian and \nTCGA white breast cancer patients\nquantile corrected", xlim=c(-50,50), ylim=c(0,70)))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(df_limma, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))
```

As per Wei Shi's preferred method: Counts were converted to log2 counts per million, quantile normalized and precision weighted with the 'voom' function of the limma package46,47. A linear model was fitted to each gene, and empirical Bayes moderated t-statistics were used to assess differences in expression48. Empirical Bayes moderated-t P values were computed relative to a fold-change cutoff of 1.5-fold. P values were adjusted to control the global FDR across all comparisons with the 'global' option of the limma package. Genes were considered differentially expressed if they had an FDR of 0.05.

##Limma's removeBatchEffect for visualization only
```{r Removing batch effect with Limma:removeBatchEffect (visualization only), message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
mat <- assay(vsd) 
mat <- limma::removeBatchEffect(mat, vsd$condition1)
assay(vsd) <- mat
plotPCA(vsd, intgroup=c("condition1", "condition2"))
```

```{r Heatmap to assess clusters after Limma:removeBatchEffect (visualization only), message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
#Heatmap to assess clusters post-batch effect removal
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 250)
hm  <- assay(vsd)[ topVarGenes, ]
hm  <- hm - rowMeans(hm)
anno <- as.data.frame(colData(vsd)[, c("condition1", "condition2")])
colnames(anno) <- c("Ethnicity", "PAM50-subtype")
rownames(anno) <- colnames(vsd)
pheatmap(hm, annotation_col = anno, cluster_rows=TRUE, cluster_cols=TRUE, clustering_distance_rows = "correlation", clustering_distance_cols = "correlation", clustering_method = "complete", show_rownames=FALSE, show_colnames = FALSE, main="Heat map: Relative VST-transformed\ncounts across samples after Limma:removeBatchEffect")
```
This heatmap after batch correction with limma's removeBatchEffect method is using the top 250 genes and is competely heirarchically clustered. These clusters are difficult to separate from each other, and it is not clear that the Limma removeBatchEffect method does not overcompensate and competely flatten existing biological signal.

In addition, the voom method is considered more appropriate for data with significant noise / abnormalities. 

#Internal data validity assessments
We will test the internal validity of using DESeq2 using comparisons between subtypes, comparisons within the Nigerian cohort. This will not use the batch effect corrected approach with SVA given the demonstrations we showed earlier of the inappropriate exaggeration of positive findings.

##Inter-subtype gene expression testing
```{r DE test: Nigerian basal vs Nigerian Her2, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
rescrosstype<- results(ddsMF, contrast=c("sampleCondition", "Nigerian.Basal", "Nigerian.Her2"), alpha=0.05)
nrow(rescrosstype)

#resBasal2$foldChange <- NA
#row.pos <- which(! is.na(resBasal2$log2FoldChange) & 
#                resBasal2$log2FoldChange >= 0)
#row.neg <- which(! is.na(resBasal2$log2FoldChange) & 
#                resBasal2$log2FoldChange < 0)
#resBasal2$foldChange[row.pos] <- 2^resBasal2$log2FoldChange[row.pos]
#resBasal2$foldChange[row.neg] <- -2^((-1) * resBasal2$log2FoldChange[row.neg])
#resBasal2 <- data.frame(id = row.names(resBasal2), resBasal2)

#before <- nrow(resBasal2)
#resBasal2 <- resBasal2[!is.na(resBasal2$foldChange) & ! is.na(resBasal2$padj),];
#after <- nrow(resBasal2)
#print(paste0('Genes removed = ', (before - after), 
#             ' (fold change is NA)'))


rescrosstype <- rescrosstype[(rescrosstype$log2FoldChange >= fc | rescrosstype$log2FoldChange <= -fc),]
rescrosstype <- subset(rescrosstype, padj < fdr)
nrow(rescrosstype)

restempcheck <- NULL
restempcheck <- lfcShrink(ddsMF, contrast=c("sampleCondition", "Nigerian.Basal", "Nigerian.Her2"), res = rescrosstype, type="ashr", optmethod = "mixEM")

restempcheck$temp <- row.names(restempcheck)
restempcheck$temp <- gsub("[.].+", "", restempcheck$temp)

restempcheck$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restempcheck$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

hist(restempcheck$pvalue, main="Histogram of unadjusted p-values of differential\n gene expression between \n Nigerian basal and Nigerian Her2 breast cancer patients")

with(restempcheck, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential\n gene expression between \n Nigerian basal and Nigerian Her2 breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(restempcheck, padj<0.01 & abs(log2FoldChange)>25), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(restempcheck, padj<0.01 & abs(log2FoldChange)>25), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))
```

The unadjusted p-value histogram and volcano plot appear appropriate without batch correction in the cross-subtype test. It is reassuring that we are seeing more normal volcano plots on the within-assay comparison. 

```{r Hand code value check of Her2, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
genename <- "ENSG00000141736.9"
her2comparematrix <- plotCounts(ddsMF, gene = genename, intgroup=c("condition1", "condition2"), main="Distribution of HER2 expression across groups", returnData = TRUE)

her2comparematrix <- her2comparematrix %>% subset(condition1=="Nigerian" & (condition2=="Her2" | condition2=="Basal"))
ggplot(her2comparematrix, aes (x=condition2, y=count)) + geom_point() + labs(title="Comparison of normalized RNAseq counts of HER2 expression between\nNigerian basal tumors and Nigerian HER2 tumors") + xlab("Breast cancer Subtype") + ylab("Normalized RNAseq count (pre DE)") + scale_y_log10()

her2tempassay <- subset(rescrosstype, rownames(rescrosstype)=="ENSG00000141736.9")
her2tempassay[,1:6]

tempassay <- as.data.frame(assay(ddsHTSeqMF))
tempassay <- dplyr::select(tempassay, contains("LIB"))
tempassay <- dplyr::select(tempassay, matches('Basal|Her2'))
tempassay <- subset(tempassay, rownames(tempassay)=="ENSG00000141736.9")
tempassaybasal <- dplyr::select(tempassay, matches('Basal'))
tempassayher2 <- dplyr::select(tempassay, matches('Her2'))

tempassaybasal <-as.numeric(tempassaybasal)
tempassayher2 <-as.numeric(tempassayher2)
t.test (tempassaybasal, tempassayher2)
```
The cross subtype comparison did find a statistically significant difference between HER2 expression between the Nigerian basal and Her tumors, which is appropriate when the data is visualized as well.

We will now assess arbitrary within-Nigerian samples to identify any other within sample sources of bias. 

##Arbitrary grouping of Nigerian patients cross-checked with each other
```{r Nigerian group cross check, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
table(batchval, sampleConditionPAM50)

sampletableNigerian <- sampleTable2
sampletableNigerian <- subset(sampletableNigerian, batch=="batch23" | batch=="batch5")
sampletableNigerian$condition1 <- NULL

dsHTSeqMFNigerian <- DESeqDataSetFromHTSeqCount(sampleTable=sampletableNigerian,
                                       directory=FOLDER,
                                       design=~batch)
dsHTSeqMFNigerian <- dsHTSeqMFNigerian[rowSums(counts(dsHTSeqMFNigerian)) > 0, ] #Pre-filtering the dataset by removing the rows without information about gene expression
ddsMFNigerianexample <- DESeq(dsHTSeqMFNigerian)

resNigerianNigerian<- results(ddsMFNigerianexample, contrast=c("batch", "batch23", "batch5"), alpha=0.05)

resNigerianNigerian <- resNigerianNigerian[(resNigerianNigerian$log2FoldChange >= fc | resNigerianNigerian$log2FoldChange <= -fc),]
resNigerianNigerian <- subset(resNigerianNigerian, padj < fdr)
nrow(resNigerianNigerian) #More genes are considered significant with the batch correction method

restempcheck <- NULL
restempcheck <- lfcShrink(ddsMFNigerianexample, contrast=c("batch", "batch23", "batch5"), res = resNigerianNigerian, type="ashr", optmethod = "mixEM")

restempcheck$temp <- row.names(restempcheck)
restempcheck$temp <- gsub("[.].+", "", restempcheck$temp)

restempcheck$symbol <- mapIds(EnsDb.Hsapiens.v75,
                     keys=restempcheck$temp,
                     column="SYMBOL",
                     keytype="GENEID",           
                     multiVals="first")

hist(restempcheck$pvalue, main="Histogram of unadjusted p-values of differential\n gene expression between \n grouped Nigerian breast cancer patients")

with(restempcheck, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot of differential\n gene expression between \n grouped Nigerian breast cancer patients", xlim=c(-50,50), ylim=c(0,70)))
with(subset(restempcheck, padj<0.05 & (2^(abs(log2FoldChange))>50)), points(log2FoldChange, -log10(padj), pch=20, col="blue"))
with(subset(restempcheck, padj<0.05 & (2^(abs(log2FoldChange))>50)), textxy(log2FoldChange, -log10(padj), labs=symbol, cex=.5))
```

#Methods cross check
To compare results between methods to identify the best approach to normalization, we noted the top and bottom log2foldchange genes per method. 
```{r Methods cross check, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE}
print("SVA Batch effect correction")
top_n(resBasal2print, 10, log2FoldChange)
top_n(resBasal2print, -10, log2FoldChange)

print("RUV housekeeping gene Batch effect correction")
top_n(resBasal3print, 10, log2FoldChange)
top_n(resBasal3print, -10, log2FoldChange)

print("Limma-voom with quantile normalization batch effect correction")
top_n(df_limmaprint, 10, log2FoldChange)
top_n(df_limmaprint, -10, log2FoldChange)

testgenename="ENSG00000126549.5"
plotCounts(dds, gene = testgenename, intgroup=c("condition1"), main="Distribution of STATH expression across groups (before DE analysis)")

testgenename="ENSG00000171560.10"
plotCounts(dds, gene = testgenename, intgroup=c("condition1"), main="Distribution of FGA expression across groups (before DE analysis)")

testgenename="ENSG00000102891.3"
plotCounts(dds, gene = testgenename, intgroup=c("condition1"), main="Distribution of MT4 expression across groups (before DE analysis)")

testgenename="ENSG00000257594.2"
plotCounts(dds, gene = testgenename, intgroup=c("condition1"), main="Distribution of GALNT4 expression across groups (before DE analysis)")

testgenename="ENSG00000183426.11"
plotCounts(dds, gene = testgenename, intgroup=c("condition1"), main="Distribution of NPIPA1 expression across groups (before DE analysis)")

```

#ADDENDUM: HRD signature analysis
```{r: HRD signature analysis, message=FALSE, warning=FALSE, echo=FALSE, cache=TRUE, eval=FALSE}
sampleTable3 <- sampleTable %>% dplyr::filter(condition1=="Nigerian")
sampleTable3 <- sampleTable3 %>% dplyr::select("sampleName", "fileName", "condition2", "batch")
colnames(sampleTable3)[3] <- "sampleCondition"

ddsHTSeqMFNigerianonly <- DESeqDataSetFromHTSeqCount(sampleTable=sampleTable3,
                                       directory=FOLDER,
                                       design=~sampleCondition)

ddsHTSeqMFNigerianonly <- ddsHTSeqMFNigerianonly[rowSums(counts(ddsHTSeqMFNigerianonly)) > 0, ] #Pre-filtering the dataset by removing the rows without information about gene expression

dds <- estimateSizeFactors(ddsHTSeqMFNigerianonly)

HRDgenes=c("ENSG00000149311.13", "ENSG00000105393.11", "ENSG00000138376.6", "ENSG00000197299.6", "ENSG00000012048.15", "ENSG00000139618.10", "ENSG00000158019.16", "ENSG00000136492.4", "ENSG00000183765.16", "ENSG00000154920.10", "ENSG00000163322.9", "ENSG00000187741.10", "ENSG00000187790.6", "ENSG00000020922.8", "ENSG00000104320.9", "ENSG00000083093.5", "ENSG00000062822.8", "ENSG00000113522.9", "ENSG00000182185.14", "ENSG00000108384.10", "ENSG00000185379.16", "ENSG00000002016.12", "ENSG00000197275.8", "ENSG00000085999.7", "ENSG00000101773.12", "ENSG00000132383.7", "ENSG00000139351.10", "ENSG00000177302.10", "ENSG00000100038.15", "ENSG00000163781.8", "ENSG00000087206.12", "ENSG00000196584.2", "ENSG00000126215.9")

ATMcounts <- plotCounts(dds, gene="ENSG00000149311.13", intgroup="sampleCondition", 
                returnData=TRUE)
ATMcounts$sampleName <- row.names(ATMcounts)
ATMcounts$percent10 <- ntile(ATMcounts$count, 10)
temp <- ATMcounts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="ATMlowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

BARD1counts <- plotCounts(dds, gene="ENSG00000138376.6", intgroup="sampleCondition", 
                returnData=TRUE)
BARD1counts$sampleName <- row.names(BARD1counts)
BARD1counts$percent10 <- ntile(BARD1counts$count, 10)
temp <- BARD1counts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="BARD1lowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

BLMcounts <- plotCounts(dds, gene="ENSG00000197299.6", intgroup="sampleCondition", 
                returnData=TRUE)
BLMcounts$sampleName <- row.names(BLMcounts)
BLMcounts$percent10 <- ntile(BLMcounts$count, 10)
temp <- BLMcounts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="BLMlowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

BRCA1counts <- plotCounts(dds, gene="ENSG00000012048.15", intgroup="sampleCondition", 
                returnData=TRUE)
BRCA1counts$sampleName <- row.names(BRCA1counts)
BRCA1counts$percent10 <- ntile(BRCA1counts$count, 10)
temp <- BRCA1counts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="BRCA1lowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

BRCA2counts <- plotCounts(dds, gene="ENSG00000139618.10", intgroup="sampleCondition", 
                returnData=TRUE)
BRCA2counts$sampleName <- row.names(BRCA2counts)
BRCA2counts$percent10 <- ntile(BRCA2counts$count, 10)
temp <- BRCA2counts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="BRCA2lowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

BRIP1counts <- plotCounts(dds, gene="ENSG00000136492.4", intgroup="sampleCondition", 
                returnData=TRUE)
BRIP1counts$sampleName <- row.names(BRIP1counts)
BRIP1counts$percent10 <- ntile(BRIP1counts$count, 10)
temp <- BRIP1counts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="BRIP1lowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

CHEK2counts <- plotCounts(dds, gene="ENSG00000183765.16", intgroup="sampleCondition", 
                returnData=TRUE)
CHEK2counts$sampleName <- row.names(CHEK2counts)
CHEK2counts$percent10 <- ntile(CHEK2counts$count, 10)
temp <- CHEK2counts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="CHEK2lowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

FAM175Acounts <- plotCounts(dds, gene="ENSG00000163322.9", intgroup="sampleCondition", 
                returnData=TRUE)
FAM175Acounts$sampleName <- row.names(FAM175Acounts)
FAM175Acounts$percent10 <- ntile(FAM175Acounts$count, 10)
temp <- FAM175Acounts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="FAM175Alowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)


FANCAcounts <- plotCounts(dds, gene="ENSG00000187741.10", intgroup="sampleCondition", 
                returnData=TRUE)
FANCAcounts$sampleName <- row.names(FANCAcounts)
FANCAcounts$percent10 <- ntile(FANCAcounts$count, 10)
temp <- FANCAcounts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="FANCAlowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

FANCMcounts <- plotCounts(dds, gene="ENSG00000187790.6", intgroup="sampleCondition", 
                returnData=TRUE)
FANCMcounts$sampleName <- row.names(FANCMcounts)
FANCMcounts$percent10 <- ntile(FANCMcounts$count, 10)
temp <- FANCMcounts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="FANCMlowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

MRE11Acounts <- plotCounts(dds, gene="ENSG00000020922.8", intgroup="sampleCondition", 
                returnData=TRUE)
MRE11Acounts$sampleName <- row.names(MRE11Acounts)
MRE11Acounts$percent10 <- ntile(MRE11Acounts$count, 10)
temp <- MRE11Acounts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="MRE11Alowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

NBNcounts <- plotCounts(dds, gene="ENSG00000104320.9", intgroup="sampleCondition", 
                returnData=TRUE)
NBNcounts$sampleName <- row.names(NBNcounts)
NBNcounts$percent10 <- ntile(NBNcounts$count, 10)
temp <- NBNcounts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="NBNlowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

PALB2counts <- plotCounts(dds, gene="ENSG00000083093.5", intgroup="sampleCondition", 
                returnData=TRUE)
PALB2counts$sampleName <- row.names(PALB2counts)
PALB2counts$percent10 <- ntile(PALB2counts$count, 10)
temp <- PALB2counts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="PALB2lowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

POLD1counts <- plotCounts(dds, gene="ENSG00000062822.8", intgroup="sampleCondition", 
                returnData=TRUE)
POLD1counts$sampleName <- row.names(POLD1counts)
POLD1counts$percent10 <- ntile(POLD1counts$count, 10)
temp <- POLD1counts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="POLD1lowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

RAD51Ccounts <- plotCounts(dds, gene="ENSG00000108384.10", intgroup="sampleCondition", 
                returnData=TRUE)
RAD51Ccounts$sampleName <- row.names(RAD51Ccounts)
RAD51Ccounts$percent10 <- ntile(RAD51Ccounts$count, 10)
temp <- RAD51Ccounts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="RAD51Clowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

XRCCcounts <- plotCounts(dds, gene="ENSG00000196584.2", intgroup="sampleCondition", 
                returnData=TRUE)
XRCCcounts$sampleName <- row.names(XRCCcounts)
XRCCcounts$percent10 <- ntile(XRCCcounts$count, 10)
temp <- XRCCcounts %>% dplyr::filter(percent10==1) %>% dplyr::select(sampleName, sampleCondition, count)
#write.table(temp, file="XRCClowest10thpercentile.csv", sep=",", row.names=FALSE, col.names=TRUE)

testgenename="ENSG00000149311.13"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of ATM expression in Nigerian cohort")

testgenename="ENSG00000105393.11"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of BABAM1 expression in Nigerian cohort")

testgenename="ENSG00000138376.6"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of BARD1 expression in Nigerian cohort")

testgenename="ENSG00000197299.6"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of BLM expression in Nigerian cohort")

testgenename="ENSG00000012048.15"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of BRCA1 expression in Nigerian cohort")

testgenename="ENSG00000139618.10"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of BRCA2 expression in Nigerian cohort")

testgenename="ENSG00000158019.16"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of BRE expression in Nigerian cohort")

testgenename="ENSG00000136492.4"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of BRIP1 expression in Nigerian cohort")

testgenename="ENSG00000183765.16"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of CHEK2 expression in Nigerian cohort")

testgenename="ENSG00000154920.10"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of EME1 expression in Nigerian cohort")

testgenename="ENSG00000163322.9"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of FAM175A expression in Nigerian cohort")

testgenename="ENSG00000187741.10"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of FANCA expression in Nigerian cohort")

testgenename="ENSG00000187790.6"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of FANCM expression in Nigerian cohort")

testgenename="ENSG00000020922.8"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of MRE11A expression in Nigerian cohort")

testgenename="ENSG00000104320.9"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of NBN expression in Nigerian cohort")

testgenename="ENSG00000083093.5"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of PALB2 expression in Nigerian cohort")

testgenename="ENSG00000062822.8"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of POLD1 expression in Nigerian cohort")

testgenename="ENSG00000113522.9"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of RAD50 expression in Nigerian cohort")

testgenename="ENSG00000182185.14"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of RAD51B expression in Nigerian cohort")

testgenename="ENSG00000108384.10"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of RAD51C expression in Nigerian cohort")

testgenename="ENSG00000185379.16"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of RAD51D expression in Nigerian cohort")

testgenename="ENSG00000002016.12"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of RAD52 expression in Nigerian cohort")

testgenename="ENSG00000197275.8"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of RAD54B expression in Nigerian cohort")

testgenename="ENSG00000085999.7"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of RAD54L expression in Nigerian cohort")

testgenename="ENSG00000101773.12"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of RBBP8 expression in Nigerian cohort")

testgenename="ENSG00000132383.7"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of RPA1 expression in Nigerian cohort")

testgenename="ENSG00000139351.10"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of SYCP3 expression in Nigerian cohort")

testgenename="ENSG00000177302.10"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of TOP3A expression in Nigerian cohort")

testgenename="ENSG00000100038.15"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of TOP3B expression in Nigerian cohort")

testgenename="ENSG00000163781.8"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of TOPBP1 expression in Nigerian cohort")

testgenename="ENSG00000087206.12"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of UIMC1 expression in Nigerian cohort")

testgenename="ENSG00000196584.2"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of XRCC2 expression in Nigerian cohort")

testgenename="ENSG00000126215.9"
plotCounts(dds, gene = testgenename, intgroup="sampleCondition", main="Distribution of XRCC3 expression in Nigerian cohort")
```
